// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/artifactsSetVisibility.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useArtifactsSetVisibilityMutation = useArtifactsSetVisibilityMutation;
exports.mutationKeyArtifactsSetVisibility = mutationKeyArtifactsSetVisibility;
exports.buildArtifactsSetVisibilityMutation = buildArtifactsSetVisibilityMutation;
const react_query_1 = require("@tanstack/react-query");
const artifactsSetVisibility_js_1 = require("../funcs/artifactsSetVisibility.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Set visibility of a namespace with an existing metadata entry
 */
function useArtifactsSetVisibilityMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildArtifactsSetVisibilityMutation(client, options),
        ...options,
    });
}
function mutationKeyArtifactsSetVisibility() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Artifacts",
        "setVisibility",
    ];
}
function buildArtifactsSetVisibilityMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyArtifactsSetVisibility(),
        mutationFn: function artifactsSetVisibilityMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, artifactsSetVisibility_js_1.artifactsSetVisibility)(client$, request, mergedOptions));
        },
    };
}
