// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/artifactsPostTags.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useArtifactsPostTagsMutation = useArtifactsPostTagsMutation;
exports.mutationKeyArtifactsPostTags = mutationKeyArtifactsPostTags;
exports.buildArtifactsPostTagsMutation = buildArtifactsPostTagsMutation;
const react_query_1 = require("@tanstack/react-query");
const artifactsPostTags_js_1 = require("../funcs/artifactsPostTags.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Add tags to an existing revision
 */
function useArtifactsPostTagsMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildArtifactsPostTagsMutation(client, options),
        ...options,
    });
}
function mutationKeyArtifactsPostTags() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Artifacts",
        "postTags",
    ];
}
function buildArtifactsPostTagsMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyArtifactsPostTags(),
        mutationFn: function artifactsPostTagsMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, artifactsPostTags_js_1.artifactsPostTags)(client$, request, mergedOptions));
        },
    };
}
