// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/artifactsSetArchived.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useArtifactsSetArchivedMutation = useArtifactsSetArchivedMutation;
exports.mutationKeyArtifactsSetArchived = mutationKeyArtifactsSetArchived;
exports.buildArtifactsSetArchivedMutation = buildArtifactsSetArchivedMutation;
const react_query_1 = require("@tanstack/react-query");
const artifactsSetArchived_js_1 = require("../funcs/artifactsSetArchived.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Set whether a namespace is archived
 */
function useArtifactsSetArchivedMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildArtifactsSetArchivedMutation(client, options),
        ...options,
    });
}
function mutationKeyArtifactsSetArchived() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Artifacts",
        "setArchived",
    ];
}
function buildArtifactsSetArchivedMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyArtifactsSetArchived(),
        mutationFn: function artifactsSetArchivedMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, artifactsSetArchived_js_1.artifactsSetArchived)(client$, request, mergedOptions));
        },
    };
}
