// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/artifactsCreateRemoteSource.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useArtifactsCreateRemoteSourceMutation = useArtifactsCreateRemoteSourceMutation;
exports.mutationKeyArtifactsCreateRemoteSource = mutationKeyArtifactsCreateRemoteSource;
exports.buildArtifactsCreateRemoteSourceMutation = buildArtifactsCreateRemoteSourceMutation;
const react_query_1 = require("@tanstack/react-query");
const artifactsCreateRemoteSource_js_1 = require("../funcs/artifactsCreateRemoteSource.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Configure a new remote source
 */
function useArtifactsCreateRemoteSourceMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildArtifactsCreateRemoteSourceMutation(client, options),
        ...options,
    });
}
function mutationKeyArtifactsCreateRemoteSource() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Artifacts",
        "createRemoteSource",
    ];
}
function buildArtifactsCreateRemoteSourceMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyArtifactsCreateRemoteSource(),
        mutationFn: function artifactsCreateRemoteSourceMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, artifactsCreateRemoteSource_js_1.artifactsCreateRemoteSource)(client$, request, mergedOptions));
        },
    };
}
