// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/artifactsGetRevisions.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useArtifactsGetRevisions = useArtifactsGetRevisions;
exports.useArtifactsGetRevisionsSuspense = useArtifactsGetRevisionsSuspense;
exports.prefetchArtifactsGetRevisions = prefetchArtifactsGetRevisions;
exports.setArtifactsGetRevisionsData = setArtifactsGetRevisionsData;
exports.invalidateArtifactsGetRevisions = invalidateArtifactsGetRevisions;
exports.invalidateAllArtifactsGetRevisions = invalidateAllArtifactsGetRevisions;
exports.buildArtifactsGetRevisionsQuery = buildArtifactsGetRevisionsQuery;
exports.queryKeyArtifactsGetRevisions = queryKeyArtifactsGetRevisions;
const react_query_1 = require("@tanstack/react-query");
const artifactsGetRevisions_js_1 = require("../funcs/artifactsGetRevisions.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
function useArtifactsGetRevisions(request, options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useQuery)({
        ...buildArtifactsGetRevisionsQuery(client, request, options),
        ...options,
    });
}
function useArtifactsGetRevisionsSuspense(request, options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useSuspenseQuery)({
        ...buildArtifactsGetRevisionsQuery(client, request, options),
        ...options,
    });
}
function prefetchArtifactsGetRevisions(queryClient, client$, request) {
    return queryClient.prefetchQuery({
        ...buildArtifactsGetRevisionsQuery(client$, request),
    });
}
function setArtifactsGetRevisionsData(client, queryKeyBase, data) {
    const key = queryKeyArtifactsGetRevisions(...queryKeyBase);
    return client.setQueryData(key, data);
}
function invalidateArtifactsGetRevisions(client, queryKeyBase, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: [
            "@speakeasy-api/speakeasy-client-sdk-typescript",
            "Artifacts",
            "getRevisions",
            ...queryKeyBase,
        ],
    });
}
function invalidateAllArtifactsGetRevisions(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: [
            "@speakeasy-api/speakeasy-client-sdk-typescript",
            "Artifacts",
            "getRevisions",
        ],
    });
}
function buildArtifactsGetRevisionsQuery(client$, request, options) {
    return {
        queryKey: queryKeyArtifactsGetRevisions(request.namespaceName, {
            nextPageToken: request.nextPageToken,
        }),
        queryFn: async function artifactsGetRevisionsQueryFn(ctx) {
            const sig = (0, primitives_js_1.combineSignals)(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return (0, fp_js_1.unwrapAsync)((0, artifactsGetRevisions_js_1.artifactsGetRevisions)(client$, request, mergedOptions));
        },
    };
}
function queryKeyArtifactsGetRevisions(namespaceName, parameters) {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Artifacts",
        "getRevisions",
        namespaceName,
        parameters,
    ];
}
