// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/artifactsGetNamespaces.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useArtifactsGetNamespaces = useArtifactsGetNamespaces;
exports.useArtifactsGetNamespacesSuspense = useArtifactsGetNamespacesSuspense;
exports.prefetchArtifactsGetNamespaces = prefetchArtifactsGetNamespaces;
exports.setArtifactsGetNamespacesData = setArtifactsGetNamespacesData;
exports.invalidateAllArtifactsGetNamespaces = invalidateAllArtifactsGetNamespaces;
exports.buildArtifactsGetNamespacesQuery = buildArtifactsGetNamespacesQuery;
exports.queryKeyArtifactsGetNamespaces = queryKeyArtifactsGetNamespaces;
const react_query_1 = require("@tanstack/react-query");
const artifactsGetNamespaces_js_1 = require("../funcs/artifactsGetNamespaces.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Each namespace contains many revisions.
 */
function useArtifactsGetNamespaces(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useQuery)({
        ...buildArtifactsGetNamespacesQuery(client, options),
        ...options,
    });
}
/**
 * Each namespace contains many revisions.
 */
function useArtifactsGetNamespacesSuspense(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useSuspenseQuery)({
        ...buildArtifactsGetNamespacesQuery(client, options),
        ...options,
    });
}
function prefetchArtifactsGetNamespaces(queryClient, client$) {
    return queryClient.prefetchQuery({
        ...buildArtifactsGetNamespacesQuery(client$),
    });
}
function setArtifactsGetNamespacesData(client, data) {
    const key = queryKeyArtifactsGetNamespaces();
    return client.setQueryData(key, data);
}
function invalidateAllArtifactsGetNamespaces(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: [
            "@speakeasy-api/speakeasy-client-sdk-typescript",
            "Artifacts",
            "getNamespaces",
        ],
    });
}
function buildArtifactsGetNamespacesQuery(client$, options) {
    return {
        queryKey: queryKeyArtifactsGetNamespaces(),
        queryFn: async function artifactsGetNamespacesQueryFn(ctx) {
            const sig = (0, primitives_js_1.combineSignals)(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return (0, fp_js_1.unwrapAsync)((0, artifactsGetNamespaces_js_1.artifactsGetNamespaces)(client$, mergedOptions));
        },
    };
}
function queryKeyArtifactsGetNamespaces() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Artifacts",
        "getNamespaces",
    ];
}
