// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/artifactsPreflight.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useArtifactsPreflightMutation = useArtifactsPreflightMutation;
exports.mutationKeyArtifactsPreflight = mutationKeyArtifactsPreflight;
exports.buildArtifactsPreflightMutation = buildArtifactsPreflightMutation;
const react_query_1 = require("@tanstack/react-query");
const artifactsPreflight_js_1 = require("../funcs/artifactsPreflight.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Get access token for communicating with OCI distribution endpoints
 */
function useArtifactsPreflightMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildArtifactsPreflightMutation(client, options),
        ...options,
    });
}
function mutationKeyArtifactsPreflight() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Artifacts",
        "preflight",
    ];
}
function buildArtifactsPreflightMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyArtifactsPreflight(),
        mutationFn: function artifactsPreflightMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, artifactsPreflight_js_1.artifactsPreflight)(client$, request, mergedOptions));
        },
    };
}
