// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/codeSamplesGenerateCodeSamplePreviewAsync.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useCodeSamplesGenerateCodeSamplePreviewAsyncMutation = useCodeSamplesGenerateCodeSamplePreviewAsyncMutation;
exports.mutationKeyCodeSamplesGenerateCodeSamplePreviewAsync = mutationKeyCodeSamplesGenerateCodeSamplePreviewAsync;
exports.buildCodeSamplesGenerateCodeSamplePreviewAsyncMutation = buildCodeSamplesGenerateCodeSamplePreviewAsyncMutation;
const react_query_1 = require("@tanstack/react-query");
const codeSamplesGenerateCodeSamplePreviewAsync_js_1 = require("../funcs/codeSamplesGenerateCodeSamplePreviewAsync.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Initiate asynchronous Code Sample preview generation from a file and configuration parameters, receiving an async JobID response for polling.
 *
 * @remarks
 * This endpoint generates Code Sample previews from a file and configuration parameters, receiving an async JobID response for polling.
 */
function useCodeSamplesGenerateCodeSamplePreviewAsyncMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildCodeSamplesGenerateCodeSamplePreviewAsyncMutation(client, options),
        ...options,
    });
}
function mutationKeyCodeSamplesGenerateCodeSamplePreviewAsync() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "CodeSamples",
        "generateCodeSamplePreviewAsync",
    ];
}
function buildCodeSamplesGenerateCodeSamplePreviewAsyncMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyCodeSamplesGenerateCodeSamplePreviewAsync(),
        mutationFn: function codeSamplesGenerateCodeSamplePreviewAsyncMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, codeSamplesGenerateCodeSamplePreviewAsync_js_1.codeSamplesGenerateCodeSamplePreviewAsync)(client$, request, mergedOptions));
        },
    };
}
