// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/eventsGetTargets.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useEventsGetTargets = useEventsGetTargets;
exports.useEventsGetTargetsSuspense = useEventsGetTargetsSuspense;
exports.prefetchEventsGetTargets = prefetchEventsGetTargets;
exports.setEventsGetTargetsData = setEventsGetTargetsData;
exports.invalidateEventsGetTargets = invalidateEventsGetTargets;
exports.invalidateAllEventsGetTargets = invalidateAllEventsGetTargets;
exports.buildEventsGetTargetsQuery = buildEventsGetTargetsQuery;
exports.queryKeyEventsGetTargets = queryKeyEventsGetTargets;
const react_query_1 = require("@tanstack/react-query");
const eventsGetTargets_js_1 = require("../funcs/eventsGetTargets.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Load targets for a particular workspace
 */
function useEventsGetTargets(request, options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useQuery)({
        ...buildEventsGetTargetsQuery(client, request, options),
        ...options,
    });
}
/**
 * Load targets for a particular workspace
 */
function useEventsGetTargetsSuspense(request, options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useSuspenseQuery)({
        ...buildEventsGetTargetsQuery(client, request, options),
        ...options,
    });
}
function prefetchEventsGetTargets(queryClient, client$, request) {
    return queryClient.prefetchQuery({
        ...buildEventsGetTargetsQuery(client$, request),
    });
}
function setEventsGetTargetsData(client, queryKeyBase, data) {
    const key = queryKeyEventsGetTargets(...queryKeyBase);
    return client.setQueryData(key, data);
}
function invalidateEventsGetTargets(client, queryKeyBase, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: [
            "@speakeasy-api/speakeasy-client-sdk-typescript",
            "Events",
            "getTargets",
            ...queryKeyBase,
        ],
    });
}
function invalidateAllEventsGetTargets(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: [
            "@speakeasy-api/speakeasy-client-sdk-typescript",
            "Events",
            "getTargets",
        ],
    });
}
function buildEventsGetTargetsQuery(client$, request, options) {
    return {
        queryKey: queryKeyEventsGetTargets({
            afterLastEventCreatedAt: request.afterLastEventCreatedAt,
        }),
        queryFn: async function eventsGetTargetsQueryFn(ctx) {
            const sig = (0, primitives_js_1.combineSignals)(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return (0, fp_js_1.unwrapAsync)((0, eventsGetTargets_js_1.eventsGetTargets)(client$, request, mergedOptions));
        },
    };
}
function queryKeyEventsGetTargets(parameters) {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Events",
        "getTargets",
        parameters,
    ];
}
