// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/eventsPost.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useEventsPostMutation = useEventsPostMutation;
exports.mutationKeyEventsPost = mutationKeyEventsPost;
exports.buildEventsPostMutation = buildEventsPostMutation;
const react_query_1 = require("@tanstack/react-query");
const eventsPost_js_1 = require("../funcs/eventsPost.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Post events for a specific workspace
 *
 * @remarks
 * Sends an array of events to be stored for a particular workspace.
 */
function useEventsPostMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildEventsPostMutation(client, options),
        ...options,
    });
}
function mutationKeyEventsPost() {
    return ["@speakeasy-api/speakeasy-client-sdk-typescript", "Events", "post"];
}
function buildEventsPostMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyEventsPost(),
        mutationFn: function eventsPostMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, eventsPost_js_1.eventsPost)(client$, request, mergedOptions));
        },
    };
}
