// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/githubConfigureMintlifyRepo.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useGithubConfigureMintlifyRepoMutation = useGithubConfigureMintlifyRepoMutation;
exports.mutationKeyGithubConfigureMintlifyRepo = mutationKeyGithubConfigureMintlifyRepo;
exports.buildGithubConfigureMintlifyRepoMutation = buildGithubConfigureMintlifyRepoMutation;
const react_query_1 = require("@tanstack/react-query");
const githubConfigureMintlifyRepo_js_1 = require("../funcs/githubConfigureMintlifyRepo.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
function useGithubConfigureMintlifyRepoMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildGithubConfigureMintlifyRepoMutation(client, options),
        ...options,
    });
}
function mutationKeyGithubConfigureMintlifyRepo() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Github",
        "configureMintlifyRepo",
    ];
}
function buildGithubConfigureMintlifyRepoMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyGithubConfigureMintlifyRepo(),
        mutationFn: function githubConfigureMintlifyRepoMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, githubConfigureMintlifyRepo_js_1.githubConfigureMintlifyRepo)(client$, request, mergedOptions));
        },
    };
}
