// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/githubTriggerAction.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useGithubTriggerActionMutation = useGithubTriggerActionMutation;
exports.mutationKeyGithubTriggerAction = mutationKeyGithubTriggerAction;
exports.buildGithubTriggerActionMutation = buildGithubTriggerActionMutation;
const react_query_1 = require("@tanstack/react-query");
const githubTriggerAction_js_1 = require("../funcs/githubTriggerAction.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
function useGithubTriggerActionMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildGithubTriggerActionMutation(client, options),
        ...options,
    });
}
function mutationKeyGithubTriggerAction() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Github",
        "triggerAction",
    ];
}
function buildGithubTriggerActionMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyGithubTriggerAction(),
        mutationFn: function githubTriggerActionMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, githubTriggerAction_js_1.githubTriggerAction)(client$, request, mergedOptions));
        },
    };
}
