// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/githubCheckPublishingSecrets.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useGithubCheckPublishingSecrets = useGithubCheckPublishingSecrets;
exports.useGithubCheckPublishingSecretsSuspense = useGithubCheckPublishingSecretsSuspense;
exports.prefetchGithubCheckPublishingSecrets = prefetchGithubCheckPublishingSecrets;
exports.setGithubCheckPublishingSecretsData = setGithubCheckPublishingSecretsData;
exports.invalidateGithubCheckPublishingSecrets = invalidateGithubCheckPublishingSecrets;
exports.invalidateAllGithubCheckPublishingSecrets = invalidateAllGithubCheckPublishingSecrets;
exports.buildGithubCheckPublishingSecretsQuery = buildGithubCheckPublishingSecretsQuery;
exports.queryKeyGithubCheckPublishingSecrets = queryKeyGithubCheckPublishingSecrets;
const react_query_1 = require("@tanstack/react-query");
const githubCheckPublishingSecrets_js_1 = require("../funcs/githubCheckPublishingSecrets.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
function useGithubCheckPublishingSecrets(request, options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useQuery)({
        ...buildGithubCheckPublishingSecretsQuery(client, request, options),
        ...options,
    });
}
function useGithubCheckPublishingSecretsSuspense(request, options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useSuspenseQuery)({
        ...buildGithubCheckPublishingSecretsQuery(client, request, options),
        ...options,
    });
}
function prefetchGithubCheckPublishingSecrets(queryClient, client$, request) {
    return queryClient.prefetchQuery({
        ...buildGithubCheckPublishingSecretsQuery(client$, request),
    });
}
function setGithubCheckPublishingSecretsData(client, queryKeyBase, data) {
    const key = queryKeyGithubCheckPublishingSecrets(...queryKeyBase);
    return client.setQueryData(key, data);
}
function invalidateGithubCheckPublishingSecrets(client, queryKeyBase, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: [
            "@speakeasy-api/speakeasy-client-sdk-typescript",
            "Github",
            "checkPublishingSecrets",
            ...queryKeyBase,
        ],
    });
}
function invalidateAllGithubCheckPublishingSecrets(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: [
            "@speakeasy-api/speakeasy-client-sdk-typescript",
            "Github",
            "checkPublishingSecrets",
        ],
    });
}
function buildGithubCheckPublishingSecretsQuery(client$, request, options) {
    return {
        queryKey: queryKeyGithubCheckPublishingSecrets({
            generateGenLockId: request.generateGenLockId,
        }),
        queryFn: async function githubCheckPublishingSecretsQueryFn(ctx) {
            const sig = (0, primitives_js_1.combineSignals)(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return (0, fp_js_1.unwrapAsync)((0, githubCheckPublishingSecrets_js_1.githubCheckPublishingSecrets)(client$, request, mergedOptions));
        },
    };
}
function queryKeyGithubCheckPublishingSecrets(parameters) {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Github",
        "checkPublishingSecrets",
        parameters,
    ];
}
