// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/githubConfigureCodeSamples.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useGithubConfigureCodeSamplesMutation = useGithubConfigureCodeSamplesMutation;
exports.mutationKeyGithubConfigureCodeSamples = mutationKeyGithubConfigureCodeSamples;
exports.buildGithubConfigureCodeSamplesMutation = buildGithubConfigureCodeSamplesMutation;
const react_query_1 = require("@tanstack/react-query");
const githubConfigureCodeSamples_js_1 = require("../funcs/githubConfigureCodeSamples.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
function useGithubConfigureCodeSamplesMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildGithubConfigureCodeSamplesMutation(client, options),
        ...options,
    });
}
function mutationKeyGithubConfigureCodeSamples() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Github",
        "configureCodeSamples",
    ];
}
function buildGithubConfigureCodeSamplesMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyGithubConfigureCodeSamples(),
        mutationFn: function githubConfigureCodeSamplesMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, githubConfigureCodeSamples_js_1.githubConfigureCodeSamples)(client$, request, mergedOptions));
        },
    };
}
