// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/githubLinkGithub.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useGithubLinkGithubMutation = useGithubLinkGithubMutation;
exports.mutationKeyGithubLinkGithub = mutationKeyGithubLinkGithub;
exports.buildGithubLinkGithubMutation = buildGithubLinkGithubMutation;
const react_query_1 = require("@tanstack/react-query");
const githubLinkGithub_js_1 = require("../funcs/githubLinkGithub.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
function useGithubLinkGithubMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildGithubLinkGithubMutation(client, options),
        ...options,
    });
}
function mutationKeyGithubLinkGithub() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Github",
        "linkGithub",
    ];
}
function buildGithubLinkGithubMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyGithubLinkGithub(),
        mutationFn: function githubLinkGithubMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, githubLinkGithub_js_1.githubLinkGithub)(client$, request, mergedOptions));
        },
    };
}
