// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/githubStorePublishingSecrets.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useGithubStorePublishingSecretsMutation = useGithubStorePublishingSecretsMutation;
exports.mutationKeyGithubStorePublishingSecrets = mutationKeyGithubStorePublishingSecrets;
exports.buildGithubStorePublishingSecretsMutation = buildGithubStorePublishingSecretsMutation;
const react_query_1 = require("@tanstack/react-query");
const githubStorePublishingSecrets_js_1 = require("../funcs/githubStorePublishingSecrets.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
function useGithubStorePublishingSecretsMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildGithubStorePublishingSecretsMutation(client, options),
        ...options,
    });
}
function mutationKeyGithubStorePublishingSecrets() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Github",
        "storePublishingSecrets",
    ];
}
function buildGithubStorePublishingSecretsMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyGithubStorePublishingSecrets(),
        mutationFn: function githubStorePublishingSecretsMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, githubStorePublishingSecrets_js_1.githubStorePublishingSecrets)(client$, request, mergedOptions));
        },
    };
}
