// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/githubConfigureTarget.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useGithubConfigureTargetMutation = useGithubConfigureTargetMutation;
exports.mutationKeyGithubConfigureTarget = mutationKeyGithubConfigureTarget;
exports.buildGithubConfigureTargetMutation = buildGithubConfigureTargetMutation;
const react_query_1 = require("@tanstack/react-query");
const githubConfigureTarget_js_1 = require("../funcs/githubConfigureTarget.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
function useGithubConfigureTargetMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildGithubConfigureTargetMutation(client, options),
        ...options,
    });
}
function mutationKeyGithubConfigureTarget() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Github",
        "configureTarget",
    ];
}
function buildGithubConfigureTargetMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyGithubConfigureTarget(),
        mutationFn: function githubConfigureTargetMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, githubConfigureTarget_js_1.githubConfigureTarget)(client$, request, mergedOptions));
        },
    };
}
