// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/githubCheckAccess.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useGithubCheckAccess = useGithubCheckAccess;
exports.useGithubCheckAccessSuspense = useGithubCheckAccessSuspense;
exports.prefetchGithubCheckAccess = prefetchGithubCheckAccess;
exports.setGithubCheckAccessData = setGithubCheckAccessData;
exports.invalidateGithubCheckAccess = invalidateGithubCheckAccess;
exports.invalidateAllGithubCheckAccess = invalidateAllGithubCheckAccess;
exports.buildGithubCheckAccessQuery = buildGithubCheckAccessQuery;
exports.queryKeyGithubCheckAccess = queryKeyGithubCheckAccess;
const react_query_1 = require("@tanstack/react-query");
const githubCheckAccess_js_1 = require("../funcs/githubCheckAccess.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
function useGithubCheckAccess(request, options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useQuery)({
        ...buildGithubCheckAccessQuery(client, request, options),
        ...options,
    });
}
function useGithubCheckAccessSuspense(request, options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useSuspenseQuery)({
        ...buildGithubCheckAccessQuery(client, request, options),
        ...options,
    });
}
function prefetchGithubCheckAccess(queryClient, client$, request) {
    return queryClient.prefetchQuery({
        ...buildGithubCheckAccessQuery(client$, request),
    });
}
function setGithubCheckAccessData(client, queryKeyBase, data) {
    const key = queryKeyGithubCheckAccess(...queryKeyBase);
    return client.setQueryData(key, data);
}
function invalidateGithubCheckAccess(client, queryKeyBase, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: [
            "@speakeasy-api/speakeasy-client-sdk-typescript",
            "Github",
            "checkAccess",
            ...queryKeyBase,
        ],
    });
}
function invalidateAllGithubCheckAccess(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: [
            "@speakeasy-api/speakeasy-client-sdk-typescript",
            "Github",
            "checkAccess",
        ],
    });
}
function buildGithubCheckAccessQuery(client$, request, options) {
    return {
        queryKey: queryKeyGithubCheckAccess({
            org: request.org,
            repo: request.repo,
        }),
        queryFn: async function githubCheckAccessQueryFn(ctx) {
            const sig = (0, primitives_js_1.combineSignals)(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return (0, fp_js_1.unwrapAsync)((0, githubCheckAccess_js_1.githubCheckAccess)(client$, request, mergedOptions));
        },
    };
}
function queryKeyGithubCheckAccess(parameters) {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Github",
        "checkAccess",
        parameters,
    ];
}
