// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/organizationsCreateFreeTrial.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useOrganizationsCreateFreeTrialMutation = useOrganizationsCreateFreeTrialMutation;
exports.mutationKeyOrganizationsCreateFreeTrial = mutationKeyOrganizationsCreateFreeTrial;
exports.buildOrganizationsCreateFreeTrialMutation = buildOrganizationsCreateFreeTrialMutation;
const react_query_1 = require("@tanstack/react-query");
const organizationsCreateFreeTrial_js_1 = require("../funcs/organizationsCreateFreeTrial.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Create a free trial for an organization
 *
 * @remarks
 * Creates a free trial for an organization
 */
function useOrganizationsCreateFreeTrialMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildOrganizationsCreateFreeTrialMutation(client, options),
        ...options,
    });
}
function mutationKeyOrganizationsCreateFreeTrial() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Organizations",
        "createFreeTrial",
    ];
}
function buildOrganizationsCreateFreeTrialMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyOrganizationsCreateFreeTrial(),
        mutationFn: function organizationsCreateFreeTrialMutationFn({ options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, organizationsCreateFreeTrial_js_1.organizationsCreateFreeTrial)(client$, mergedOptions));
        },
    };
}
