// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/organizationsCreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useOrganizationsCreateMutation = useOrganizationsCreateMutation;
exports.mutationKeyOrganizationsCreate = mutationKeyOrganizationsCreate;
exports.buildOrganizationsCreateMutation = buildOrganizationsCreateMutation;
const react_query_1 = require("@tanstack/react-query");
const organizationsCreate_js_1 = require("../funcs/organizationsCreate.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Create an organization
 *
 * @remarks
 * Creates an organization
 */
function useOrganizationsCreateMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildOrganizationsCreateMutation(client, options),
        ...options,
    });
}
function mutationKeyOrganizationsCreate() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Organizations",
        "create",
    ];
}
function buildOrganizationsCreateMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyOrganizationsCreate(),
        mutationFn: function organizationsCreateMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, organizationsCreate_js_1.organizationsCreate)(client$, request, mergedOptions));
        },
    };
}
