// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/organizationsCreateBillingAddOns.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useOrganizationsCreateBillingAddOnsMutation = useOrganizationsCreateBillingAddOnsMutation;
exports.mutationKeyOrganizationsCreateBillingAddOns = mutationKeyOrganizationsCreateBillingAddOns;
exports.buildOrganizationsCreateBillingAddOnsMutation = buildOrganizationsCreateBillingAddOnsMutation;
const react_query_1 = require("@tanstack/react-query");
const organizationsCreateBillingAddOns_js_1 = require("../funcs/organizationsCreateBillingAddOns.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Create billing add ons
 */
function useOrganizationsCreateBillingAddOnsMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildOrganizationsCreateBillingAddOnsMutation(client, options),
        ...options,
    });
}
function mutationKeyOrganizationsCreateBillingAddOns() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Organizations",
        "createBillingAddOns",
    ];
}
function buildOrganizationsCreateBillingAddOnsMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyOrganizationsCreateBillingAddOns(),
        mutationFn: function organizationsCreateBillingAddOnsMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, organizationsCreateBillingAddOns_js_1.organizationsCreateBillingAddOns)(client$, request, mergedOptions));
        },
    };
}
