// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/organizationsDeleteBillingAddOn.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useOrganizationsDeleteBillingAddOnMutation = useOrganizationsDeleteBillingAddOnMutation;
exports.mutationKeyOrganizationsDeleteBillingAddOn = mutationKeyOrganizationsDeleteBillingAddOn;
exports.buildOrganizationsDeleteBillingAddOnMutation = buildOrganizationsDeleteBillingAddOnMutation;
const react_query_1 = require("@tanstack/react-query");
const organizationsDeleteBillingAddOn_js_1 = require("../funcs/organizationsDeleteBillingAddOn.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Delete billing add ons
 */
function useOrganizationsDeleteBillingAddOnMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildOrganizationsDeleteBillingAddOnMutation(client, options),
        ...options,
    });
}
function mutationKeyOrganizationsDeleteBillingAddOn() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Organizations",
        "deleteBillingAddOn",
    ];
}
function buildOrganizationsDeleteBillingAddOnMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyOrganizationsDeleteBillingAddOn(),
        mutationFn: function organizationsDeleteBillingAddOnMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, organizationsDeleteBillingAddOn_js_1.organizationsDeleteBillingAddOn)(client$, request, mergedOptions));
        },
    };
}
