// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/organizationsGetBillingAddOns.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useOrganizationsGetBillingAddOns = useOrganizationsGetBillingAddOns;
exports.useOrganizationsGetBillingAddOnsSuspense = useOrganizationsGetBillingAddOnsSuspense;
exports.prefetchOrganizationsGetBillingAddOns = prefetchOrganizationsGetBillingAddOns;
exports.setOrganizationsGetBillingAddOnsData = setOrganizationsGetBillingAddOnsData;
exports.invalidateAllOrganizationsGetBillingAddOns = invalidateAllOrganizationsGetBillingAddOns;
exports.buildOrganizationsGetBillingAddOnsQuery = buildOrganizationsGetBillingAddOnsQuery;
exports.queryKeyOrganizationsGetBillingAddOns = queryKeyOrganizationsGetBillingAddOns;
const react_query_1 = require("@tanstack/react-query");
const organizationsGetBillingAddOns_js_1 = require("../funcs/organizationsGetBillingAddOns.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Get billing add ons
 */
function useOrganizationsGetBillingAddOns(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useQuery)({
        ...buildOrganizationsGetBillingAddOnsQuery(client, options),
        ...options,
    });
}
/**
 * Get billing add ons
 */
function useOrganizationsGetBillingAddOnsSuspense(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useSuspenseQuery)({
        ...buildOrganizationsGetBillingAddOnsQuery(client, options),
        ...options,
    });
}
function prefetchOrganizationsGetBillingAddOns(queryClient, client$) {
    return queryClient.prefetchQuery({
        ...buildOrganizationsGetBillingAddOnsQuery(client$),
    });
}
function setOrganizationsGetBillingAddOnsData(client, data) {
    const key = queryKeyOrganizationsGetBillingAddOns();
    return client.setQueryData(key, data);
}
function invalidateAllOrganizationsGetBillingAddOns(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: [
            "@speakeasy-api/speakeasy-client-sdk-typescript",
            "Organizations",
            "getBillingAddOns",
        ],
    });
}
function buildOrganizationsGetBillingAddOnsQuery(client$, options) {
    return {
        queryKey: queryKeyOrganizationsGetBillingAddOns(),
        queryFn: async function organizationsGetBillingAddOnsQueryFn(ctx) {
            const sig = (0, primitives_js_1.combineSignals)(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return (0, fp_js_1.unwrapAsync)((0, organizationsGetBillingAddOns_js_1.organizationsGetBillingAddOns)(client$, mergedOptions));
        },
    };
}
function queryKeyOrganizationsGetBillingAddOns() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Organizations",
        "getBillingAddOns",
    ];
}
