// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/publishingTokensDelete.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.usePublishingTokensDeleteMutation = usePublishingTokensDeleteMutation;
exports.mutationKeyPublishingTokensDelete = mutationKeyPublishingTokensDelete;
exports.buildPublishingTokensDeleteMutation = buildPublishingTokensDeleteMutation;
const react_query_1 = require("@tanstack/react-query");
const publishingTokensDelete_js_1 = require("../funcs/publishingTokensDelete.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Delete a specific publishing token
 *
 * @remarks
 * Delete a particular publishing token.
 */
function usePublishingTokensDeleteMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildPublishingTokensDeleteMutation(client, options),
        ...options,
    });
}
function mutationKeyPublishingTokensDelete() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "PublishingTokens",
        "delete",
    ];
}
function buildPublishingTokensDeleteMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyPublishingTokensDelete(),
        mutationFn: function publishingTokensDeleteMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, publishingTokensDelete_js_1.publishingTokensDelete)(client$, request, mergedOptions));
        },
    };
}
