// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/publishingTokensUpdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.usePublishingTokensUpdateMutation = usePublishingTokensUpdateMutation;
exports.mutationKeyPublishingTokensUpdate = mutationKeyPublishingTokensUpdate;
exports.buildPublishingTokensUpdateMutation = buildPublishingTokensUpdateMutation;
const react_query_1 = require("@tanstack/react-query");
const publishingTokensUpdate_js_1 = require("../funcs/publishingTokensUpdate.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Updates the validitity period of a publishing token
 *
 * @remarks
 * Updates the validity period of a particular publishing token.
 */
function usePublishingTokensUpdateMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildPublishingTokensUpdateMutation(client, options),
        ...options,
    });
}
function mutationKeyPublishingTokensUpdate() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "PublishingTokens",
        "update",
    ];
}
function buildPublishingTokensUpdateMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyPublishingTokensUpdate(),
        mutationFn: function publishingTokensUpdateMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, publishingTokensUpdate_js_1.publishingTokensUpdate)(client$, request, mergedOptions));
        },
    };
}
