// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/publishingTokensResolveTarget.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.usePublishingTokensResolveTarget = usePublishingTokensResolveTarget;
exports.usePublishingTokensResolveTargetSuspense = usePublishingTokensResolveTargetSuspense;
exports.prefetchPublishingTokensResolveTarget = prefetchPublishingTokensResolveTarget;
exports.setPublishingTokensResolveTargetData = setPublishingTokensResolveTargetData;
exports.invalidatePublishingTokensResolveTarget = invalidatePublishingTokensResolveTarget;
exports.invalidateAllPublishingTokensResolveTarget = invalidateAllPublishingTokensResolveTarget;
exports.buildPublishingTokensResolveTargetQuery = buildPublishingTokensResolveTargetQuery;
exports.queryKeyPublishingTokensResolveTarget = queryKeyPublishingTokensResolveTarget;
const react_query_1 = require("@tanstack/react-query");
const publishingTokensResolveTarget_js_1 = require("../funcs/publishingTokensResolveTarget.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Get a specific publishing token target
 *
 * @remarks
 * Get information about a particular publishing token target.
 */
function usePublishingTokensResolveTarget(request, options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useQuery)({
        ...buildPublishingTokensResolveTargetQuery(client, request, options),
        ...options,
    });
}
/**
 * Get a specific publishing token target
 *
 * @remarks
 * Get information about a particular publishing token target.
 */
function usePublishingTokensResolveTargetSuspense(request, options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useSuspenseQuery)({
        ...buildPublishingTokensResolveTargetQuery(client, request, options),
        ...options,
    });
}
function prefetchPublishingTokensResolveTarget(queryClient, client$, request) {
    return queryClient.prefetchQuery({
        ...buildPublishingTokensResolveTargetQuery(client$, request),
    });
}
function setPublishingTokensResolveTargetData(client, queryKeyBase, data) {
    const key = queryKeyPublishingTokensResolveTarget(...queryKeyBase);
    return client.setQueryData(key, data);
}
function invalidatePublishingTokensResolveTarget(client, queryKeyBase, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: [
            "@speakeasy-api/speakeasy-client-sdk-typescript",
            "PublishingTokens",
            "resolveTarget",
            ...queryKeyBase,
        ],
    });
}
function invalidateAllPublishingTokensResolveTarget(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: [
            "@speakeasy-api/speakeasy-client-sdk-typescript",
            "PublishingTokens",
            "resolveTarget",
        ],
    });
}
function buildPublishingTokensResolveTargetQuery(client$, request, options) {
    return {
        queryKey: queryKeyPublishingTokensResolveTarget(request.tokenID),
        queryFn: async function publishingTokensResolveTargetQueryFn(ctx) {
            const sig = (0, primitives_js_1.combineSignals)(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return (0, fp_js_1.unwrapAsync)((0, publishingTokensResolveTarget_js_1.publishingTokensResolveTarget)(client$, request, mergedOptions));
        },
    };
}
function queryKeyPublishingTokensResolveTarget(tokenID) {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "PublishingTokens",
        "resolveTarget",
        tokenID,
    ];
}
