// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/publishingTokensGet.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.usePublishingTokensGet = usePublishingTokensGet;
exports.usePublishingTokensGetSuspense = usePublishingTokensGetSuspense;
exports.prefetchPublishingTokensGet = prefetchPublishingTokensGet;
exports.setPublishingTokensGetData = setPublishingTokensGetData;
exports.invalidatePublishingTokensGet = invalidatePublishingTokensGet;
exports.invalidateAllPublishingTokensGet = invalidateAllPublishingTokensGet;
exports.buildPublishingTokensGetQuery = buildPublishingTokensGetQuery;
exports.queryKeyPublishingTokensGet = queryKeyPublishingTokensGet;
const react_query_1 = require("@tanstack/react-query");
const publishingTokensGet_js_1 = require("../funcs/publishingTokensGet.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Get a specific publishing token
 *
 * @remarks
 * Get information about a particular publishing token.
 */
function usePublishingTokensGet(request, options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useQuery)({
        ...buildPublishingTokensGetQuery(client, request, options),
        ...options,
    });
}
/**
 * Get a specific publishing token
 *
 * @remarks
 * Get information about a particular publishing token.
 */
function usePublishingTokensGetSuspense(request, options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useSuspenseQuery)({
        ...buildPublishingTokensGetQuery(client, request, options),
        ...options,
    });
}
function prefetchPublishingTokensGet(queryClient, client$, request) {
    return queryClient.prefetchQuery({
        ...buildPublishingTokensGetQuery(client$, request),
    });
}
function setPublishingTokensGetData(client, queryKeyBase, data) {
    const key = queryKeyPublishingTokensGet(...queryKeyBase);
    return client.setQueryData(key, data);
}
function invalidatePublishingTokensGet(client, queryKeyBase, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: [
            "@speakeasy-api/speakeasy-client-sdk-typescript",
            "PublishingTokens",
            "get",
            ...queryKeyBase,
        ],
    });
}
function invalidateAllPublishingTokensGet(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: [
            "@speakeasy-api/speakeasy-client-sdk-typescript",
            "PublishingTokens",
            "get",
        ],
    });
}
function buildPublishingTokensGetQuery(client$, request, options) {
    return {
        queryKey: queryKeyPublishingTokensGet(request.tokenID),
        queryFn: async function publishingTokensGetQueryFn(ctx) {
            const sig = (0, primitives_js_1.combineSignals)(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return (0, fp_js_1.unwrapAsync)((0, publishingTokensGet_js_1.publishingTokensGet)(client$, request, mergedOptions));
        },
    };
}
function queryKeyPublishingTokensGet(tokenID) {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "PublishingTokens",
        "get",
        tokenID,
    ];
}
