// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/publishingTokensList.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.usePublishingTokensList = usePublishingTokensList;
exports.usePublishingTokensListSuspense = usePublishingTokensListSuspense;
exports.prefetchPublishingTokensList = prefetchPublishingTokensList;
exports.setPublishingTokensListData = setPublishingTokensListData;
exports.invalidateAllPublishingTokensList = invalidateAllPublishingTokensList;
exports.buildPublishingTokensListQuery = buildPublishingTokensListQuery;
exports.queryKeyPublishingTokensList = queryKeyPublishingTokensList;
const react_query_1 = require("@tanstack/react-query");
const publishingTokensList_js_1 = require("../funcs/publishingTokensList.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Get publishing tokens for a workspace
 *
 * @remarks
 * Returns a publishing token for the current workspace
 */
function usePublishingTokensList(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useQuery)({
        ...buildPublishingTokensListQuery(client, options),
        ...options,
    });
}
/**
 * Get publishing tokens for a workspace
 *
 * @remarks
 * Returns a publishing token for the current workspace
 */
function usePublishingTokensListSuspense(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useSuspenseQuery)({
        ...buildPublishingTokensListQuery(client, options),
        ...options,
    });
}
function prefetchPublishingTokensList(queryClient, client$) {
    return queryClient.prefetchQuery({
        ...buildPublishingTokensListQuery(client$),
    });
}
function setPublishingTokensListData(client, data) {
    const key = queryKeyPublishingTokensList();
    return client.setQueryData(key, data);
}
function invalidateAllPublishingTokensList(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: [
            "@speakeasy-api/speakeasy-client-sdk-typescript",
            "PublishingTokens",
            "list",
        ],
    });
}
function buildPublishingTokensListQuery(client$, options) {
    return {
        queryKey: queryKeyPublishingTokensList(),
        queryFn: async function publishingTokensListQueryFn(ctx) {
            const sig = (0, primitives_js_1.combineSignals)(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return (0, fp_js_1.unwrapAsync)((0, publishingTokensList_js_1.publishingTokensList)(client$, mergedOptions));
        },
    };
}
function queryKeyPublishingTokensList() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "PublishingTokens",
        "list",
    ];
}
