// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/publishingTokensCreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.usePublishingTokensCreateMutation = usePublishingTokensCreateMutation;
exports.mutationKeyPublishingTokensCreate = mutationKeyPublishingTokensCreate;
exports.buildPublishingTokensCreateMutation = buildPublishingTokensCreateMutation;
const react_query_1 = require("@tanstack/react-query");
const publishingTokensCreate_js_1 = require("../funcs/publishingTokensCreate.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Create a publishing token for a workspace
 *
 * @remarks
 * Creates a publishing token for the current workspace
 */
function usePublishingTokensCreateMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildPublishingTokensCreateMutation(client, options),
        ...options,
    });
}
function mutationKeyPublishingTokensCreate() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "PublishingTokens",
        "create",
    ];
}
function buildPublishingTokensCreateMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyPublishingTokensCreate(),
        mutationFn: function publishingTokensCreateMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, publishingTokensCreate_js_1.publishingTokensCreate)(client$, request, mergedOptions));
        },
    };
}
