// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/reportsUploadReport.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useReportsUploadReportMutation = useReportsUploadReportMutation;
exports.mutationKeyReportsUploadReport = mutationKeyReportsUploadReport;
exports.buildReportsUploadReportMutation = buildReportsUploadReportMutation;
const react_query_1 = require("@tanstack/react-query");
const reportsUploadReport_js_1 = require("../funcs/reportsUploadReport.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Upload a report.
 */
function useReportsUploadReportMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildReportsUploadReportMutation(client, options),
        ...options,
    });
}
function mutationKeyReportsUploadReport() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Reports",
        "uploadReport",
    ];
}
function buildReportsUploadReportMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyReportsUploadReport(),
        mutationFn: function reportsUploadReportMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, reportsUploadReport_js_1.reportsUploadReport)(client$, request, mergedOptions));
        },
    };
}
