// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/schemaStoreCreateSchemaStoreItem.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useSchemaStoreCreateSchemaStoreItemMutation = useSchemaStoreCreateSchemaStoreItemMutation;
exports.mutationKeySchemaStoreCreateSchemaStoreItem = mutationKeySchemaStoreCreateSchemaStoreItem;
exports.buildSchemaStoreCreateSchemaStoreItemMutation = buildSchemaStoreCreateSchemaStoreItemMutation;
const react_query_1 = require("@tanstack/react-query");
const schemaStoreCreateSchemaStoreItem_js_1 = require("../funcs/schemaStoreCreateSchemaStoreItem.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Create a schema in the schema store
 */
function useSchemaStoreCreateSchemaStoreItemMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildSchemaStoreCreateSchemaStoreItemMutation(client, options),
        ...options,
    });
}
function mutationKeySchemaStoreCreateSchemaStoreItem() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "SchemaStore",
        "createSchemaStoreItem",
    ];
}
function buildSchemaStoreCreateSchemaStoreItemMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeySchemaStoreCreateSchemaStoreItem(),
        mutationFn: function schemaStoreCreateSchemaStoreItemMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, schemaStoreCreateSchemaStoreItem_js_1.schemaStoreCreateSchemaStoreItem)(client$, request, mergedOptions));
        },
    };
}
