// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/shortURLsCreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useShortURLsCreateMutation = useShortURLsCreateMutation;
exports.mutationKeyShortURLsCreate = mutationKeyShortURLsCreate;
exports.buildShortURLsCreateMutation = buildShortURLsCreateMutation;
const react_query_1 = require("@tanstack/react-query");
const shortURLsCreate_js_1 = require("../funcs/shortURLsCreate.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Shorten a URL.
 */
function useShortURLsCreateMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildShortURLsCreateMutation(client, options),
        ...options,
    });
}
function mutationKeyShortURLsCreate() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "ShortURLs",
        "create",
    ];
}
function buildShortURLsCreateMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyShortURLsCreate(),
        mutationFn: function shortURLsCreateMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, shortURLsCreate_js_1.shortURLsCreate)(client$, request, mergedOptions));
        },
    };
}
