// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/subscriptionsIgnoreSubscriptionNamespace.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useSubscriptionsIgnoreSubscriptionNamespaceMutation = useSubscriptionsIgnoreSubscriptionNamespaceMutation;
exports.mutationKeySubscriptionsIgnoreSubscriptionNamespace = mutationKeySubscriptionsIgnoreSubscriptionNamespace;
exports.buildSubscriptionsIgnoreSubscriptionNamespaceMutation = buildSubscriptionsIgnoreSubscriptionNamespaceMutation;
const react_query_1 = require("@tanstack/react-query");
const subscriptionsIgnoreSubscriptionNamespace_js_1 = require("../funcs/subscriptionsIgnoreSubscriptionNamespace.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Ignored a namespace for a subscription
 */
function useSubscriptionsIgnoreSubscriptionNamespaceMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildSubscriptionsIgnoreSubscriptionNamespaceMutation(client, options),
        ...options,
    });
}
function mutationKeySubscriptionsIgnoreSubscriptionNamespace() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Subscriptions",
        "ignoreSubscriptionNamespace",
    ];
}
function buildSubscriptionsIgnoreSubscriptionNamespaceMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeySubscriptionsIgnoreSubscriptionNamespace(),
        mutationFn: function subscriptionsIgnoreSubscriptionNamespaceMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, subscriptionsIgnoreSubscriptionNamespace_js_1.subscriptionsIgnoreSubscriptionNamespace)(client$, request, mergedOptions));
        },
    };
}
