// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/subscriptionsActivateSubscriptionNamespace.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useSubscriptionsActivateSubscriptionNamespaceMutation = useSubscriptionsActivateSubscriptionNamespaceMutation;
exports.mutationKeySubscriptionsActivateSubscriptionNamespace = mutationKeySubscriptionsActivateSubscriptionNamespace;
exports.buildSubscriptionsActivateSubscriptionNamespaceMutation = buildSubscriptionsActivateSubscriptionNamespaceMutation;
const react_query_1 = require("@tanstack/react-query");
const subscriptionsActivateSubscriptionNamespace_js_1 = require("../funcs/subscriptionsActivateSubscriptionNamespace.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Activate an ignored namespace for a subscription
 */
function useSubscriptionsActivateSubscriptionNamespaceMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildSubscriptionsActivateSubscriptionNamespaceMutation(client, options),
        ...options,
    });
}
function mutationKeySubscriptionsActivateSubscriptionNamespace() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Subscriptions",
        "activateSubscriptionNamespace",
    ];
}
function buildSubscriptionsActivateSubscriptionNamespaceMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeySubscriptionsActivateSubscriptionNamespace(),
        mutationFn: function subscriptionsActivateSubscriptionNamespaceMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, subscriptionsActivateSubscriptionNamespace_js_1.subscriptionsActivateSubscriptionNamespace)(client$, request, mergedOptions));
        },
    };
}
