// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/suggestSuggestOpenAPIRegistry.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useSuggestSuggestOpenAPIRegistryMutation = useSuggestSuggestOpenAPIRegistryMutation;
exports.mutationKeySuggestSuggestOpenAPIRegistry = mutationKeySuggestSuggestOpenAPIRegistry;
exports.buildSuggestSuggestOpenAPIRegistryMutation = buildSuggestSuggestOpenAPIRegistryMutation;
const react_query_1 = require("@tanstack/react-query");
const suggestSuggestOpenAPIRegistry_js_1 = require("../funcs/suggestSuggestOpenAPIRegistry.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Generate suggestions for improving an OpenAPI document stored in the registry.
 *
 * @remarks
 * Get suggestions from an LLM model for improving an OpenAPI document stored in the registry.
 */
function useSuggestSuggestOpenAPIRegistryMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildSuggestSuggestOpenAPIRegistryMutation(client, options),
        ...options,
    });
}
function mutationKeySuggestSuggestOpenAPIRegistry() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Suggest",
        "suggestOpenAPIRegistry",
    ];
}
function buildSuggestSuggestOpenAPIRegistryMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeySuggestSuggestOpenAPIRegistry(),
        mutationFn: function suggestSuggestOpenAPIRegistryMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, suggestSuggestOpenAPIRegistry_js_1.suggestSuggestOpenAPIRegistry)(client$, request, mergedOptions));
        },
    };
}
