// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/suggestSuggestOpenAPI.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useSuggestSuggestOpenAPIMutation = useSuggestSuggestOpenAPIMutation;
exports.mutationKeySuggestSuggestOpenAPI = mutationKeySuggestSuggestOpenAPI;
exports.buildSuggestSuggestOpenAPIMutation = buildSuggestSuggestOpenAPIMutation;
const react_query_1 = require("@tanstack/react-query");
const suggestSuggestOpenAPI_js_1 = require("../funcs/suggestSuggestOpenAPI.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * (DEPRECATED) Generate suggestions for improving an OpenAPI document.
 *
 * @remarks
 * Get suggestions from an LLM model for improving an OpenAPI document.
 */
function useSuggestSuggestOpenAPIMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildSuggestSuggestOpenAPIMutation(client, options),
        ...options,
    });
}
function mutationKeySuggestSuggestOpenAPI() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Suggest",
        "suggestOpenAPI",
    ];
}
function buildSuggestSuggestOpenAPIMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeySuggestSuggestOpenAPI(),
        mutationFn: function suggestSuggestOpenAPIMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, suggestSuggestOpenAPI_js_1.suggestSuggestOpenAPI)(client$, request, mergedOptions));
        },
    };
}
