// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/suggestSuggestItems.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useSuggestSuggestItemsMutation = useSuggestSuggestItemsMutation;
exports.mutationKeySuggestSuggestItems = mutationKeySuggestSuggestItems;
exports.buildSuggestSuggestItemsMutation = buildSuggestSuggestItemsMutation;
const react_query_1 = require("@tanstack/react-query");
const suggestSuggestItems_js_1 = require("../funcs/suggestSuggestItems.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Generate generic suggestions for a list of items.
 */
function useSuggestSuggestItemsMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildSuggestSuggestItemsMutation(client, options),
        ...options,
    });
}
function mutationKeySuggestSuggestItems() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Suggest",
        "suggestItems",
    ];
}
function buildSuggestSuggestItemsMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeySuggestSuggestItems(),
        mutationFn: function suggestSuggestItemsMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, suggestSuggestItems_js_1.suggestSuggestItems)(client$, request, mergedOptions));
        },
    };
}
