// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/suggestSuggest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useSuggestSuggestMutation = useSuggestSuggestMutation;
exports.mutationKeySuggestSuggest = mutationKeySuggestSuggest;
exports.buildSuggestSuggestMutation = buildSuggestSuggestMutation;
const react_query_1 = require("@tanstack/react-query");
const suggestSuggest_js_1 = require("../funcs/suggestSuggest.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Generate suggestions for improving an OpenAPI document.
 *
 * @remarks
 * Get suggestions from an LLM model for improving an OpenAPI document.
 */
function useSuggestSuggestMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildSuggestSuggestMutation(client, options),
        ...options,
    });
}
function mutationKeySuggestSuggest() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Suggest",
        "suggest",
    ];
}
function buildSuggestSuggestMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeySuggestSuggest(),
        mutationFn: function suggestSuggestMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, suggestSuggest_js_1.suggestSuggest)(client$, request, mergedOptions));
        },
    };
}
