// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/workspacesCreateToken.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useWorkspacesCreateTokenMutation = useWorkspacesCreateTokenMutation;
exports.mutationKeyWorkspacesCreateToken = mutationKeyWorkspacesCreateToken;
exports.buildWorkspacesCreateTokenMutation = buildWorkspacesCreateTokenMutation;
const react_query_1 = require("@tanstack/react-query");
const workspacesCreateToken_js_1 = require("../funcs/workspacesCreateToken.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Create a token for a particular workspace
 */
function useWorkspacesCreateTokenMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildWorkspacesCreateTokenMutation(client, options),
        ...options,
    });
}
function mutationKeyWorkspacesCreateToken() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Workspaces",
        "createToken",
    ];
}
function buildWorkspacesCreateTokenMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyWorkspacesCreateToken(),
        mutationFn: function workspacesCreateTokenMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, workspacesCreateToken_js_1.workspacesCreateToken)(client$, request, mergedOptions));
        },
    };
}
