// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/workspacesGetTokens.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useWorkspacesGetTokens = useWorkspacesGetTokens;
exports.useWorkspacesGetTokensSuspense = useWorkspacesGetTokensSuspense;
exports.prefetchWorkspacesGetTokens = prefetchWorkspacesGetTokens;
exports.setWorkspacesGetTokensData = setWorkspacesGetTokensData;
exports.invalidateWorkspacesGetTokens = invalidateWorkspacesGetTokens;
exports.invalidateAllWorkspacesGetTokens = invalidateAllWorkspacesGetTokens;
exports.buildWorkspacesGetTokensQuery = buildWorkspacesGetTokensQuery;
exports.queryKeyWorkspacesGetTokens = queryKeyWorkspacesGetTokens;
const react_query_1 = require("@tanstack/react-query");
const workspacesGetTokens_js_1 = require("../funcs/workspacesGetTokens.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Get tokens for a particular workspace
 */
function useWorkspacesGetTokens(request, options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useQuery)({
        ...buildWorkspacesGetTokensQuery(client, request, options),
        ...options,
    });
}
/**
 * Get tokens for a particular workspace
 */
function useWorkspacesGetTokensSuspense(request, options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useSuspenseQuery)({
        ...buildWorkspacesGetTokensQuery(client, request, options),
        ...options,
    });
}
function prefetchWorkspacesGetTokens(queryClient, client$, request) {
    return queryClient.prefetchQuery({
        ...buildWorkspacesGetTokensQuery(client$, request),
    });
}
function setWorkspacesGetTokensData(client, queryKeyBase, data) {
    const key = queryKeyWorkspacesGetTokens(...queryKeyBase);
    return client.setQueryData(key, data);
}
function invalidateWorkspacesGetTokens(client, queryKeyBase, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: [
            "@speakeasy-api/speakeasy-client-sdk-typescript",
            "Workspaces",
            "getTokens",
            ...queryKeyBase,
        ],
    });
}
function invalidateAllWorkspacesGetTokens(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: [
            "@speakeasy-api/speakeasy-client-sdk-typescript",
            "Workspaces",
            "getTokens",
        ],
    });
}
function buildWorkspacesGetTokensQuery(client$, request, options) {
    return {
        queryKey: queryKeyWorkspacesGetTokens(request.workspaceId),
        queryFn: async function workspacesGetTokensQueryFn(ctx) {
            const sig = (0, primitives_js_1.combineSignals)(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return (0, fp_js_1.unwrapAsync)((0, workspacesGetTokens_js_1.workspacesGetTokens)(client$, request, mergedOptions));
        },
    };
}
function queryKeyWorkspacesGetTokens(workspaceId) {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Workspaces",
        "getTokens",
        workspaceId,
    ];
}
