// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/workspacesDeleteToken.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useWorkspacesDeleteTokenMutation = useWorkspacesDeleteTokenMutation;
exports.mutationKeyWorkspacesDeleteToken = mutationKeyWorkspacesDeleteToken;
exports.buildWorkspacesDeleteTokenMutation = buildWorkspacesDeleteTokenMutation;
const react_query_1 = require("@tanstack/react-query");
const workspacesDeleteToken_js_1 = require("../funcs/workspacesDeleteToken.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Delete a token for a particular workspace
 */
function useWorkspacesDeleteTokenMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildWorkspacesDeleteTokenMutation(client, options),
        ...options,
    });
}
function mutationKeyWorkspacesDeleteToken() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Workspaces",
        "deleteToken",
    ];
}
function buildWorkspacesDeleteTokenMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyWorkspacesDeleteToken(),
        mutationFn: function workspacesDeleteTokenMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, workspacesDeleteToken_js_1.workspacesDeleteToken)(client$, request, mergedOptions));
        },
    };
}
