// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/workspacesRevokeAccess.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useWorkspacesRevokeAccessMutation = useWorkspacesRevokeAccessMutation;
exports.mutationKeyWorkspacesRevokeAccess = mutationKeyWorkspacesRevokeAccess;
exports.buildWorkspacesRevokeAccessMutation = buildWorkspacesRevokeAccessMutation;
const react_query_1 = require("@tanstack/react-query");
const workspacesRevokeAccess_js_1 = require("../funcs/workspacesRevokeAccess.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Revoke a user's access to a particular workspace
 */
function useWorkspacesRevokeAccessMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildWorkspacesRevokeAccessMutation(client, options),
        ...options,
    });
}
function mutationKeyWorkspacesRevokeAccess() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Workspaces",
        "revokeAccess",
    ];
}
function buildWorkspacesRevokeAccessMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyWorkspacesRevokeAccess(),
        mutationFn: function workspacesRevokeAccessMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, workspacesRevokeAccess_js_1.workspacesRevokeAccess)(client$, request, mergedOptions));
        },
    };
}
