// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/workspacesUpdateSettings.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useWorkspacesUpdateSettingsMutation = useWorkspacesUpdateSettingsMutation;
exports.mutationKeyWorkspacesUpdateSettings = mutationKeyWorkspacesUpdateSettings;
exports.buildWorkspacesUpdateSettingsMutation = buildWorkspacesUpdateSettingsMutation;
const react_query_1 = require("@tanstack/react-query");
const workspacesUpdateSettings_js_1 = require("../funcs/workspacesUpdateSettings.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Update workspace settings
 *
 * @remarks
 * Update settings about a particular workspace.
 */
function useWorkspacesUpdateSettingsMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildWorkspacesUpdateSettingsMutation(client, options),
        ...options,
    });
}
function mutationKeyWorkspacesUpdateSettings() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Workspaces",
        "updateSettings",
    ];
}
function buildWorkspacesUpdateSettingsMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyWorkspacesUpdateSettings(),
        mutationFn: function workspacesUpdateSettingsMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, workspacesUpdateSettings_js_1.workspacesUpdateSettings)(client$, request, mergedOptions));
        },
    };
}
