// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/workspacesGetAll.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useWorkspacesGetAll = useWorkspacesGetAll;
exports.useWorkspacesGetAllSuspense = useWorkspacesGetAllSuspense;
exports.prefetchWorkspacesGetAll = prefetchWorkspacesGetAll;
exports.setWorkspacesGetAllData = setWorkspacesGetAllData;
exports.invalidateAllWorkspacesGetAll = invalidateAllWorkspacesGetAll;
exports.buildWorkspacesGetAllQuery = buildWorkspacesGetAllQuery;
exports.queryKeyWorkspacesGetAll = queryKeyWorkspacesGetAll;
const react_query_1 = require("@tanstack/react-query");
const workspacesGetAll_js_1 = require("../funcs/workspacesGetAll.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Get workspaces for a user
 *
 * @remarks
 * Returns a list of workspaces a user has access too
 */
function useWorkspacesGetAll(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useQuery)({
        ...buildWorkspacesGetAllQuery(client, options),
        ...options,
    });
}
/**
 * Get workspaces for a user
 *
 * @remarks
 * Returns a list of workspaces a user has access too
 */
function useWorkspacesGetAllSuspense(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useSuspenseQuery)({
        ...buildWorkspacesGetAllQuery(client, options),
        ...options,
    });
}
function prefetchWorkspacesGetAll(queryClient, client$) {
    return queryClient.prefetchQuery({
        ...buildWorkspacesGetAllQuery(client$),
    });
}
function setWorkspacesGetAllData(client, data) {
    const key = queryKeyWorkspacesGetAll();
    return client.setQueryData(key, data);
}
function invalidateAllWorkspacesGetAll(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: [
            "@speakeasy-api/speakeasy-client-sdk-typescript",
            "Workspaces",
            "getAll",
        ],
    });
}
function buildWorkspacesGetAllQuery(client$, options) {
    return {
        queryKey: queryKeyWorkspacesGetAll(),
        queryFn: async function workspacesGetAllQueryFn(ctx) {
            const sig = (0, primitives_js_1.combineSignals)(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return (0, fp_js_1.unwrapAsync)((0, workspacesGetAll_js_1.workspacesGetAll)(client$, mergedOptions));
        },
    };
}
function queryKeyWorkspacesGetAll() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Workspaces",
        "getAll",
    ];
}
