// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/workspacesGetTeam.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useWorkspacesGetTeam = useWorkspacesGetTeam;
exports.useWorkspacesGetTeamSuspense = useWorkspacesGetTeamSuspense;
exports.prefetchWorkspacesGetTeam = prefetchWorkspacesGetTeam;
exports.setWorkspacesGetTeamData = setWorkspacesGetTeamData;
exports.invalidateWorkspacesGetTeam = invalidateWorkspacesGetTeam;
exports.invalidateAllWorkspacesGetTeam = invalidateAllWorkspacesGetTeam;
exports.buildWorkspacesGetTeamQuery = buildWorkspacesGetTeamQuery;
exports.queryKeyWorkspacesGetTeam = queryKeyWorkspacesGetTeam;
const react_query_1 = require("@tanstack/react-query");
const workspacesGetTeam_js_1 = require("../funcs/workspacesGetTeam.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Get team members for a particular workspace
 */
function useWorkspacesGetTeam(request, options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useQuery)({
        ...buildWorkspacesGetTeamQuery(client, request, options),
        ...options,
    });
}
/**
 * Get team members for a particular workspace
 */
function useWorkspacesGetTeamSuspense(request, options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useSuspenseQuery)({
        ...buildWorkspacesGetTeamQuery(client, request, options),
        ...options,
    });
}
function prefetchWorkspacesGetTeam(queryClient, client$, request) {
    return queryClient.prefetchQuery({
        ...buildWorkspacesGetTeamQuery(client$, request),
    });
}
function setWorkspacesGetTeamData(client, queryKeyBase, data) {
    const key = queryKeyWorkspacesGetTeam(...queryKeyBase);
    return client.setQueryData(key, data);
}
function invalidateWorkspacesGetTeam(client, queryKeyBase, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: [
            "@speakeasy-api/speakeasy-client-sdk-typescript",
            "Workspaces",
            "getTeam",
            ...queryKeyBase,
        ],
    });
}
function invalidateAllWorkspacesGetTeam(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: [
            "@speakeasy-api/speakeasy-client-sdk-typescript",
            "Workspaces",
            "getTeam",
        ],
    });
}
function buildWorkspacesGetTeamQuery(client$, request, options) {
    return {
        queryKey: queryKeyWorkspacesGetTeam(request.workspaceId),
        queryFn: async function workspacesGetTeamQueryFn(ctx) {
            const sig = (0, primitives_js_1.combineSignals)(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return (0, fp_js_1.unwrapAsync)((0, workspacesGetTeam_js_1.workspacesGetTeam)(client$, request, mergedOptions));
        },
    };
}
function queryKeyWorkspacesGetTeam(workspaceId) {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Workspaces",
        "getTeam",
        workspaceId,
    ];
}
