// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/workspacesCreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useWorkspacesCreateMutation = useWorkspacesCreateMutation;
exports.mutationKeyWorkspacesCreate = mutationKeyWorkspacesCreate;
exports.buildWorkspacesCreateMutation = buildWorkspacesCreateMutation;
const react_query_1 = require("@tanstack/react-query");
const workspacesCreate_js_1 = require("../funcs/workspacesCreate.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Create a workspace
 *
 * @remarks
 * Creates a workspace
 */
function useWorkspacesCreateMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildWorkspacesCreateMutation(client, options),
        ...options,
    });
}
function mutationKeyWorkspacesCreate() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Workspaces",
        "create",
    ];
}
function buildWorkspacesCreateMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyWorkspacesCreate(),
        mutationFn: function workspacesCreateMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, workspacesCreate_js_1.workspacesCreate)(client$, request, mergedOptions));
        },
    };
}
