// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/workspacesSetFeatureFlags.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useWorkspacesSetFeatureFlagsMutation = useWorkspacesSetFeatureFlagsMutation;
exports.mutationKeyWorkspacesSetFeatureFlags = mutationKeyWorkspacesSetFeatureFlags;
exports.buildWorkspacesSetFeatureFlagsMutation = buildWorkspacesSetFeatureFlagsMutation;
const react_query_1 = require("@tanstack/react-query");
const workspacesSetFeatureFlags_js_1 = require("../funcs/workspacesSetFeatureFlags.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Set workspace feature flags
 */
function useWorkspacesSetFeatureFlagsMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildWorkspacesSetFeatureFlagsMutation(client, options),
        ...options,
    });
}
function mutationKeyWorkspacesSetFeatureFlags() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Workspaces",
        "setFeatureFlags",
    ];
}
function buildWorkspacesSetFeatureFlagsMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyWorkspacesSetFeatureFlags(),
        mutationFn: function workspacesSetFeatureFlagsMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, workspacesSetFeatureFlags_js_1.workspacesSetFeatureFlags)(client$, request, mergedOptions));
        },
    };
}
