// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/workspacesUpdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useWorkspacesUpdateMutation = useWorkspacesUpdateMutation;
exports.mutationKeyWorkspacesUpdate = mutationKeyWorkspacesUpdate;
exports.buildWorkspacesUpdateMutation = buildWorkspacesUpdateMutation;
const react_query_1 = require("@tanstack/react-query");
const workspacesUpdate_js_1 = require("../funcs/workspacesUpdate.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Update workspace details
 *
 * @remarks
 * Update information about a particular workspace.
 */
function useWorkspacesUpdateMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildWorkspacesUpdateMutation(client, options),
        ...options,
    });
}
function mutationKeyWorkspacesUpdate() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Workspaces",
        "update",
    ];
}
function buildWorkspacesUpdateMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyWorkspacesUpdate(),
        mutationFn: function workspacesUpdateMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, workspacesUpdate_js_1.workspacesUpdate)(client$, request, mergedOptions));
        },
    };
}
