// @speakeasy-api/speakeasy-client-sdk-typescript/react-query/workspacesGrantAccess.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useWorkspacesGrantAccessMutation = useWorkspacesGrantAccessMutation;
exports.mutationKeyWorkspacesGrantAccess = mutationKeyWorkspacesGrantAccess;
exports.buildWorkspacesGrantAccessMutation = buildWorkspacesGrantAccessMutation;
const react_query_1 = require("@tanstack/react-query");
const workspacesGrantAccess_js_1 = require("../funcs/workspacesGrantAccess.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../sdk/types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Grant a user access to a particular workspace
 */
function useWorkspacesGrantAccessMutation(options) {
    const client = (0, _context_js_1.useSpeakeasyContext)();
    return (0, react_query_1.useMutation)({
        ...buildWorkspacesGrantAccessMutation(client, options),
        ...options,
    });
}
function mutationKeyWorkspacesGrantAccess() {
    return [
        "@speakeasy-api/speakeasy-client-sdk-typescript",
        "Workspaces",
        "grantAccess",
    ];
}
function buildWorkspacesGrantAccessMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyWorkspacesGrantAccess(),
        mutationFn: function workspacesGrantAccessMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, workspacesGrantAccess_js_1.workspacesGrantAccess)(client$, request, mergedOptions));
        },
    };
}
