// @speakeasy-api/speakeasy-client-sdk-typescript/mcp-server/tools/organizationsGetUsage.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$organizationsGetUsage = void 0;
const organizationsGetUsage_js_1 = require("../../funcs/organizationsGetUsage.js");
const tools_js_1 = require("../tools.js");
exports.tool$organizationsGetUsage = {
    name: "organizations-get-usage",
    description: `Get billing usage summary for a particular organization

Returns a billing usage summary by target languages for a particular organization`,
    tool: async (client, ctx) => {
        const [result, apiCall] = await (0, organizationsGetUsage_js_1.organizationsGetUsage)(client, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
