// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/billingaddon.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BillingAddOn$ = exports.BillingAddOn$outboundSchema = exports.BillingAddOn$inboundSchema = exports.BillingAddOn = void 0;
const z = __importStar(require("zod"));
const enums_js_1 = require("../../types/enums.js");
exports.BillingAddOn = {
    Webhooks: "webhooks",
    SDKTesting: "sdk_testing",
    CustomCodeRegions: "custom_code_regions",
    SnippetAi: "snippet_ai",
};
/** @internal */
exports.BillingAddOn$inboundSchema = z
    .union([
    z.nativeEnum(exports.BillingAddOn),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.BillingAddOn$outboundSchema = z.union([
    z.nativeEnum(exports.BillingAddOn),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BillingAddOn$;
(function (BillingAddOn$) {
    /** @deprecated use `BillingAddOn$inboundSchema` instead. */
    BillingAddOn$.inboundSchema = exports.BillingAddOn$inboundSchema;
    /** @deprecated use `BillingAddOn$outboundSchema` instead. */
    BillingAddOn$.outboundSchema = exports.BillingAddOn$outboundSchema;
})(BillingAddOn$ || (exports.BillingAddOn$ = BillingAddOn$ = {}));
