// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/codesamplesjobstatus.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CodeSamplesJobStatus$ = exports.CodeSamplesJobStatus$outboundSchema = exports.CodeSamplesJobStatus$inboundSchema = exports.CodeSamplesJobStatus = void 0;
const z = __importStar(require("zod"));
/**
 * The current status of the job. Possible values are `pending` or `running`.
 */
exports.CodeSamplesJobStatus = {
    Pending: "pending",
    Running: "running",
};
/** @internal */
exports.CodeSamplesJobStatus$inboundSchema = z.nativeEnum(exports.CodeSamplesJobStatus);
/** @internal */
exports.CodeSamplesJobStatus$outboundSchema = exports.CodeSamplesJobStatus$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CodeSamplesJobStatus$;
(function (CodeSamplesJobStatus$) {
    /** @deprecated use `CodeSamplesJobStatus$inboundSchema` instead. */
    CodeSamplesJobStatus$.inboundSchema = exports.CodeSamplesJobStatus$inboundSchema;
    /** @deprecated use `CodeSamplesJobStatus$outboundSchema` instead. */
    CodeSamplesJobStatus$.outboundSchema = exports.CodeSamplesJobStatus$outboundSchema;
})(CodeSamplesJobStatus$ || (exports.CodeSamplesJobStatus$ = CodeSamplesJobStatus$ = {}));
