// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/diagnostic.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Diagnostic$ = exports.Diagnostic$outboundSchema = exports.Diagnostic$inboundSchema = void 0;
exports.diagnosticToJSON = diagnosticToJSON;
exports.diagnosticFromJSON = diagnosticFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.Diagnostic$inboundSchema = z.object({
    helpMessage: z.string().optional(),
    message: z.string(),
    path: z.array(z.string()),
    type: z.string(),
});
/** @internal */
exports.Diagnostic$outboundSchema = z.object({
    helpMessage: z.string().optional(),
    message: z.string(),
    path: z.array(z.string()),
    type: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Diagnostic$;
(function (Diagnostic$) {
    /** @deprecated use `Diagnostic$inboundSchema` instead. */
    Diagnostic$.inboundSchema = exports.Diagnostic$inboundSchema;
    /** @deprecated use `Diagnostic$outboundSchema` instead. */
    Diagnostic$.outboundSchema = exports.Diagnostic$outboundSchema;
})(Diagnostic$ || (exports.Diagnostic$ = Diagnostic$ = {}));
function diagnosticToJSON(diagnostic) {
    return JSON.stringify(exports.Diagnostic$outboundSchema.parse(diagnostic));
}
function diagnosticFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Diagnostic$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Diagnostic' from JSON`);
}
