// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/featureflag.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeatureFlag$ = exports.FeatureFlag$outboundSchema = exports.FeatureFlag$inboundSchema = void 0;
exports.featureFlagToJSON = featureFlagToJSON;
exports.featureFlagFromJSON = featureFlagFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
const workspacefeatureflag_js_1 = require("./workspacefeatureflag.js");
/** @internal */
exports.FeatureFlag$inboundSchema = z.object({
    feature_flag: workspacefeatureflag_js_1.WorkspaceFeatureFlag$inboundSchema,
    trial_ends_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "feature_flag": "featureFlag",
        "trial_ends_at": "trialEndsAt",
    });
});
/** @internal */
exports.FeatureFlag$outboundSchema = z.object({
    featureFlag: workspacefeatureflag_js_1.WorkspaceFeatureFlag$outboundSchema,
    trialEndsAt: z.nullable(z.date().transform(v => v.toISOString())).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        featureFlag: "feature_flag",
        trialEndsAt: "trial_ends_at",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FeatureFlag$;
(function (FeatureFlag$) {
    /** @deprecated use `FeatureFlag$inboundSchema` instead. */
    FeatureFlag$.inboundSchema = exports.FeatureFlag$inboundSchema;
    /** @deprecated use `FeatureFlag$outboundSchema` instead. */
    FeatureFlag$.outboundSchema = exports.FeatureFlag$outboundSchema;
})(FeatureFlag$ || (exports.FeatureFlag$ = FeatureFlag$ = {}));
function featureFlagToJSON(featureFlag) {
    return JSON.stringify(exports.FeatureFlag$outboundSchema.parse(featureFlag));
}
function featureFlagFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.FeatureFlag$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FeatureFlag' from JSON`);
}
