// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/getnamespacesresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetNamespacesResponse$ = exports.GetNamespacesResponse$outboundSchema = exports.GetNamespacesResponse$inboundSchema = void 0;
exports.getNamespacesResponseToJSON = getNamespacesResponseToJSON;
exports.getNamespacesResponseFromJSON = getNamespacesResponseFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../../lib/schemas.js");
const namespace_js_1 = require("./namespace.js");
/** @internal */
exports.GetNamespacesResponse$inboundSchema = z.object({
    items: z.array(namespace_js_1.Namespace$inboundSchema),
});
/** @internal */
exports.GetNamespacesResponse$outboundSchema = z.object({
    items: z.array(namespace_js_1.Namespace$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetNamespacesResponse$;
(function (GetNamespacesResponse$) {
    /** @deprecated use `GetNamespacesResponse$inboundSchema` instead. */
    GetNamespacesResponse$.inboundSchema = exports.GetNamespacesResponse$inboundSchema;
    /** @deprecated use `GetNamespacesResponse$outboundSchema` instead. */
    GetNamespacesResponse$.outboundSchema = exports.GetNamespacesResponse$outboundSchema;
})(GetNamespacesResponse$ || (exports.GetNamespacesResponse$ = GetNamespacesResponse$ = {}));
function getNamespacesResponseToJSON(getNamespacesResponse) {
    return JSON.stringify(exports.GetNamespacesResponse$outboundSchema.parse(getNamespacesResponse));
}
function getNamespacesResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetNamespacesResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetNamespacesResponse' from JSON`);
}
