// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/getrevisionsresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetRevisionsResponse$ = exports.GetRevisionsResponse$outboundSchema = exports.GetRevisionsResponse$inboundSchema = void 0;
exports.getRevisionsResponseToJSON = getRevisionsResponseToJSON;
exports.getRevisionsResponseFromJSON = getRevisionsResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
const revision_js_1 = require("./revision.js");
/** @internal */
exports.GetRevisionsResponse$inboundSchema = z.object({
    items: z.array(revision_js_1.Revision$inboundSchema),
    next_page_token: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "next_page_token": "nextPageToken",
    });
});
/** @internal */
exports.GetRevisionsResponse$outboundSchema = z.object({
    items: z.array(revision_js_1.Revision$outboundSchema),
    nextPageToken: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        nextPageToken: "next_page_token",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetRevisionsResponse$;
(function (GetRevisionsResponse$) {
    /** @deprecated use `GetRevisionsResponse$inboundSchema` instead. */
    GetRevisionsResponse$.inboundSchema = exports.GetRevisionsResponse$inboundSchema;
    /** @deprecated use `GetRevisionsResponse$outboundSchema` instead. */
    GetRevisionsResponse$.outboundSchema = exports.GetRevisionsResponse$outboundSchema;
})(GetRevisionsResponse$ || (exports.GetRevisionsResponse$ = GetRevisionsResponse$ = {}));
function getRevisionsResponseToJSON(getRevisionsResponse) {
    return JSON.stringify(exports.GetRevisionsResponse$outboundSchema.parse(getRevisionsResponse));
}
function getRevisionsResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetRevisionsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRevisionsResponse' from JSON`);
}
