// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/gettagsresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetTagsResponse$ = exports.GetTagsResponse$outboundSchema = exports.GetTagsResponse$inboundSchema = void 0;
exports.getTagsResponseToJSON = getTagsResponseToJSON;
exports.getTagsResponseFromJSON = getTagsResponseFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../../lib/schemas.js");
const tag_js_1 = require("./tag.js");
/** @internal */
exports.GetTagsResponse$inboundSchema = z.object({
    items: z.array(tag_js_1.Tag$inboundSchema),
});
/** @internal */
exports.GetTagsResponse$outboundSchema = z.object({
    items: z.array(tag_js_1.Tag$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetTagsResponse$;
(function (GetTagsResponse$) {
    /** @deprecated use `GetTagsResponse$inboundSchema` instead. */
    GetTagsResponse$.inboundSchema = exports.GetTagsResponse$inboundSchema;
    /** @deprecated use `GetTagsResponse$outboundSchema` instead. */
    GetTagsResponse$.outboundSchema = exports.GetTagsResponse$outboundSchema;
})(GetTagsResponse$ || (exports.GetTagsResponse$ = GetTagsResponse$ = {}));
function getTagsResponseToJSON(getTagsResponse) {
    return JSON.stringify(exports.GetTagsResponse$outboundSchema.parse(getTagsResponse));
}
function getTagsResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetTagsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetTagsResponse' from JSON`);
}
