// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/githubconfiguremintlifyreporequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GithubConfigureMintlifyRepoRequest$ = exports.GithubConfigureMintlifyRepoRequest$outboundSchema = exports.GithubConfigureMintlifyRepoRequest$inboundSchema = void 0;
exports.githubConfigureMintlifyRepoRequestToJSON = githubConfigureMintlifyRepoRequestToJSON;
exports.githubConfigureMintlifyRepoRequestFromJSON = githubConfigureMintlifyRepoRequestFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.GithubConfigureMintlifyRepoRequest$inboundSchema = z.object({
    input: z.string(),
    org: z.string(),
    overlays: z.array(z.string()),
    repo: z.string(),
    subdirectory: z.string().optional(),
});
/** @internal */
exports.GithubConfigureMintlifyRepoRequest$outboundSchema = z.object({
    input: z.string(),
    org: z.string(),
    overlays: z.array(z.string()),
    repo: z.string(),
    subdirectory: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GithubConfigureMintlifyRepoRequest$;
(function (GithubConfigureMintlifyRepoRequest$) {
    /** @deprecated use `GithubConfigureMintlifyRepoRequest$inboundSchema` instead. */
    GithubConfigureMintlifyRepoRequest$.inboundSchema = exports.GithubConfigureMintlifyRepoRequest$inboundSchema;
    /** @deprecated use `GithubConfigureMintlifyRepoRequest$outboundSchema` instead. */
    GithubConfigureMintlifyRepoRequest$.outboundSchema = exports.GithubConfigureMintlifyRepoRequest$outboundSchema;
})(GithubConfigureMintlifyRepoRequest$ || (exports.GithubConfigureMintlifyRepoRequest$ = GithubConfigureMintlifyRepoRequest$ = {}));
function githubConfigureMintlifyRepoRequestToJSON(githubConfigureMintlifyRepoRequest) {
    return JSON.stringify(exports.GithubConfigureMintlifyRepoRequest$outboundSchema.parse(githubConfigureMintlifyRepoRequest));
}
function githubConfigureMintlifyRepoRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GithubConfigureMintlifyRepoRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GithubConfigureMintlifyRepoRequest' from JSON`);
}
