// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/githubconfigurecodesamplesresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GithubConfigureCodeSamplesResponse$ = exports.GithubConfigureCodeSamplesResponse$outboundSchema = exports.GithubConfigureCodeSamplesResponse$inboundSchema = void 0;
exports.githubConfigureCodeSamplesResponseToJSON = githubConfigureCodeSamplesResponseToJSON;
exports.githubConfigureCodeSamplesResponseFromJSON = githubConfigureCodeSamplesResponseFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../../lib/schemas.js");
const workflowdocument_js_1 = require("./workflowdocument.js");
/** @internal */
exports.GithubConfigureCodeSamplesResponse$inboundSchema = z.object({
    codeSampleOverlayRegistryURL: z.string(),
    ghActionID: z.string().optional(),
    source: workflowdocument_js_1.WorkflowDocument$inboundSchema,
});
/** @internal */
exports.GithubConfigureCodeSamplesResponse$outboundSchema = z.object({
    codeSampleOverlayRegistryURL: z.string(),
    ghActionID: z.string().optional(),
    source: workflowdocument_js_1.WorkflowDocument$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GithubConfigureCodeSamplesResponse$;
(function (GithubConfigureCodeSamplesResponse$) {
    /** @deprecated use `GithubConfigureCodeSamplesResponse$inboundSchema` instead. */
    GithubConfigureCodeSamplesResponse$.inboundSchema = exports.GithubConfigureCodeSamplesResponse$inboundSchema;
    /** @deprecated use `GithubConfigureCodeSamplesResponse$outboundSchema` instead. */
    GithubConfigureCodeSamplesResponse$.outboundSchema = exports.GithubConfigureCodeSamplesResponse$outboundSchema;
})(GithubConfigureCodeSamplesResponse$ || (exports.GithubConfigureCodeSamplesResponse$ = GithubConfigureCodeSamplesResponse$ = {}));
function githubConfigureCodeSamplesResponseToJSON(githubConfigureCodeSamplesResponse) {
    return JSON.stringify(exports.GithubConfigureCodeSamplesResponse$outboundSchema.parse(githubConfigureCodeSamplesResponse));
}
function githubConfigureCodeSamplesResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GithubConfigureCodeSamplesResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GithubConfigureCodeSamplesResponse' from JSON`);
}
