// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/interactiontype.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.InteractionType$ = exports.InteractionType$outboundSchema = exports.InteractionType$inboundSchema = exports.InteractionType = void 0;
const z = __importStar(require("zod"));
/**
 * Type of interaction.
 */
exports.InteractionType = {
    CiExec: "CI_EXEC",
    CliExec: "CLI_EXEC",
    Lint: "LINT",
    OpenapiDiff: "OPENAPI_DIFF",
    TargetGenerate: "TARGET_GENERATE",
    Tombstone: "TOMBSTONE",
    Authenticate: "AUTHENTICATE",
    Quickstart: "QUICKSTART",
    Run: "RUN",
    Configure: "CONFIGURE",
    Publish: "PUBLISH",
    Test: "TEST",
};
/** @internal */
exports.InteractionType$inboundSchema = z.nativeEnum(exports.InteractionType);
/** @internal */
exports.InteractionType$outboundSchema = exports.InteractionType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var InteractionType$;
(function (InteractionType$) {
    /** @deprecated use `InteractionType$inboundSchema` instead. */
    InteractionType$.inboundSchema = exports.InteractionType$inboundSchema;
    /** @deprecated use `InteractionType$outboundSchema` instead. */
    InteractionType$.outboundSchema = exports.InteractionType$outboundSchema;
})(InteractionType$ || (exports.InteractionType$ = InteractionType$ = {}));
