// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/manifest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Manifest$ = exports.Manifest$outboundSchema = exports.Manifest$inboundSchema = void 0;
exports.manifestToJSON = manifestToJSON;
exports.manifestFromJSON = manifestFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../../lib/schemas.js");
const annotations_js_1 = require("./annotations.js");
const v2descriptor_js_1 = require("./v2descriptor.js");
/** @internal */
exports.Manifest$inboundSchema = z.object({
    annotations: annotations_js_1.Annotations$inboundSchema.optional(),
    artifactType: z.string().optional(),
    layers: z.array(v2descriptor_js_1.V2Descriptor$inboundSchema).optional(),
    mediaType: z.string().optional(),
    schemaVersion: z.number().int().optional(),
});
/** @internal */
exports.Manifest$outboundSchema = z.object({
    annotations: annotations_js_1.Annotations$outboundSchema.optional(),
    artifactType: z.string().optional(),
    layers: z.array(v2descriptor_js_1.V2Descriptor$outboundSchema).optional(),
    mediaType: z.string().optional(),
    schemaVersion: z.number().int().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Manifest$;
(function (Manifest$) {
    /** @deprecated use `Manifest$inboundSchema` instead. */
    Manifest$.inboundSchema = exports.Manifest$inboundSchema;
    /** @deprecated use `Manifest$outboundSchema` instead. */
    Manifest$.outboundSchema = exports.Manifest$outboundSchema;
})(Manifest$ || (exports.Manifest$ = Manifest$ = {}));
function manifestToJSON(manifest) {
    return JSON.stringify(exports.Manifest$outboundSchema.parse(manifest));
}
function manifestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Manifest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Manifest' from JSON`);
}
