// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/oasinfo.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OASInfo$ = exports.OASInfo$outboundSchema = exports.OASInfo$inboundSchema = exports.License$ = exports.License$outboundSchema = exports.License$inboundSchema = void 0;
exports.licenseToJSON = licenseToJSON;
exports.licenseFromJSON = licenseFromJSON;
exports.oasInfoToJSON = oasInfoToJSON;
exports.oasInfoFromJSON = oasInfoFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.License$inboundSchema = z.object({
    identifier: z.string().optional(),
});
/** @internal */
exports.License$outboundSchema = z.object({
    identifier: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var License$;
(function (License$) {
    /** @deprecated use `License$inboundSchema` instead. */
    License$.inboundSchema = exports.License$inboundSchema;
    /** @deprecated use `License$outboundSchema` instead. */
    License$.outboundSchema = exports.License$outboundSchema;
})(License$ || (exports.License$ = License$ = {}));
function licenseToJSON(license) {
    return JSON.stringify(exports.License$outboundSchema.parse(license));
}
function licenseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.License$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'License' from JSON`);
}
/** @internal */
exports.OASInfo$inboundSchema = z.object({
    description: z.string(),
    license: z.lazy(() => exports.License$inboundSchema),
    summary: z.string(),
    title: z.string(),
    version: z.string(),
});
/** @internal */
exports.OASInfo$outboundSchema = z.object({
    description: z.string(),
    license: z.lazy(() => exports.License$outboundSchema),
    summary: z.string(),
    title: z.string(),
    version: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OASInfo$;
(function (OASInfo$) {
    /** @deprecated use `OASInfo$inboundSchema` instead. */
    OASInfo$.inboundSchema = exports.OASInfo$inboundSchema;
    /** @deprecated use `OASInfo$outboundSchema` instead. */
    OASInfo$.outboundSchema = exports.OASInfo$outboundSchema;
})(OASInfo$ || (exports.OASInfo$ = OASInfo$ = {}));
function oasInfoToJSON(oasInfo) {
    return JSON.stringify(exports.OASInfo$outboundSchema.parse(oasInfo));
}
function oasInfoFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OASInfo$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OASInfo' from JSON`);
}
