// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/oasoperation.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OASOperation$ = exports.OASOperation$outboundSchema = exports.OASOperation$inboundSchema = void 0;
exports.oasOperationToJSON = oasOperationToJSON;
exports.oasOperationFromJSON = oasOperationFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.OASOperation$inboundSchema = z.object({
    description: z.string(),
    group_override: z.string().optional(),
    method: z.string(),
    method_name_override: z.string().optional(),
    operation_id: z.string(),
    path: z.string(),
    tags: z.array(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "group_override": "groupOverride",
        "method_name_override": "methodNameOverride",
        "operation_id": "operationId",
    });
});
/** @internal */
exports.OASOperation$outboundSchema = z.object({
    description: z.string(),
    groupOverride: z.string().optional(),
    method: z.string(),
    methodNameOverride: z.string().optional(),
    operationId: z.string(),
    path: z.string(),
    tags: z.array(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        groupOverride: "group_override",
        methodNameOverride: "method_name_override",
        operationId: "operation_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OASOperation$;
(function (OASOperation$) {
    /** @deprecated use `OASOperation$inboundSchema` instead. */
    OASOperation$.inboundSchema = exports.OASOperation$inboundSchema;
    /** @deprecated use `OASOperation$outboundSchema` instead. */
    OASOperation$.outboundSchema = exports.OASOperation$outboundSchema;
})(OASOperation$ || (exports.OASOperation$ = OASOperation$ = {}));
function oasOperationToJSON(oasOperation) {
    return JSON.stringify(exports.OASOperation$outboundSchema.parse(oasOperation));
}
function oasOperationFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OASOperation$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OASOperation' from JSON`);
}
