// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/oassummary.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OASSummary$ = exports.OASSummary$outboundSchema = exports.OASSummary$inboundSchema = void 0;
exports.oasSummaryToJSON = oasSummaryToJSON;
exports.oasSummaryFromJSON = oasSummaryFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../../lib/schemas.js");
const oasinfo_js_1 = require("./oasinfo.js");
const oasoperation_js_1 = require("./oasoperation.js");
/** @internal */
exports.OASSummary$inboundSchema = z.object({
    info: oasinfo_js_1.OASInfo$inboundSchema,
    operations: z.array(oasoperation_js_1.OASOperation$inboundSchema),
});
/** @internal */
exports.OASSummary$outboundSchema = z.object({
    info: oasinfo_js_1.OASInfo$outboundSchema,
    operations: z.array(oasoperation_js_1.OASOperation$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OASSummary$;
(function (OASSummary$) {
    /** @deprecated use `OASSummary$inboundSchema` instead. */
    OASSummary$.inboundSchema = exports.OASSummary$inboundSchema;
    /** @deprecated use `OASSummary$outboundSchema` instead. */
    OASSummary$.outboundSchema = exports.OASSummary$outboundSchema;
})(OASSummary$ || (exports.OASSummary$ = OASSummary$ = {}));
function oasSummaryToJSON(oasSummary) {
    return JSON.stringify(exports.OASSummary$outboundSchema.parse(oasSummary));
}
function oasSummaryFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OASSummary$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OASSummary' from JSON`);
}
