// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/organization.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Organization$ = exports.Organization$outboundSchema = exports.Organization$inboundSchema = void 0;
exports.organizationToJSON = organizationToJSON;
exports.organizationFromJSON = organizationFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
const accounttype_js_1 = require("./accounttype.js");
/** @internal */
exports.Organization$inboundSchema = z.object({
    account_type: accounttype_js_1.AccountType$inboundSchema,
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    free_trial_expiry: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    id: z.string(),
    internal: z.boolean().optional(),
    name: z.string(),
    slug: z.string(),
    sso_activated: z.boolean(),
    sso_connection_id: z.nullable(z.string()).optional(),
    telemetry_disabled: z.boolean(),
    updated_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_type": "accountType",
        "created_at": "createdAt",
        "free_trial_expiry": "freeTrialExpiry",
        "sso_activated": "ssoActivated",
        "sso_connection_id": "ssoConnectionId",
        "telemetry_disabled": "telemetryDisabled",
        "updated_at": "updatedAt",
    });
});
/** @internal */
exports.Organization$outboundSchema = z.object({
    accountType: accounttype_js_1.AccountType$outboundSchema,
    createdAt: z.date().transform(v => v.toISOString()),
    freeTrialExpiry: z.nullable(z.date().transform(v => v.toISOString()))
        .optional(),
    id: z.string(),
    internal: z.boolean().optional(),
    name: z.string(),
    slug: z.string(),
    ssoActivated: z.boolean(),
    ssoConnectionId: z.nullable(z.string()).optional(),
    telemetryDisabled: z.boolean(),
    updatedAt: z.date().transform(v => v.toISOString()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountType: "account_type",
        createdAt: "created_at",
        freeTrialExpiry: "free_trial_expiry",
        ssoActivated: "sso_activated",
        ssoConnectionId: "sso_connection_id",
        telemetryDisabled: "telemetry_disabled",
        updatedAt: "updated_at",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Organization$;
(function (Organization$) {
    /** @deprecated use `Organization$inboundSchema` instead. */
    Organization$.inboundSchema = exports.Organization$inboundSchema;
    /** @deprecated use `Organization$outboundSchema` instead. */
    Organization$.outboundSchema = exports.Organization$outboundSchema;
})(Organization$ || (exports.Organization$ = Organization$ = {}));
function organizationToJSON(organization) {
    return JSON.stringify(exports.Organization$outboundSchema.parse(organization));
}
function organizationFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Organization$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Organization' from JSON`);
}
