// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/organizationbillingaddonresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationBillingAddOnResponse$ = exports.OrganizationBillingAddOnResponse$outboundSchema = exports.OrganizationBillingAddOnResponse$inboundSchema = void 0;
exports.organizationBillingAddOnResponseToJSON = organizationBillingAddOnResponseToJSON;
exports.organizationBillingAddOnResponseFromJSON = organizationBillingAddOnResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
const billingaddon_js_1 = require("./billingaddon.js");
/** @internal */
exports.OrganizationBillingAddOnResponse$inboundSchema = z.object({
    add_ons: z.array(billingaddon_js_1.BillingAddOn$inboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "add_ons": "addOns",
    });
});
/** @internal */
exports.OrganizationBillingAddOnResponse$outboundSchema = z.object({
    addOns: z.array(billingaddon_js_1.BillingAddOn$outboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        addOns: "add_ons",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrganizationBillingAddOnResponse$;
(function (OrganizationBillingAddOnResponse$) {
    /** @deprecated use `OrganizationBillingAddOnResponse$inboundSchema` instead. */
    OrganizationBillingAddOnResponse$.inboundSchema = exports.OrganizationBillingAddOnResponse$inboundSchema;
    /** @deprecated use `OrganizationBillingAddOnResponse$outboundSchema` instead. */
    OrganizationBillingAddOnResponse$.outboundSchema = exports.OrganizationBillingAddOnResponse$outboundSchema;
})(OrganizationBillingAddOnResponse$ || (exports.OrganizationBillingAddOnResponse$ = OrganizationBillingAddOnResponse$ = {}));
function organizationBillingAddOnResponseToJSON(organizationBillingAddOnResponse) {
    return JSON.stringify(exports.OrganizationBillingAddOnResponse$outboundSchema.parse(organizationBillingAddOnResponse));
}
function organizationBillingAddOnResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OrganizationBillingAddOnResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrganizationBillingAddOnResponse' from JSON`);
}
