// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/organizationbillingaddonrequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationBillingAddOnRequest$ = exports.OrganizationBillingAddOnRequest$outboundSchema = exports.OrganizationBillingAddOnRequest$inboundSchema = void 0;
exports.organizationBillingAddOnRequestToJSON = organizationBillingAddOnRequestToJSON;
exports.organizationBillingAddOnRequestFromJSON = organizationBillingAddOnRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
const billingaddon_js_1 = require("./billingaddon.js");
/** @internal */
exports.OrganizationBillingAddOnRequest$inboundSchema = z.object({
    add_ons: z.array(billingaddon_js_1.BillingAddOn$inboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "add_ons": "addOns",
    });
});
/** @internal */
exports.OrganizationBillingAddOnRequest$outboundSchema = z.object({
    addOns: z.array(billingaddon_js_1.BillingAddOn$outboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        addOns: "add_ons",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrganizationBillingAddOnRequest$;
(function (OrganizationBillingAddOnRequest$) {
    /** @deprecated use `OrganizationBillingAddOnRequest$inboundSchema` instead. */
    OrganizationBillingAddOnRequest$.inboundSchema = exports.OrganizationBillingAddOnRequest$inboundSchema;
    /** @deprecated use `OrganizationBillingAddOnRequest$outboundSchema` instead. */
    OrganizationBillingAddOnRequest$.outboundSchema = exports.OrganizationBillingAddOnRequest$outboundSchema;
})(OrganizationBillingAddOnRequest$ || (exports.OrganizationBillingAddOnRequest$ = OrganizationBillingAddOnRequest$ = {}));
function organizationBillingAddOnRequestToJSON(organizationBillingAddOnRequest) {
    return JSON.stringify(exports.OrganizationBillingAddOnRequest$outboundSchema.parse(organizationBillingAddOnRequest));
}
function organizationBillingAddOnRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OrganizationBillingAddOnRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrganizationBillingAddOnRequest' from JSON`);
}
