// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/organizationusageresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationUsageResponse$ = exports.OrganizationUsageResponse$outboundSchema = exports.OrganizationUsageResponse$inboundSchema = void 0;
exports.organizationUsageResponseToJSON = organizationUsageResponseToJSON;
exports.organizationUsageResponseFromJSON = organizationUsageResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
const organizationusage_js_1 = require("./organizationusage.js");
/** @internal */
exports.OrganizationUsageResponse$inboundSchema = z.object({
    allowed_languages: z.array(z.string()),
    free_trial_expiry: z.string().datetime({ offset: true }).transform(v => new Date(v)).optional(),
    total_allowed_languages: z.number().int(),
    usage: z.array(organizationusage_js_1.OrganizationUsage$inboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "allowed_languages": "allowedLanguages",
        "free_trial_expiry": "freeTrialExpiry",
        "total_allowed_languages": "totalAllowedLanguages",
    });
});
/** @internal */
exports.OrganizationUsageResponse$outboundSchema = z.object({
    allowedLanguages: z.array(z.string()),
    freeTrialExpiry: z.date().transform(v => v.toISOString()).optional(),
    totalAllowedLanguages: z.number().int(),
    usage: z.array(organizationusage_js_1.OrganizationUsage$outboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        allowedLanguages: "allowed_languages",
        freeTrialExpiry: "free_trial_expiry",
        totalAllowedLanguages: "total_allowed_languages",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrganizationUsageResponse$;
(function (OrganizationUsageResponse$) {
    /** @deprecated use `OrganizationUsageResponse$inboundSchema` instead. */
    OrganizationUsageResponse$.inboundSchema = exports.OrganizationUsageResponse$inboundSchema;
    /** @deprecated use `OrganizationUsageResponse$outboundSchema` instead. */
    OrganizationUsageResponse$.outboundSchema = exports.OrganizationUsageResponse$outboundSchema;
})(OrganizationUsageResponse$ || (exports.OrganizationUsageResponse$ = OrganizationUsageResponse$ = {}));
function organizationUsageResponseToJSON(organizationUsageResponse) {
    return JSON.stringify(exports.OrganizationUsageResponse$outboundSchema.parse(organizationUsageResponse));
}
function organizationUsageResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OrganizationUsageResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrganizationUsageResponse' from JSON`);
}
