// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/preflighttoken.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PreflightToken$ = exports.PreflightToken$outboundSchema = exports.PreflightToken$inboundSchema = void 0;
exports.preflightTokenToJSON = preflightTokenToJSON;
exports.preflightTokenFromJSON = preflightTokenFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.PreflightToken$inboundSchema = z.object({
    auth_token: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "auth_token": "authToken",
    });
});
/** @internal */
exports.PreflightToken$outboundSchema = z.object({
    authToken: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        authToken: "auth_token",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PreflightToken$;
(function (PreflightToken$) {
    /** @deprecated use `PreflightToken$inboundSchema` instead. */
    PreflightToken$.inboundSchema = exports.PreflightToken$inboundSchema;
    /** @deprecated use `PreflightToken$outboundSchema` instead. */
    PreflightToken$.outboundSchema = exports.PreflightToken$outboundSchema;
})(PreflightToken$ || (exports.PreflightToken$ = PreflightToken$ = {}));
function preflightTokenToJSON(preflightToken) {
    return JSON.stringify(exports.PreflightToken$outboundSchema.parse(preflightToken));
}
function preflightTokenFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.PreflightToken$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PreflightToken' from JSON`);
}
