// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/remotesource.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RemoteSource$ = exports.RemoteSource$outboundSchema = exports.RemoteSource$inboundSchema = void 0;
exports.remoteSourceToJSON = remoteSourceToJSON;
exports.remoteSourceFromJSON = remoteSourceFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../../lib/schemas.js");
const remotedocument_js_1 = require("./remotedocument.js");
/** @internal */
exports.RemoteSource$inboundSchema = z.object({
    inputs: z.array(remotedocument_js_1.RemoteDocument$inboundSchema),
    output: remotedocument_js_1.RemoteDocument$inboundSchema,
    overlays: z.array(remotedocument_js_1.RemoteDocument$inboundSchema).optional(),
});
/** @internal */
exports.RemoteSource$outboundSchema = z.object({
    inputs: z.array(remotedocument_js_1.RemoteDocument$outboundSchema),
    output: remotedocument_js_1.RemoteDocument$outboundSchema,
    overlays: z.array(remotedocument_js_1.RemoteDocument$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RemoteSource$;
(function (RemoteSource$) {
    /** @deprecated use `RemoteSource$inboundSchema` instead. */
    RemoteSource$.inboundSchema = exports.RemoteSource$inboundSchema;
    /** @deprecated use `RemoteSource$outboundSchema` instead. */
    RemoteSource$.outboundSchema = exports.RemoteSource$outboundSchema;
})(RemoteSource$ || (exports.RemoteSource$ = RemoteSource$ = {}));
function remoteSourceToJSON(remoteSource) {
    return JSON.stringify(exports.RemoteSource$outboundSchema.parse(remoteSource));
}
function remoteSourceFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.RemoteSource$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoteSource' from JSON`);
}
