// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/remotesourcesubscriptionsettings.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RemoteSourceSubscriptionSettings$ = exports.RemoteSourceSubscriptionSettings$outboundSchema = exports.RemoteSourceSubscriptionSettings$inboundSchema = void 0;
exports.remoteSourceSubscriptionSettingsToJSON = remoteSourceSubscriptionSettingsToJSON;
exports.remoteSourceSubscriptionSettingsFromJSON = remoteSourceSubscriptionSettingsFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.RemoteSourceSubscriptionSettings$inboundSchema = z.object({
    base_spec_namespaces: z.array(z.string()),
    ignored_namespaces: z.array(z.string()).optional(),
    output_namespace: z.string(),
    overlay_namespaces: z.array(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "base_spec_namespaces": "baseSpecNamespaces",
        "ignored_namespaces": "ignoredNamespaces",
        "output_namespace": "outputNamespace",
        "overlay_namespaces": "overlayNamespaces",
    });
});
/** @internal */
exports.RemoteSourceSubscriptionSettings$outboundSchema = z.object({
    baseSpecNamespaces: z.array(z.string()),
    ignoredNamespaces: z.array(z.string()).optional(),
    outputNamespace: z.string(),
    overlayNamespaces: z.array(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        baseSpecNamespaces: "base_spec_namespaces",
        ignoredNamespaces: "ignored_namespaces",
        outputNamespace: "output_namespace",
        overlayNamespaces: "overlay_namespaces",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RemoteSourceSubscriptionSettings$;
(function (RemoteSourceSubscriptionSettings$) {
    /** @deprecated use `RemoteSourceSubscriptionSettings$inboundSchema` instead. */
    RemoteSourceSubscriptionSettings$.inboundSchema = exports.RemoteSourceSubscriptionSettings$inboundSchema;
    /** @deprecated use `RemoteSourceSubscriptionSettings$outboundSchema` instead. */
    RemoteSourceSubscriptionSettings$.outboundSchema = exports.RemoteSourceSubscriptionSettings$outboundSchema;
})(RemoteSourceSubscriptionSettings$ || (exports.RemoteSourceSubscriptionSettings$ = RemoteSourceSubscriptionSettings$ = {}));
function remoteSourceSubscriptionSettingsToJSON(remoteSourceSubscriptionSettings) {
    return JSON.stringify(exports.RemoteSourceSubscriptionSettings$outboundSchema.parse(remoteSourceSubscriptionSettings));
}
function remoteSourceSubscriptionSettingsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.RemoteSourceSubscriptionSettings$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoteSourceSubscriptionSettings' from JSON`);
}
