// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/remotedocument.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RemoteDocument$ = exports.RemoteDocument$outboundSchema = exports.RemoteDocument$inboundSchema = void 0;
exports.remoteDocumentToJSON = remoteDocumentToJSON;
exports.remoteDocumentFromJSON = remoteDocumentFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.RemoteDocument$inboundSchema = z.object({
    registry_url: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "registry_url": "registryUrl",
    });
});
/** @internal */
exports.RemoteDocument$outboundSchema = z.object({
    registryUrl: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        registryUrl: "registry_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RemoteDocument$;
(function (RemoteDocument$) {
    /** @deprecated use `RemoteDocument$inboundSchema` instead. */
    RemoteDocument$.inboundSchema = exports.RemoteDocument$inboundSchema;
    /** @deprecated use `RemoteDocument$outboundSchema` instead. */
    RemoteDocument$.outboundSchema = exports.RemoteDocument$outboundSchema;
})(RemoteDocument$ || (exports.RemoteDocument$ = RemoteDocument$ = {}));
function remoteDocumentToJSON(remoteDocument) {
    return JSON.stringify(exports.RemoteDocument$outboundSchema.parse(remoteDocument));
}
function remoteDocumentFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.RemoteDocument$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoteDocument' from JSON`);
}
