// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/revision.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Revision$ = exports.Revision$outboundSchema = exports.Revision$inboundSchema = void 0;
exports.revisionToJSON = revisionToJSON;
exports.revisionFromJSON = revisionFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
const revisioncontentsmetadata_js_1 = require("./revisioncontentsmetadata.js");
/** @internal */
exports.Revision$inboundSchema = z.object({
    contents_metadata: revisioncontentsmetadata_js_1.RevisionContentsMetadata$inboundSchema.optional(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    digest: z.string(),
    id: z.string(),
    namespace_name: z.string(),
    tags: z.array(z.string()),
    updated_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "contents_metadata": "contentsMetadata",
        "created_at": "createdAt",
        "namespace_name": "namespaceName",
        "updated_at": "updatedAt",
    });
});
/** @internal */
exports.Revision$outboundSchema = z.object({
    contentsMetadata: revisioncontentsmetadata_js_1.RevisionContentsMetadata$outboundSchema.optional(),
    createdAt: z.date().transform(v => v.toISOString()),
    digest: z.string(),
    id: z.string(),
    namespaceName: z.string(),
    tags: z.array(z.string()),
    updatedAt: z.date().transform(v => v.toISOString()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentsMetadata: "contents_metadata",
        createdAt: "created_at",
        namespaceName: "namespace_name",
        updatedAt: "updated_at",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Revision$;
(function (Revision$) {
    /** @deprecated use `Revision$inboundSchema` instead. */
    Revision$.inboundSchema = exports.Revision$inboundSchema;
    /** @deprecated use `Revision$outboundSchema` instead. */
    Revision$.outboundSchema = exports.Revision$outboundSchema;
})(Revision$ || (exports.Revision$ = Revision$ = {}));
function revisionToJSON(revision) {
    return JSON.stringify(exports.Revision$outboundSchema.parse(revision));
}
function revisionFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Revision$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Revision' from JSON`);
}
