// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/schemastoreitem.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SchemaStoreItem$ = exports.SchemaStoreItem$outboundSchema = exports.SchemaStoreItem$inboundSchema = exports.Format$ = exports.Format$outboundSchema = exports.Format$inboundSchema = exports.Format = void 0;
exports.schemaStoreItemToJSON = schemaStoreItemToJSON;
exports.schemaStoreItemFromJSON = schemaStoreItemFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
exports.Format = {
    Json: "json",
    Yaml: "yaml",
};
/** @internal */
exports.Format$inboundSchema = z
    .nativeEnum(exports.Format);
/** @internal */
exports.Format$outboundSchema = exports.Format$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Format$;
(function (Format$) {
    /** @deprecated use `Format$inboundSchema` instead. */
    Format$.inboundSchema = exports.Format$inboundSchema;
    /** @deprecated use `Format$outboundSchema` instead. */
    Format$.outboundSchema = exports.Format$outboundSchema;
})(Format$ || (exports.Format$ = Format$ = {}));
/** @internal */
exports.SchemaStoreItem$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    format: exports.Format$inboundSchema,
    id: z.string(),
    packageName: z.string(),
    sdkClassname: z.string(),
    spec: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
    });
});
/** @internal */
exports.SchemaStoreItem$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    format: exports.Format$outboundSchema,
    id: z.string(),
    packageName: z.string(),
    sdkClassname: z.string(),
    spec: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SchemaStoreItem$;
(function (SchemaStoreItem$) {
    /** @deprecated use `SchemaStoreItem$inboundSchema` instead. */
    SchemaStoreItem$.inboundSchema = exports.SchemaStoreItem$inboundSchema;
    /** @deprecated use `SchemaStoreItem$outboundSchema` instead. */
    SchemaStoreItem$.outboundSchema = exports.SchemaStoreItem$outboundSchema;
})(SchemaStoreItem$ || (exports.SchemaStoreItem$ = SchemaStoreItem$ = {}));
function schemaStoreItemToJSON(schemaStoreItem) {
    return JSON.stringify(exports.SchemaStoreItem$outboundSchema.parse(schemaStoreItem));
}
function schemaStoreItemFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SchemaStoreItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SchemaStoreItem' from JSON`);
}
