// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/shorturl.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShortURL$ = exports.ShortURL$outboundSchema = exports.ShortURL$inboundSchema = void 0;
exports.shortURLToJSON = shortURLToJSON;
exports.shortURLFromJSON = shortURLFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.ShortURL$inboundSchema = z.object({
    full_url: z.string(),
    short_url: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "full_url": "fullUrl",
        "short_url": "shortUrl",
    });
});
/** @internal */
exports.ShortURL$outboundSchema = z.object({
    fullUrl: z.string(),
    shortUrl: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        fullUrl: "full_url",
        shortUrl: "short_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ShortURL$;
(function (ShortURL$) {
    /** @deprecated use `ShortURL$inboundSchema` instead. */
    ShortURL$.inboundSchema = exports.ShortURL$inboundSchema;
    /** @deprecated use `ShortURL$outboundSchema` instead. */
    ShortURL$.outboundSchema = exports.ShortURL$outboundSchema;
})(ShortURL$ || (exports.ShortURL$ = ShortURL$ = {}));
function shortURLToJSON(shortURL) {
    return JSON.stringify(exports.ShortURL$outboundSchema.parse(shortURL));
}
function shortURLFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ShortURL$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ShortURL' from JSON`);
}
