// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/ssometadata.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SSOMetadata$ = exports.SSOMetadata$outboundSchema = exports.SSOMetadata$inboundSchema = void 0;
exports.ssoMetadataToJSON = ssoMetadataToJSON;
exports.ssoMetadataFromJSON = ssoMetadataFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.SSOMetadata$inboundSchema = z.object({
    sso_activated: z.boolean(),
    sso_domains: z.array(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "sso_activated": "ssoActivated",
        "sso_domains": "ssoDomains",
    });
});
/** @internal */
exports.SSOMetadata$outboundSchema = z.object({
    ssoActivated: z.boolean(),
    ssoDomains: z.array(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        ssoActivated: "sso_activated",
        ssoDomains: "sso_domains",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SSOMetadata$;
(function (SSOMetadata$) {
    /** @deprecated use `SSOMetadata$inboundSchema` instead. */
    SSOMetadata$.inboundSchema = exports.SSOMetadata$inboundSchema;
    /** @deprecated use `SSOMetadata$outboundSchema` instead. */
    SSOMetadata$.outboundSchema = exports.SSOMetadata$outboundSchema;
})(SSOMetadata$ || (exports.SSOMetadata$ = SSOMetadata$ = {}));
function ssoMetadataToJSON(ssoMetadata) {
    return JSON.stringify(exports.SSOMetadata$outboundSchema.parse(ssoMetadata));
}
function ssoMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SSOMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SSOMetadata' from JSON`);
}
