// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/suggestoptsold.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SuggestOptsOld$ = exports.SuggestOptsOld$outboundSchema = exports.SuggestOptsOld$inboundSchema = exports.SuggestionType$ = exports.SuggestionType$outboundSchema = exports.SuggestionType$inboundSchema = exports.SuggestionType = void 0;
exports.suggestOptsOldToJSON = suggestOptsOldToJSON;
exports.suggestOptsOldFromJSON = suggestOptsOldFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
const diagnostic_js_1 = require("./diagnostic.js");
exports.SuggestionType = {
    MethodNames: "method-names",
    DiagnosticsOnly: "diagnostics-only",
};
/** @internal */
exports.SuggestionType$inboundSchema = z.nativeEnum(exports.SuggestionType);
/** @internal */
exports.SuggestionType$outboundSchema = exports.SuggestionType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SuggestionType$;
(function (SuggestionType$) {
    /** @deprecated use `SuggestionType$inboundSchema` instead. */
    SuggestionType$.inboundSchema = exports.SuggestionType$inboundSchema;
    /** @deprecated use `SuggestionType$outboundSchema` instead. */
    SuggestionType$.outboundSchema = exports.SuggestionType$outboundSchema;
})(SuggestionType$ || (exports.SuggestionType$ = SuggestionType$ = {}));
/** @internal */
exports.SuggestOptsOld$inboundSchema = z.object({
    diagnostics: z.array(diagnostic_js_1.Diagnostic$inboundSchema).optional(),
    suggestion_type: exports.SuggestionType$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "suggestion_type": "suggestionType",
    });
});
/** @internal */
exports.SuggestOptsOld$outboundSchema = z.object({
    diagnostics: z.array(diagnostic_js_1.Diagnostic$outboundSchema).optional(),
    suggestionType: exports.SuggestionType$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        suggestionType: "suggestion_type",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SuggestOptsOld$;
(function (SuggestOptsOld$) {
    /** @deprecated use `SuggestOptsOld$inboundSchema` instead. */
    SuggestOptsOld$.inboundSchema = exports.SuggestOptsOld$inboundSchema;
    /** @deprecated use `SuggestOptsOld$outboundSchema` instead. */
    SuggestOptsOld$.outboundSchema = exports.SuggestOptsOld$outboundSchema;
})(SuggestOptsOld$ || (exports.SuggestOptsOld$ = SuggestOptsOld$ = {}));
function suggestOptsOldToJSON(suggestOptsOld) {
    return JSON.stringify(exports.SuggestOptsOld$outboundSchema.parse(suggestOptsOld));
}
function suggestOptsOldFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SuggestOptsOld$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SuggestOptsOld' from JSON`);
}
