// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/tag.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tag$ = exports.Tag$outboundSchema = exports.Tag$inboundSchema = void 0;
exports.tagToJSON = tagToJSON;
exports.tagFromJSON = tagFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.Tag$inboundSchema = z
    .object({
    id: z.string(),
    name: z.string(),
    namespace_name: z.string(),
    revision_digest: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "namespace_name": "namespaceName",
        "revision_digest": "revisionDigest",
    });
});
/** @internal */
exports.Tag$outboundSchema = z
    .object({
    id: z.string(),
    name: z.string(),
    namespaceName: z.string(),
    revisionDigest: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        namespaceName: "namespace_name",
        revisionDigest: "revision_digest",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Tag$;
(function (Tag$) {
    /** @deprecated use `Tag$inboundSchema` instead. */
    Tag$.inboundSchema = exports.Tag$inboundSchema;
    /** @deprecated use `Tag$outboundSchema` instead. */
    Tag$.outboundSchema = exports.Tag$outboundSchema;
})(Tag$ || (exports.Tag$ = Tag$ = {}));
function tagToJSON(tag) {
    return JSON.stringify(exports.Tag$outboundSchema.parse(tag));
}
function tagFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Tag$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Tag' from JSON`);
}
