// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/usagesnippets.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsageSnippets$ = exports.UsageSnippets$outboundSchema = exports.UsageSnippets$inboundSchema = void 0;
exports.usageSnippetsToJSON = usageSnippetsToJSON;
exports.usageSnippetsFromJSON = usageSnippetsFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../../lib/schemas.js");
const usagesnippet_js_1 = require("./usagesnippet.js");
/** @internal */
exports.UsageSnippets$inboundSchema = z.object({
    snippets: z.array(usagesnippet_js_1.UsageSnippet$inboundSchema),
});
/** @internal */
exports.UsageSnippets$outboundSchema = z.object({
    snippets: z.array(usagesnippet_js_1.UsageSnippet$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UsageSnippets$;
(function (UsageSnippets$) {
    /** @deprecated use `UsageSnippets$inboundSchema` instead. */
    UsageSnippets$.inboundSchema = exports.UsageSnippets$inboundSchema;
    /** @deprecated use `UsageSnippets$outboundSchema` instead. */
    UsageSnippets$.outboundSchema = exports.UsageSnippets$outboundSchema;
})(UsageSnippets$ || (exports.UsageSnippets$ = UsageSnippets$ = {}));
function usageSnippetsToJSON(usageSnippets) {
    return JSON.stringify(exports.UsageSnippets$outboundSchema.parse(usageSnippets));
}
function usageSnippetsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.UsageSnippets$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UsageSnippets' from JSON`);
}
