// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/v2descriptor.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2Descriptor$ = exports.V2Descriptor$outboundSchema = exports.V2Descriptor$inboundSchema = void 0;
exports.v2DescriptorToJSON = v2DescriptorToJSON;
exports.v2DescriptorFromJSON = v2DescriptorFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../../lib/schemas.js");
const annotations_js_1 = require("./annotations.js");
/** @internal */
exports.V2Descriptor$inboundSchema = z.object({
    annotations: annotations_js_1.Annotations$inboundSchema.optional(),
    digest: z.string().optional(),
    mediaType: z.string().optional(),
    size: z.number().int().optional(),
});
/** @internal */
exports.V2Descriptor$outboundSchema = z.object({
    annotations: annotations_js_1.Annotations$outboundSchema.optional(),
    digest: z.string().optional(),
    mediaType: z.string().optional(),
    size: z.number().int().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var V2Descriptor$;
(function (V2Descriptor$) {
    /** @deprecated use `V2Descriptor$inboundSchema` instead. */
    V2Descriptor$.inboundSchema = exports.V2Descriptor$inboundSchema;
    /** @deprecated use `V2Descriptor$outboundSchema` instead. */
    V2Descriptor$.outboundSchema = exports.V2Descriptor$outboundSchema;
})(V2Descriptor$ || (exports.V2Descriptor$ = V2Descriptor$ = {}));
function v2DescriptorToJSON(v2Descriptor) {
    return JSON.stringify(exports.V2Descriptor$outboundSchema.parse(v2Descriptor));
}
function v2DescriptorFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.V2Descriptor$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'V2Descriptor' from JSON`);
}
