// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/workspacefeatureflagresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceFeatureFlagResponse$ = exports.WorkspaceFeatureFlagResponse$outboundSchema = exports.WorkspaceFeatureFlagResponse$inboundSchema = void 0;
exports.workspaceFeatureFlagResponseToJSON = workspaceFeatureFlagResponseToJSON;
exports.workspaceFeatureFlagResponseFromJSON = workspaceFeatureFlagResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
const featureflag_js_1 = require("./featureflag.js");
/** @internal */
exports.WorkspaceFeatureFlagResponse$inboundSchema = z.object({
    feature_flags: z.array(featureflag_js_1.FeatureFlag$inboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "feature_flags": "featureFlags",
    });
});
/** @internal */
exports.WorkspaceFeatureFlagResponse$outboundSchema = z.object({
    featureFlags: z.array(featureflag_js_1.FeatureFlag$outboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        featureFlags: "feature_flags",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WorkspaceFeatureFlagResponse$;
(function (WorkspaceFeatureFlagResponse$) {
    /** @deprecated use `WorkspaceFeatureFlagResponse$inboundSchema` instead. */
    WorkspaceFeatureFlagResponse$.inboundSchema = exports.WorkspaceFeatureFlagResponse$inboundSchema;
    /** @deprecated use `WorkspaceFeatureFlagResponse$outboundSchema` instead. */
    WorkspaceFeatureFlagResponse$.outboundSchema = exports.WorkspaceFeatureFlagResponse$outboundSchema;
})(WorkspaceFeatureFlagResponse$ || (exports.WorkspaceFeatureFlagResponse$ = WorkspaceFeatureFlagResponse$ = {}));
function workspaceFeatureFlagResponseToJSON(workspaceFeatureFlagResponse) {
    return JSON.stringify(exports.WorkspaceFeatureFlagResponse$outboundSchema.parse(workspaceFeatureFlagResponse));
}
function workspaceFeatureFlagResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WorkspaceFeatureFlagResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WorkspaceFeatureFlagResponse' from JSON`);
}
