// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/workspacesettings.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceSettings$ = exports.WorkspaceSettings$outboundSchema = exports.WorkspaceSettings$inboundSchema = void 0;
exports.workspaceSettingsToJSON = workspaceSettingsToJSON;
exports.workspaceSettingsFromJSON = workspaceSettingsFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.WorkspaceSettings$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    updated_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    webhook_url: z.string(),
    workspace_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "updated_at": "updatedAt",
        "webhook_url": "webhookUrl",
        "workspace_id": "workspaceId",
    });
});
/** @internal */
exports.WorkspaceSettings$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    updatedAt: z.date().transform(v => v.toISOString()),
    webhookUrl: z.string(),
    workspaceId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        updatedAt: "updated_at",
        webhookUrl: "webhook_url",
        workspaceId: "workspace_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WorkspaceSettings$;
(function (WorkspaceSettings$) {
    /** @deprecated use `WorkspaceSettings$inboundSchema` instead. */
    WorkspaceSettings$.inboundSchema = exports.WorkspaceSettings$inboundSchema;
    /** @deprecated use `WorkspaceSettings$outboundSchema` instead. */
    WorkspaceSettings$.outboundSchema = exports.WorkspaceSettings$outboundSchema;
})(WorkspaceSettings$ || (exports.WorkspaceSettings$ = WorkspaceSettings$ = {}));
function workspaceSettingsToJSON(workspaceSettings) {
    return JSON.stringify(exports.WorkspaceSettings$outboundSchema.parse(workspaceSettings));
}
function workspaceSettingsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WorkspaceSettings$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WorkspaceSettings' from JSON`);
}
