// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/workspacefeatureflagrequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceFeatureFlagRequest$ = exports.WorkspaceFeatureFlagRequest$outboundSchema = exports.WorkspaceFeatureFlagRequest$inboundSchema = void 0;
exports.workspaceFeatureFlagRequestToJSON = workspaceFeatureFlagRequestToJSON;
exports.workspaceFeatureFlagRequestFromJSON = workspaceFeatureFlagRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
const workspacefeatureflag_js_1 = require("./workspacefeatureflag.js");
/** @internal */
exports.WorkspaceFeatureFlagRequest$inboundSchema = z.object({
    feature_flags: z.array(workspacefeatureflag_js_1.WorkspaceFeatureFlag$inboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "feature_flags": "featureFlags",
    });
});
/** @internal */
exports.WorkspaceFeatureFlagRequest$outboundSchema = z.object({
    featureFlags: z.array(workspacefeatureflag_js_1.WorkspaceFeatureFlag$outboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        featureFlags: "feature_flags",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WorkspaceFeatureFlagRequest$;
(function (WorkspaceFeatureFlagRequest$) {
    /** @deprecated use `WorkspaceFeatureFlagRequest$inboundSchema` instead. */
    WorkspaceFeatureFlagRequest$.inboundSchema = exports.WorkspaceFeatureFlagRequest$inboundSchema;
    /** @deprecated use `WorkspaceFeatureFlagRequest$outboundSchema` instead. */
    WorkspaceFeatureFlagRequest$.outboundSchema = exports.WorkspaceFeatureFlagRequest$outboundSchema;
})(WorkspaceFeatureFlagRequest$ || (exports.WorkspaceFeatureFlagRequest$ = WorkspaceFeatureFlagRequest$ = {}));
function workspaceFeatureFlagRequestToJSON(workspaceFeatureFlagRequest) {
    return JSON.stringify(exports.WorkspaceFeatureFlagRequest$outboundSchema.parse(workspaceFeatureFlagRequest));
}
function workspaceFeatureFlagRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WorkspaceFeatureFlagRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WorkspaceFeatureFlagRequest' from JSON`);
}
