// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/workspaceinviteresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceInviteResponse$ = exports.WorkspaceInviteResponse$outboundSchema = exports.WorkspaceInviteResponse$inboundSchema = exports.Relationship$ = exports.Relationship$outboundSchema = exports.Relationship$inboundSchema = void 0;
exports.relationshipToJSON = relationshipToJSON;
exports.relationshipFromJSON = relationshipFromJSON;
exports.workspaceInviteResponseToJSON = workspaceInviteResponseToJSON;
exports.workspaceInviteResponseFromJSON = workspaceInviteResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.Relationship$inboundSchema = z.object({
    user_id: z.string(),
    workspace_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "user_id": "userId",
        "workspace_id": "workspaceId",
    });
});
/** @internal */
exports.Relationship$outboundSchema = z.object({
    userId: z.string(),
    workspaceId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        userId: "user_id",
        workspaceId: "workspace_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Relationship$;
(function (Relationship$) {
    /** @deprecated use `Relationship$inboundSchema` instead. */
    Relationship$.inboundSchema = exports.Relationship$inboundSchema;
    /** @deprecated use `Relationship$outboundSchema` instead. */
    Relationship$.outboundSchema = exports.Relationship$outboundSchema;
})(Relationship$ || (exports.Relationship$ = Relationship$ = {}));
function relationshipToJSON(relationship) {
    return JSON.stringify(exports.Relationship$outboundSchema.parse(relationship));
}
function relationshipFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Relationship$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Relationship' from JSON`);
}
/** @internal */
exports.WorkspaceInviteResponse$inboundSchema = z.object({
    invite_link: z.nullable(z.string()).optional(),
    relationship: z.lazy(() => exports.Relationship$inboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "invite_link": "inviteLink",
    });
});
/** @internal */
exports.WorkspaceInviteResponse$outboundSchema = z.object({
    inviteLink: z.nullable(z.string()).optional(),
    relationship: z.lazy(() => exports.Relationship$outboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        inviteLink: "invite_link",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WorkspaceInviteResponse$;
(function (WorkspaceInviteResponse$) {
    /** @deprecated use `WorkspaceInviteResponse$inboundSchema` instead. */
    WorkspaceInviteResponse$.inboundSchema = exports.WorkspaceInviteResponse$inboundSchema;
    /** @deprecated use `WorkspaceInviteResponse$outboundSchema` instead. */
    WorkspaceInviteResponse$.outboundSchema = exports.WorkspaceInviteResponse$outboundSchema;
})(WorkspaceInviteResponse$ || (exports.WorkspaceInviteResponse$ = WorkspaceInviteResponse$ = {}));
function workspaceInviteResponseToJSON(workspaceInviteResponse) {
    return JSON.stringify(exports.WorkspaceInviteResponse$outboundSchema.parse(workspaceInviteResponse));
}
function workspaceInviteResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WorkspaceInviteResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WorkspaceInviteResponse' from JSON`);
}
