// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/workspace.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workspace$ = exports.Workspace$outboundSchema = exports.Workspace$inboundSchema = void 0;
exports.workspaceToJSON = workspaceToJSON;
exports.workspaceFromJSON = workspaceFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.Workspace$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    id: z.string(),
    inactive: z.boolean().optional(),
    name: z.string(),
    organization_id: z.string(),
    slug: z.string(),
    telemetry_disabled: z.boolean().optional(),
    updated_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    verified: z.boolean(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "organization_id": "organizationId",
        "telemetry_disabled": "telemetryDisabled",
        "updated_at": "updatedAt",
    });
});
/** @internal */
exports.Workspace$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    id: z.string(),
    inactive: z.boolean().optional(),
    name: z.string(),
    organizationId: z.string(),
    slug: z.string(),
    telemetryDisabled: z.boolean().optional(),
    updatedAt: z.date().transform(v => v.toISOString()),
    verified: z.boolean(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        organizationId: "organization_id",
        telemetryDisabled: "telemetry_disabled",
        updatedAt: "updated_at",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Workspace$;
(function (Workspace$) {
    /** @deprecated use `Workspace$inboundSchema` instead. */
    Workspace$.inboundSchema = exports.Workspace$inboundSchema;
    /** @deprecated use `Workspace$outboundSchema` instead. */
    Workspace$.outboundSchema = exports.Workspace$outboundSchema;
})(Workspace$ || (exports.Workspace$ = Workspace$ = {}));
function workspaceToJSON(workspace) {
    return JSON.stringify(exports.Workspace$outboundSchema.parse(workspace));
}
function workspaceFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Workspace$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Workspace' from JSON`);
}
