// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/shared/workspaceteamresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceTeamResponse$ = exports.WorkspaceTeamResponse$outboundSchema = exports.WorkspaceTeamResponse$inboundSchema = void 0;
exports.workspaceTeamResponseToJSON = workspaceTeamResponseToJSON;
exports.workspaceTeamResponseFromJSON = workspaceTeamResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
const ssometadata_js_1 = require("./ssometadata.js");
const user_js_1 = require("./user.js");
/** @internal */
exports.WorkspaceTeamResponse$inboundSchema = z.object({
    sso_metadata: ssometadata_js_1.SSOMetadata$inboundSchema.optional(),
    users: z.array(user_js_1.User$inboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "sso_metadata": "ssoMetadata",
    });
});
/** @internal */
exports.WorkspaceTeamResponse$outboundSchema = z.object({
    ssoMetadata: ssometadata_js_1.SSOMetadata$outboundSchema.optional(),
    users: z.array(user_js_1.User$outboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        ssoMetadata: "sso_metadata",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WorkspaceTeamResponse$;
(function (WorkspaceTeamResponse$) {
    /** @deprecated use `WorkspaceTeamResponse$inboundSchema` instead. */
    WorkspaceTeamResponse$.inboundSchema = exports.WorkspaceTeamResponse$inboundSchema;
    /** @deprecated use `WorkspaceTeamResponse$outboundSchema` instead. */
    WorkspaceTeamResponse$.outboundSchema = exports.WorkspaceTeamResponse$outboundSchema;
})(WorkspaceTeamResponse$ || (exports.WorkspaceTeamResponse$ = WorkspaceTeamResponse$ = {}));
function workspaceTeamResponseToJSON(workspaceTeamResponse) {
    return JSON.stringify(exports.WorkspaceTeamResponse$outboundSchema.parse(workspaceTeamResponse));
}
function workspaceTeamResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WorkspaceTeamResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WorkspaceTeamResponse' from JSON`);
}
