// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/archivenamespace.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ArchiveNamespaceRequest$ = exports.ArchiveNamespaceRequest$outboundSchema = exports.ArchiveNamespaceRequest$inboundSchema = exports.ArchiveNamespaceRequestBody$ = exports.ArchiveNamespaceRequestBody$outboundSchema = exports.ArchiveNamespaceRequestBody$inboundSchema = void 0;
exports.archiveNamespaceRequestBodyToJSON = archiveNamespaceRequestBodyToJSON;
exports.archiveNamespaceRequestBodyFromJSON = archiveNamespaceRequestBodyFromJSON;
exports.archiveNamespaceRequestToJSON = archiveNamespaceRequestToJSON;
exports.archiveNamespaceRequestFromJSON = archiveNamespaceRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.ArchiveNamespaceRequestBody$inboundSchema = z.object({
    archived: z.boolean().default(true),
});
/** @internal */
exports.ArchiveNamespaceRequestBody$outboundSchema = z.object({
    archived: z.boolean().default(true),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ArchiveNamespaceRequestBody$;
(function (ArchiveNamespaceRequestBody$) {
    /** @deprecated use `ArchiveNamespaceRequestBody$inboundSchema` instead. */
    ArchiveNamespaceRequestBody$.inboundSchema = exports.ArchiveNamespaceRequestBody$inboundSchema;
    /** @deprecated use `ArchiveNamespaceRequestBody$outboundSchema` instead. */
    ArchiveNamespaceRequestBody$.outboundSchema = exports.ArchiveNamespaceRequestBody$outboundSchema;
})(ArchiveNamespaceRequestBody$ || (exports.ArchiveNamespaceRequestBody$ = ArchiveNamespaceRequestBody$ = {}));
function archiveNamespaceRequestBodyToJSON(archiveNamespaceRequestBody) {
    return JSON.stringify(exports.ArchiveNamespaceRequestBody$outboundSchema.parse(archiveNamespaceRequestBody));
}
function archiveNamespaceRequestBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ArchiveNamespaceRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ArchiveNamespaceRequestBody' from JSON`);
}
/** @internal */
exports.ArchiveNamespaceRequest$inboundSchema = z.object({
    namespace_name: z.string(),
    RequestBody: z.lazy(() => exports.ArchiveNamespaceRequestBody$inboundSchema)
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "namespace_name": "namespaceName",
        "RequestBody": "requestBody",
    });
});
/** @internal */
exports.ArchiveNamespaceRequest$outboundSchema = z.object({
    namespaceName: z.string(),
    requestBody: z.lazy(() => exports.ArchiveNamespaceRequestBody$outboundSchema)
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        namespaceName: "namespace_name",
        requestBody: "RequestBody",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ArchiveNamespaceRequest$;
(function (ArchiveNamespaceRequest$) {
    /** @deprecated use `ArchiveNamespaceRequest$inboundSchema` instead. */
    ArchiveNamespaceRequest$.inboundSchema = exports.ArchiveNamespaceRequest$inboundSchema;
    /** @deprecated use `ArchiveNamespaceRequest$outboundSchema` instead. */
    ArchiveNamespaceRequest$.outboundSchema = exports.ArchiveNamespaceRequest$outboundSchema;
})(ArchiveNamespaceRequest$ || (exports.ArchiveNamespaceRequest$ = ArchiveNamespaceRequest$ = {}));
function archiveNamespaceRequestToJSON(archiveNamespaceRequest) {
    return JSON.stringify(exports.ArchiveNamespaceRequest$outboundSchema.parse(archiveNamespaceRequest));
}
function archiveNamespaceRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ArchiveNamespaceRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ArchiveNamespaceRequest' from JSON`);
}
