// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/checkgithubaccess.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckGithubAccessRequest$ = exports.CheckGithubAccessRequest$outboundSchema = exports.CheckGithubAccessRequest$inboundSchema = void 0;
exports.checkGithubAccessRequestToJSON = checkGithubAccessRequestToJSON;
exports.checkGithubAccessRequestFromJSON = checkGithubAccessRequestFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.CheckGithubAccessRequest$inboundSchema = z.object({
    org: z.string(),
    repo: z.string(),
});
/** @internal */
exports.CheckGithubAccessRequest$outboundSchema = z.object({
    org: z.string(),
    repo: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckGithubAccessRequest$;
(function (CheckGithubAccessRequest$) {
    /** @deprecated use `CheckGithubAccessRequest$inboundSchema` instead. */
    CheckGithubAccessRequest$.inboundSchema = exports.CheckGithubAccessRequest$inboundSchema;
    /** @deprecated use `CheckGithubAccessRequest$outboundSchema` instead. */
    CheckGithubAccessRequest$.outboundSchema = exports.CheckGithubAccessRequest$outboundSchema;
})(CheckGithubAccessRequest$ || (exports.CheckGithubAccessRequest$ = CheckGithubAccessRequest$ = {}));
function checkGithubAccessRequestToJSON(checkGithubAccessRequest) {
    return JSON.stringify(exports.CheckGithubAccessRequest$outboundSchema.parse(checkGithubAccessRequest));
}
function checkGithubAccessRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckGithubAccessRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckGithubAccessRequest' from JSON`);
}
