// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/createpublishingtoken.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreatePublishingTokenRequestBody$ = exports.CreatePublishingTokenRequestBody$outboundSchema = exports.CreatePublishingTokenRequestBody$inboundSchema = void 0;
exports.createPublishingTokenRequestBodyToJSON = createPublishingTokenRequestBodyToJSON;
exports.createPublishingTokenRequestBodyFromJSON = createPublishingTokenRequestBodyFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.CreatePublishingTokenRequestBody$inboundSchema = z.object({
    target_id: z.string(),
    target_resource: z.string(),
    token_name: z.string(),
    valid_until: z.string().datetime({ offset: true }).transform(v => new Date(v)),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "target_id": "targetId",
        "target_resource": "targetResource",
        "token_name": "tokenName",
        "valid_until": "validUntil",
    });
});
/** @internal */
exports.CreatePublishingTokenRequestBody$outboundSchema = z.object({
    targetId: z.string(),
    targetResource: z.string(),
    tokenName: z.string(),
    validUntil: z.date().transform(v => v.toISOString()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        targetId: "target_id",
        targetResource: "target_resource",
        tokenName: "token_name",
        validUntil: "valid_until",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreatePublishingTokenRequestBody$;
(function (CreatePublishingTokenRequestBody$) {
    /** @deprecated use `CreatePublishingTokenRequestBody$inboundSchema` instead. */
    CreatePublishingTokenRequestBody$.inboundSchema = exports.CreatePublishingTokenRequestBody$inboundSchema;
    /** @deprecated use `CreatePublishingTokenRequestBody$outboundSchema` instead. */
    CreatePublishingTokenRequestBody$.outboundSchema = exports.CreatePublishingTokenRequestBody$outboundSchema;
})(CreatePublishingTokenRequestBody$ || (exports.CreatePublishingTokenRequestBody$ = CreatePublishingTokenRequestBody$ = {}));
function createPublishingTokenRequestBodyToJSON(createPublishingTokenRequestBody) {
    return JSON.stringify(exports.CreatePublishingTokenRequestBody$outboundSchema.parse(createPublishingTokenRequestBody));
}
function createPublishingTokenRequestBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CreatePublishingTokenRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreatePublishingTokenRequestBody' from JSON`);
}
