// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/createschemastoreitem.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateSchemaStoreItemRequestBody$ = exports.CreateSchemaStoreItemRequestBody$outboundSchema = exports.CreateSchemaStoreItemRequestBody$inboundSchema = exports.Format$ = exports.Format$outboundSchema = exports.Format$inboundSchema = exports.Format = void 0;
exports.createSchemaStoreItemRequestBodyToJSON = createSchemaStoreItemRequestBodyToJSON;
exports.createSchemaStoreItemRequestBodyFromJSON = createSchemaStoreItemRequestBodyFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../../lib/schemas.js");
/**
 * The format of the OpenAPI specification.
 */
exports.Format = {
    Json: "json",
    Yaml: "yaml",
};
/** @internal */
exports.Format$inboundSchema = z
    .nativeEnum(exports.Format);
/** @internal */
exports.Format$outboundSchema = exports.Format$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Format$;
(function (Format$) {
    /** @deprecated use `Format$inboundSchema` instead. */
    Format$.inboundSchema = exports.Format$inboundSchema;
    /** @deprecated use `Format$outboundSchema` instead. */
    Format$.outboundSchema = exports.Format$outboundSchema;
})(Format$ || (exports.Format$ = Format$ = {}));
/** @internal */
exports.CreateSchemaStoreItemRequestBody$inboundSchema = z.object({
    format: exports.Format$inboundSchema,
    packageName: z.string(),
    sdkClassname: z.string(),
    spec: z.string(),
});
/** @internal */
exports.CreateSchemaStoreItemRequestBody$outboundSchema = z.object({
    format: exports.Format$outboundSchema,
    packageName: z.string(),
    sdkClassname: z.string(),
    spec: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateSchemaStoreItemRequestBody$;
(function (CreateSchemaStoreItemRequestBody$) {
    /** @deprecated use `CreateSchemaStoreItemRequestBody$inboundSchema` instead. */
    CreateSchemaStoreItemRequestBody$.inboundSchema = exports.CreateSchemaStoreItemRequestBody$inboundSchema;
    /** @deprecated use `CreateSchemaStoreItemRequestBody$outboundSchema` instead. */
    CreateSchemaStoreItemRequestBody$.outboundSchema = exports.CreateSchemaStoreItemRequestBody$outboundSchema;
})(CreateSchemaStoreItemRequestBody$ || (exports.CreateSchemaStoreItemRequestBody$ = CreateSchemaStoreItemRequestBody$ = {}));
function createSchemaStoreItemRequestBodyToJSON(createSchemaStoreItemRequestBody) {
    return JSON.stringify(exports.CreateSchemaStoreItemRequestBody$outboundSchema.parse(createSchemaStoreItemRequestBody));
}
function createSchemaStoreItemRequestBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CreateSchemaStoreItemRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateSchemaStoreItemRequestBody' from JSON`);
}
