// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/deleteworkspacetoken.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteWorkspaceTokenRequest$ = exports.DeleteWorkspaceTokenRequest$outboundSchema = exports.DeleteWorkspaceTokenRequest$inboundSchema = exports.DeleteWorkspaceTokenGlobals$ = exports.DeleteWorkspaceTokenGlobals$outboundSchema = exports.DeleteWorkspaceTokenGlobals$inboundSchema = void 0;
exports.deleteWorkspaceTokenGlobalsToJSON = deleteWorkspaceTokenGlobalsToJSON;
exports.deleteWorkspaceTokenGlobalsFromJSON = deleteWorkspaceTokenGlobalsFromJSON;
exports.deleteWorkspaceTokenRequestToJSON = deleteWorkspaceTokenRequestToJSON;
exports.deleteWorkspaceTokenRequestFromJSON = deleteWorkspaceTokenRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.DeleteWorkspaceTokenGlobals$inboundSchema = z.object({
    workspace_id: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "workspace_id": "workspaceId",
    });
});
/** @internal */
exports.DeleteWorkspaceTokenGlobals$outboundSchema = z.object({
    workspaceId: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        workspaceId: "workspace_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteWorkspaceTokenGlobals$;
(function (DeleteWorkspaceTokenGlobals$) {
    /** @deprecated use `DeleteWorkspaceTokenGlobals$inboundSchema` instead. */
    DeleteWorkspaceTokenGlobals$.inboundSchema = exports.DeleteWorkspaceTokenGlobals$inboundSchema;
    /** @deprecated use `DeleteWorkspaceTokenGlobals$outboundSchema` instead. */
    DeleteWorkspaceTokenGlobals$.outboundSchema = exports.DeleteWorkspaceTokenGlobals$outboundSchema;
})(DeleteWorkspaceTokenGlobals$ || (exports.DeleteWorkspaceTokenGlobals$ = DeleteWorkspaceTokenGlobals$ = {}));
function deleteWorkspaceTokenGlobalsToJSON(deleteWorkspaceTokenGlobals) {
    return JSON.stringify(exports.DeleteWorkspaceTokenGlobals$outboundSchema.parse(deleteWorkspaceTokenGlobals));
}
function deleteWorkspaceTokenGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DeleteWorkspaceTokenGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteWorkspaceTokenGlobals' from JSON`);
}
/** @internal */
exports.DeleteWorkspaceTokenRequest$inboundSchema = z.object({
    workspace_id: z.string().optional(),
    tokenID: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "workspace_id": "workspaceId",
    });
});
/** @internal */
exports.DeleteWorkspaceTokenRequest$outboundSchema = z.object({
    workspaceId: z.string().optional(),
    tokenID: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        workspaceId: "workspace_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteWorkspaceTokenRequest$;
(function (DeleteWorkspaceTokenRequest$) {
    /** @deprecated use `DeleteWorkspaceTokenRequest$inboundSchema` instead. */
    DeleteWorkspaceTokenRequest$.inboundSchema = exports.DeleteWorkspaceTokenRequest$inboundSchema;
    /** @deprecated use `DeleteWorkspaceTokenRequest$outboundSchema` instead. */
    DeleteWorkspaceTokenRequest$.outboundSchema = exports.DeleteWorkspaceTokenRequest$outboundSchema;
})(DeleteWorkspaceTokenRequest$ || (exports.DeleteWorkspaceTokenRequest$ = DeleteWorkspaceTokenRequest$ = {}));
function deleteWorkspaceTokenRequestToJSON(deleteWorkspaceTokenRequest) {
    return JSON.stringify(exports.DeleteWorkspaceTokenRequest$outboundSchema.parse(deleteWorkspaceTokenRequest));
}
function deleteWorkspaceTokenRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DeleteWorkspaceTokenRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteWorkspaceTokenRequest' from JSON`);
}
