// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/deletebillingaddon.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteBillingAddOnRequest$ = exports.DeleteBillingAddOnRequest$outboundSchema = exports.DeleteBillingAddOnRequest$inboundSchema = void 0;
exports.deleteBillingAddOnRequestToJSON = deleteBillingAddOnRequestToJSON;
exports.deleteBillingAddOnRequestFromJSON = deleteBillingAddOnRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
const shared = __importStar(require("../shared/index.js"));
/** @internal */
exports.DeleteBillingAddOnRequest$inboundSchema = z.object({
    add_on: shared.BillingAddOn$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "add_on": "addOn",
    });
});
/** @internal */
exports.DeleteBillingAddOnRequest$outboundSchema = z.object({
    addOn: shared.BillingAddOn$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        addOn: "add_on",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteBillingAddOnRequest$;
(function (DeleteBillingAddOnRequest$) {
    /** @deprecated use `DeleteBillingAddOnRequest$inboundSchema` instead. */
    DeleteBillingAddOnRequest$.inboundSchema = exports.DeleteBillingAddOnRequest$inboundSchema;
    /** @deprecated use `DeleteBillingAddOnRequest$outboundSchema` instead. */
    DeleteBillingAddOnRequest$.outboundSchema = exports.DeleteBillingAddOnRequest$outboundSchema;
})(DeleteBillingAddOnRequest$ || (exports.DeleteBillingAddOnRequest$ = DeleteBillingAddOnRequest$ = {}));
function deleteBillingAddOnRequestToJSON(deleteBillingAddOnRequest) {
    return JSON.stringify(exports.DeleteBillingAddOnRequest$outboundSchema.parse(deleteBillingAddOnRequest));
}
function deleteBillingAddOnRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DeleteBillingAddOnRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteBillingAddOnRequest' from JSON`);
}
