// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/getblob.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetBlobRequest$ = exports.GetBlobRequest$outboundSchema = exports.GetBlobRequest$inboundSchema = void 0;
exports.getBlobRequestToJSON = getBlobRequestToJSON;
exports.getBlobRequestFromJSON = getBlobRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.GetBlobRequest$inboundSchema = z.object({
    organization_slug: z.string(),
    workspace_slug: z.string(),
    namespace_name: z.string(),
    digest: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_slug": "organizationSlug",
        "workspace_slug": "workspaceSlug",
        "namespace_name": "namespaceName",
    });
});
/** @internal */
exports.GetBlobRequest$outboundSchema = z.object({
    organizationSlug: z.string(),
    workspaceSlug: z.string(),
    namespaceName: z.string(),
    digest: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationSlug: "organization_slug",
        workspaceSlug: "workspace_slug",
        namespaceName: "namespace_name",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetBlobRequest$;
(function (GetBlobRequest$) {
    /** @deprecated use `GetBlobRequest$inboundSchema` instead. */
    GetBlobRequest$.inboundSchema = exports.GetBlobRequest$inboundSchema;
    /** @deprecated use `GetBlobRequest$outboundSchema` instead. */
    GetBlobRequest$.outboundSchema = exports.GetBlobRequest$outboundSchema;
})(GetBlobRequest$ || (exports.GetBlobRequest$ = GetBlobRequest$ = {}));
function getBlobRequestToJSON(getBlobRequest) {
    return JSON.stringify(exports.GetBlobRequest$outboundSchema.parse(getBlobRequest));
}
function getBlobRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetBlobRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetBlobRequest' from JSON`);
}
