// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/getcodesamples.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetCodeSamplesRequest$ = exports.GetCodeSamplesRequest$outboundSchema = exports.GetCodeSamplesRequest$inboundSchema = exports.MethodPaths$ = exports.MethodPaths$outboundSchema = exports.MethodPaths$inboundSchema = void 0;
exports.methodPathsToJSON = methodPathsToJSON;
exports.methodPathsFromJSON = methodPathsFromJSON;
exports.getCodeSamplesRequestToJSON = getCodeSamplesRequestToJSON;
exports.getCodeSamplesRequestFromJSON = getCodeSamplesRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
const shared = __importStar(require("../shared/index.js"));
/** @internal */
exports.MethodPaths$inboundSchema = z.object({
    method: shared.HttpMethod$inboundSchema,
    path: z.string(),
});
/** @internal */
exports.MethodPaths$outboundSchema = z.object({
    method: shared.HttpMethod$outboundSchema,
    path: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var MethodPaths$;
(function (MethodPaths$) {
    /** @deprecated use `MethodPaths$inboundSchema` instead. */
    MethodPaths$.inboundSchema = exports.MethodPaths$inboundSchema;
    /** @deprecated use `MethodPaths$outboundSchema` instead. */
    MethodPaths$.outboundSchema = exports.MethodPaths$outboundSchema;
})(MethodPaths$ || (exports.MethodPaths$ = MethodPaths$ = {}));
function methodPathsToJSON(methodPaths) {
    return JSON.stringify(exports.MethodPaths$outboundSchema.parse(methodPaths));
}
function methodPathsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MethodPaths$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MethodPaths' from JSON`);
}
/** @internal */
exports.GetCodeSamplesRequest$inboundSchema = z.object({
    registry_url: z.string(),
    operation_ids: z.array(z.string()).optional(),
    method_paths: z.array(z.lazy(() => exports.MethodPaths$inboundSchema)).optional(),
    languages: z.array(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "registry_url": "registryUrl",
        "operation_ids": "operationIds",
        "method_paths": "methodPaths",
    });
});
/** @internal */
exports.GetCodeSamplesRequest$outboundSchema = z.object({
    registryUrl: z.string(),
    operationIds: z.array(z.string()).optional(),
    methodPaths: z.array(z.lazy(() => exports.MethodPaths$outboundSchema)).optional(),
    languages: z.array(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        registryUrl: "registry_url",
        operationIds: "operation_ids",
        methodPaths: "method_paths",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetCodeSamplesRequest$;
(function (GetCodeSamplesRequest$) {
    /** @deprecated use `GetCodeSamplesRequest$inboundSchema` instead. */
    GetCodeSamplesRequest$.inboundSchema = exports.GetCodeSamplesRequest$inboundSchema;
    /** @deprecated use `GetCodeSamplesRequest$outboundSchema` instead. */
    GetCodeSamplesRequest$.outboundSchema = exports.GetCodeSamplesRequest$outboundSchema;
})(GetCodeSamplesRequest$ || (exports.GetCodeSamplesRequest$ = GetCodeSamplesRequest$ = {}));
function getCodeSamplesRequestToJSON(getCodeSamplesRequest) {
    return JSON.stringify(exports.GetCodeSamplesRequest$outboundSchema.parse(getCodeSamplesRequest));
}
function getCodeSamplesRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetCodeSamplesRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCodeSamplesRequest' from JSON`);
}
