// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/getcodesamplepreviewasync.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetCodeSamplePreviewAsyncResponse$ = exports.GetCodeSamplePreviewAsyncResponse$outboundSchema = exports.GetCodeSamplePreviewAsyncResponse$inboundSchema = exports.GetCodeSamplePreviewAsyncResponseBody$ = exports.GetCodeSamplePreviewAsyncResponseBody$outboundSchema = exports.GetCodeSamplePreviewAsyncResponseBody$inboundSchema = exports.GetCodeSamplePreviewAsyncRequest$ = exports.GetCodeSamplePreviewAsyncRequest$outboundSchema = exports.GetCodeSamplePreviewAsyncRequest$inboundSchema = void 0;
exports.getCodeSamplePreviewAsyncRequestToJSON = getCodeSamplePreviewAsyncRequestToJSON;
exports.getCodeSamplePreviewAsyncRequestFromJSON = getCodeSamplePreviewAsyncRequestFromJSON;
exports.getCodeSamplePreviewAsyncResponseBodyToJSON = getCodeSamplePreviewAsyncResponseBodyToJSON;
exports.getCodeSamplePreviewAsyncResponseBodyFromJSON = getCodeSamplePreviewAsyncResponseBodyFromJSON;
exports.getCodeSamplePreviewAsyncResponseToJSON = getCodeSamplePreviewAsyncResponseToJSON;
exports.getCodeSamplePreviewAsyncResponseFromJSON = getCodeSamplePreviewAsyncResponseFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../../lib/schemas.js");
const shared = __importStar(require("../shared/index.js"));
/** @internal */
exports.GetCodeSamplePreviewAsyncRequest$inboundSchema = z.object({
    jobID: z.string(),
});
/** @internal */
exports.GetCodeSamplePreviewAsyncRequest$outboundSchema = z.object({
    jobID: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetCodeSamplePreviewAsyncRequest$;
(function (GetCodeSamplePreviewAsyncRequest$) {
    /** @deprecated use `GetCodeSamplePreviewAsyncRequest$inboundSchema` instead. */
    GetCodeSamplePreviewAsyncRequest$.inboundSchema = exports.GetCodeSamplePreviewAsyncRequest$inboundSchema;
    /** @deprecated use `GetCodeSamplePreviewAsyncRequest$outboundSchema` instead. */
    GetCodeSamplePreviewAsyncRequest$.outboundSchema = exports.GetCodeSamplePreviewAsyncRequest$outboundSchema;
})(GetCodeSamplePreviewAsyncRequest$ || (exports.GetCodeSamplePreviewAsyncRequest$ = GetCodeSamplePreviewAsyncRequest$ = {}));
function getCodeSamplePreviewAsyncRequestToJSON(getCodeSamplePreviewAsyncRequest) {
    return JSON.stringify(exports.GetCodeSamplePreviewAsyncRequest$outboundSchema.parse(getCodeSamplePreviewAsyncRequest));
}
function getCodeSamplePreviewAsyncRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetCodeSamplePreviewAsyncRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCodeSamplePreviewAsyncRequest' from JSON`);
}
/** @internal */
exports.GetCodeSamplePreviewAsyncResponseBody$inboundSchema = z.object({
    status: shared.CodeSamplesJobStatus$inboundSchema,
});
/** @internal */
exports.GetCodeSamplePreviewAsyncResponseBody$outboundSchema = z.object({
    status: shared.CodeSamplesJobStatus$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetCodeSamplePreviewAsyncResponseBody$;
(function (GetCodeSamplePreviewAsyncResponseBody$) {
    /** @deprecated use `GetCodeSamplePreviewAsyncResponseBody$inboundSchema` instead. */
    GetCodeSamplePreviewAsyncResponseBody$.inboundSchema = exports.GetCodeSamplePreviewAsyncResponseBody$inboundSchema;
    /** @deprecated use `GetCodeSamplePreviewAsyncResponseBody$outboundSchema` instead. */
    GetCodeSamplePreviewAsyncResponseBody$.outboundSchema = exports.GetCodeSamplePreviewAsyncResponseBody$outboundSchema;
})(GetCodeSamplePreviewAsyncResponseBody$ || (exports.GetCodeSamplePreviewAsyncResponseBody$ = GetCodeSamplePreviewAsyncResponseBody$ = {}));
function getCodeSamplePreviewAsyncResponseBodyToJSON(getCodeSamplePreviewAsyncResponseBody) {
    return JSON.stringify(exports.GetCodeSamplePreviewAsyncResponseBody$outboundSchema.parse(getCodeSamplePreviewAsyncResponseBody));
}
function getCodeSamplePreviewAsyncResponseBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetCodeSamplePreviewAsyncResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCodeSamplePreviewAsyncResponseBody' from JSON`);
}
/** @internal */
exports.GetCodeSamplePreviewAsyncResponse$inboundSchema = z.union([
    shared.UsageSnippets$inboundSchema,
    z.lazy(() => exports.GetCodeSamplePreviewAsyncResponseBody$inboundSchema),
]);
/** @internal */
exports.GetCodeSamplePreviewAsyncResponse$outboundSchema = z.union([
    shared.UsageSnippets$outboundSchema,
    z.lazy(() => exports.GetCodeSamplePreviewAsyncResponseBody$outboundSchema),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetCodeSamplePreviewAsyncResponse$;
(function (GetCodeSamplePreviewAsyncResponse$) {
    /** @deprecated use `GetCodeSamplePreviewAsyncResponse$inboundSchema` instead. */
    GetCodeSamplePreviewAsyncResponse$.inboundSchema = exports.GetCodeSamplePreviewAsyncResponse$inboundSchema;
    /** @deprecated use `GetCodeSamplePreviewAsyncResponse$outboundSchema` instead. */
    GetCodeSamplePreviewAsyncResponse$.outboundSchema = exports.GetCodeSamplePreviewAsyncResponse$outboundSchema;
})(GetCodeSamplePreviewAsyncResponse$ || (exports.GetCodeSamplePreviewAsyncResponse$ = GetCodeSamplePreviewAsyncResponse$ = {}));
function getCodeSamplePreviewAsyncResponseToJSON(getCodeSamplePreviewAsyncResponse) {
    return JSON.stringify(exports.GetCodeSamplePreviewAsyncResponse$outboundSchema.parse(getCodeSamplePreviewAsyncResponse));
}
function getCodeSamplePreviewAsyncResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetCodeSamplePreviewAsyncResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCodeSamplePreviewAsyncResponse' from JSON`);
}
