// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/getchangesreportsignedurl.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetChangesReportSignedUrlSignedAccess$ = exports.GetChangesReportSignedUrlSignedAccess$outboundSchema = exports.GetChangesReportSignedUrlSignedAccess$inboundSchema = exports.GetChangesReportSignedUrlRequest$ = exports.GetChangesReportSignedUrlRequest$outboundSchema = exports.GetChangesReportSignedUrlRequest$inboundSchema = void 0;
exports.getChangesReportSignedUrlRequestToJSON = getChangesReportSignedUrlRequestToJSON;
exports.getChangesReportSignedUrlRequestFromJSON = getChangesReportSignedUrlRequestFromJSON;
exports.getChangesReportSignedUrlSignedAccessToJSON = getChangesReportSignedUrlSignedAccessToJSON;
exports.getChangesReportSignedUrlSignedAccessFromJSON = getChangesReportSignedUrlSignedAccessFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.GetChangesReportSignedUrlRequest$inboundSchema = z.object({
    documentChecksum: z.string(),
});
/** @internal */
exports.GetChangesReportSignedUrlRequest$outboundSchema = z.object({
    documentChecksum: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetChangesReportSignedUrlRequest$;
(function (GetChangesReportSignedUrlRequest$) {
    /** @deprecated use `GetChangesReportSignedUrlRequest$inboundSchema` instead. */
    GetChangesReportSignedUrlRequest$.inboundSchema = exports.GetChangesReportSignedUrlRequest$inboundSchema;
    /** @deprecated use `GetChangesReportSignedUrlRequest$outboundSchema` instead. */
    GetChangesReportSignedUrlRequest$.outboundSchema = exports.GetChangesReportSignedUrlRequest$outboundSchema;
})(GetChangesReportSignedUrlRequest$ || (exports.GetChangesReportSignedUrlRequest$ = GetChangesReportSignedUrlRequest$ = {}));
function getChangesReportSignedUrlRequestToJSON(getChangesReportSignedUrlRequest) {
    return JSON.stringify(exports.GetChangesReportSignedUrlRequest$outboundSchema.parse(getChangesReportSignedUrlRequest));
}
function getChangesReportSignedUrlRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetChangesReportSignedUrlRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetChangesReportSignedUrlRequest' from JSON`);
}
/** @internal */
exports.GetChangesReportSignedUrlSignedAccess$inboundSchema = z.object({
    url: z.string(),
});
/** @internal */
exports.GetChangesReportSignedUrlSignedAccess$outboundSchema = z.object({
    url: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetChangesReportSignedUrlSignedAccess$;
(function (GetChangesReportSignedUrlSignedAccess$) {
    /** @deprecated use `GetChangesReportSignedUrlSignedAccess$inboundSchema` instead. */
    GetChangesReportSignedUrlSignedAccess$.inboundSchema = exports.GetChangesReportSignedUrlSignedAccess$inboundSchema;
    /** @deprecated use `GetChangesReportSignedUrlSignedAccess$outboundSchema` instead. */
    GetChangesReportSignedUrlSignedAccess$.outboundSchema = exports.GetChangesReportSignedUrlSignedAccess$outboundSchema;
})(GetChangesReportSignedUrlSignedAccess$ || (exports.GetChangesReportSignedUrlSignedAccess$ = GetChangesReportSignedUrlSignedAccess$ = {}));
function getChangesReportSignedUrlSignedAccessToJSON(getChangesReportSignedUrlSignedAccess) {
    return JSON.stringify(exports.GetChangesReportSignedUrlSignedAccess$outboundSchema.parse(getChangesReportSignedUrlSignedAccess));
}
function getChangesReportSignedUrlSignedAccessFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetChangesReportSignedUrlSignedAccess$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetChangesReportSignedUrlSignedAccess' from JSON`);
}
