// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/getlintingreportsignedurl.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetLintingReportSignedUrlSignedAccess$ = exports.GetLintingReportSignedUrlSignedAccess$outboundSchema = exports.GetLintingReportSignedUrlSignedAccess$inboundSchema = exports.GetLintingReportSignedUrlRequest$ = exports.GetLintingReportSignedUrlRequest$outboundSchema = exports.GetLintingReportSignedUrlRequest$inboundSchema = void 0;
exports.getLintingReportSignedUrlRequestToJSON = getLintingReportSignedUrlRequestToJSON;
exports.getLintingReportSignedUrlRequestFromJSON = getLintingReportSignedUrlRequestFromJSON;
exports.getLintingReportSignedUrlSignedAccessToJSON = getLintingReportSignedUrlSignedAccessToJSON;
exports.getLintingReportSignedUrlSignedAccessFromJSON = getLintingReportSignedUrlSignedAccessFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.GetLintingReportSignedUrlRequest$inboundSchema = z.object({
    documentChecksum: z.string(),
});
/** @internal */
exports.GetLintingReportSignedUrlRequest$outboundSchema = z.object({
    documentChecksum: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetLintingReportSignedUrlRequest$;
(function (GetLintingReportSignedUrlRequest$) {
    /** @deprecated use `GetLintingReportSignedUrlRequest$inboundSchema` instead. */
    GetLintingReportSignedUrlRequest$.inboundSchema = exports.GetLintingReportSignedUrlRequest$inboundSchema;
    /** @deprecated use `GetLintingReportSignedUrlRequest$outboundSchema` instead. */
    GetLintingReportSignedUrlRequest$.outboundSchema = exports.GetLintingReportSignedUrlRequest$outboundSchema;
})(GetLintingReportSignedUrlRequest$ || (exports.GetLintingReportSignedUrlRequest$ = GetLintingReportSignedUrlRequest$ = {}));
function getLintingReportSignedUrlRequestToJSON(getLintingReportSignedUrlRequest) {
    return JSON.stringify(exports.GetLintingReportSignedUrlRequest$outboundSchema.parse(getLintingReportSignedUrlRequest));
}
function getLintingReportSignedUrlRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetLintingReportSignedUrlRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetLintingReportSignedUrlRequest' from JSON`);
}
/** @internal */
exports.GetLintingReportSignedUrlSignedAccess$inboundSchema = z.object({
    url: z.string(),
});
/** @internal */
exports.GetLintingReportSignedUrlSignedAccess$outboundSchema = z.object({
    url: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetLintingReportSignedUrlSignedAccess$;
(function (GetLintingReportSignedUrlSignedAccess$) {
    /** @deprecated use `GetLintingReportSignedUrlSignedAccess$inboundSchema` instead. */
    GetLintingReportSignedUrlSignedAccess$.inboundSchema = exports.GetLintingReportSignedUrlSignedAccess$inboundSchema;
    /** @deprecated use `GetLintingReportSignedUrlSignedAccess$outboundSchema` instead. */
    GetLintingReportSignedUrlSignedAccess$.outboundSchema = exports.GetLintingReportSignedUrlSignedAccess$outboundSchema;
})(GetLintingReportSignedUrlSignedAccess$ || (exports.GetLintingReportSignedUrlSignedAccess$ = GetLintingReportSignedUrlSignedAccess$ = {}));
function getLintingReportSignedUrlSignedAccessToJSON(getLintingReportSignedUrlSignedAccess) {
    return JSON.stringify(exports.GetLintingReportSignedUrlSignedAccess$outboundSchema.parse(getLintingReportSignedUrlSignedAccess));
}
function getLintingReportSignedUrlSignedAccessFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetLintingReportSignedUrlSignedAccess$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetLintingReportSignedUrlSignedAccess' from JSON`);
}
