// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/getmanifest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetManifestRequest$ = exports.GetManifestRequest$outboundSchema = exports.GetManifestRequest$inboundSchema = void 0;
exports.getManifestRequestToJSON = getManifestRequestToJSON;
exports.getManifestRequestFromJSON = getManifestRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.GetManifestRequest$inboundSchema = z.object({
    organization_slug: z.string(),
    workspace_slug: z.string(),
    namespace_name: z.string(),
    revision_reference: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_slug": "organizationSlug",
        "workspace_slug": "workspaceSlug",
        "namespace_name": "namespaceName",
        "revision_reference": "revisionReference",
    });
});
/** @internal */
exports.GetManifestRequest$outboundSchema = z.object({
    organizationSlug: z.string(),
    workspaceSlug: z.string(),
    namespaceName: z.string(),
    revisionReference: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationSlug: "organization_slug",
        workspaceSlug: "workspace_slug",
        namespaceName: "namespace_name",
        revisionReference: "revision_reference",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetManifestRequest$;
(function (GetManifestRequest$) {
    /** @deprecated use `GetManifestRequest$inboundSchema` instead. */
    GetManifestRequest$.inboundSchema = exports.GetManifestRequest$inboundSchema;
    /** @deprecated use `GetManifestRequest$outboundSchema` instead. */
    GetManifestRequest$.outboundSchema = exports.GetManifestRequest$outboundSchema;
})(GetManifestRequest$ || (exports.GetManifestRequest$ = GetManifestRequest$ = {}));
function getManifestRequestToJSON(getManifestRequest) {
    return JSON.stringify(exports.GetManifestRequest$outboundSchema.parse(getManifestRequest));
}
function getManifestRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetManifestRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetManifestRequest' from JSON`);
}
