// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/getpublishingtokentargetbyid.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetPublishingTokenTargetByIDRequest$ = exports.GetPublishingTokenTargetByIDRequest$outboundSchema = exports.GetPublishingTokenTargetByIDRequest$inboundSchema = void 0;
exports.getPublishingTokenTargetByIDRequestToJSON = getPublishingTokenTargetByIDRequestToJSON;
exports.getPublishingTokenTargetByIDRequestFromJSON = getPublishingTokenTargetByIDRequestFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.GetPublishingTokenTargetByIDRequest$inboundSchema = z.object({
    tokenID: z.string(),
});
/** @internal */
exports.GetPublishingTokenTargetByIDRequest$outboundSchema = z.object({
    tokenID: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetPublishingTokenTargetByIDRequest$;
(function (GetPublishingTokenTargetByIDRequest$) {
    /** @deprecated use `GetPublishingTokenTargetByIDRequest$inboundSchema` instead. */
    GetPublishingTokenTargetByIDRequest$.inboundSchema = exports.GetPublishingTokenTargetByIDRequest$inboundSchema;
    /** @deprecated use `GetPublishingTokenTargetByIDRequest$outboundSchema` instead. */
    GetPublishingTokenTargetByIDRequest$.outboundSchema = exports.GetPublishingTokenTargetByIDRequest$outboundSchema;
})(GetPublishingTokenTargetByIDRequest$ || (exports.GetPublishingTokenTargetByIDRequest$ = GetPublishingTokenTargetByIDRequest$ = {}));
function getPublishingTokenTargetByIDRequestToJSON(getPublishingTokenTargetByIDRequest) {
    return JSON.stringify(exports.GetPublishingTokenTargetByIDRequest$outboundSchema.parse(getPublishingTokenTargetByIDRequest));
}
function getPublishingTokenTargetByIDRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetPublishingTokenTargetByIDRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetPublishingTokenTargetByIDRequest' from JSON`);
}
