// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/gettags.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetTagsRequest$ = exports.GetTagsRequest$outboundSchema = exports.GetTagsRequest$inboundSchema = void 0;
exports.getTagsRequestToJSON = getTagsRequestToJSON;
exports.getTagsRequestFromJSON = getTagsRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.GetTagsRequest$inboundSchema = z.object({
    namespace_name: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "namespace_name": "namespaceName",
    });
});
/** @internal */
exports.GetTagsRequest$outboundSchema = z.object({
    namespaceName: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        namespaceName: "namespace_name",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetTagsRequest$;
(function (GetTagsRequest$) {
    /** @deprecated use `GetTagsRequest$inboundSchema` instead. */
    GetTagsRequest$.inboundSchema = exports.GetTagsRequest$inboundSchema;
    /** @deprecated use `GetTagsRequest$outboundSchema` instead. */
    GetTagsRequest$.outboundSchema = exports.GetTagsRequest$outboundSchema;
})(GetTagsRequest$ || (exports.GetTagsRequest$ = GetTagsRequest$ = {}));
function getTagsRequestToJSON(getTagsRequest) {
    return JSON.stringify(exports.GetTagsRequest$outboundSchema.parse(getTagsRequest));
}
function getTagsRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetTagsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetTagsRequest' from JSON`);
}
