// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/getworkspacefeatureflags.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetWorkspaceFeatureFlagsRequest$ = exports.GetWorkspaceFeatureFlagsRequest$outboundSchema = exports.GetWorkspaceFeatureFlagsRequest$inboundSchema = exports.GetWorkspaceFeatureFlagsGlobals$ = exports.GetWorkspaceFeatureFlagsGlobals$outboundSchema = exports.GetWorkspaceFeatureFlagsGlobals$inboundSchema = void 0;
exports.getWorkspaceFeatureFlagsGlobalsToJSON = getWorkspaceFeatureFlagsGlobalsToJSON;
exports.getWorkspaceFeatureFlagsGlobalsFromJSON = getWorkspaceFeatureFlagsGlobalsFromJSON;
exports.getWorkspaceFeatureFlagsRequestToJSON = getWorkspaceFeatureFlagsRequestToJSON;
exports.getWorkspaceFeatureFlagsRequestFromJSON = getWorkspaceFeatureFlagsRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.GetWorkspaceFeatureFlagsGlobals$inboundSchema = z.object({
    workspace_id: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "workspace_id": "workspaceId",
    });
});
/** @internal */
exports.GetWorkspaceFeatureFlagsGlobals$outboundSchema = z.object({
    workspaceId: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        workspaceId: "workspace_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetWorkspaceFeatureFlagsGlobals$;
(function (GetWorkspaceFeatureFlagsGlobals$) {
    /** @deprecated use `GetWorkspaceFeatureFlagsGlobals$inboundSchema` instead. */
    GetWorkspaceFeatureFlagsGlobals$.inboundSchema = exports.GetWorkspaceFeatureFlagsGlobals$inboundSchema;
    /** @deprecated use `GetWorkspaceFeatureFlagsGlobals$outboundSchema` instead. */
    GetWorkspaceFeatureFlagsGlobals$.outboundSchema = exports.GetWorkspaceFeatureFlagsGlobals$outboundSchema;
})(GetWorkspaceFeatureFlagsGlobals$ || (exports.GetWorkspaceFeatureFlagsGlobals$ = GetWorkspaceFeatureFlagsGlobals$ = {}));
function getWorkspaceFeatureFlagsGlobalsToJSON(getWorkspaceFeatureFlagsGlobals) {
    return JSON.stringify(exports.GetWorkspaceFeatureFlagsGlobals$outboundSchema.parse(getWorkspaceFeatureFlagsGlobals));
}
function getWorkspaceFeatureFlagsGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetWorkspaceFeatureFlagsGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetWorkspaceFeatureFlagsGlobals' from JSON`);
}
/** @internal */
exports.GetWorkspaceFeatureFlagsRequest$inboundSchema = z.object({
    workspace_id: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "workspace_id": "workspaceId",
    });
});
/** @internal */
exports.GetWorkspaceFeatureFlagsRequest$outboundSchema = z.object({
    workspaceId: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        workspaceId: "workspace_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetWorkspaceFeatureFlagsRequest$;
(function (GetWorkspaceFeatureFlagsRequest$) {
    /** @deprecated use `GetWorkspaceFeatureFlagsRequest$inboundSchema` instead. */
    GetWorkspaceFeatureFlagsRequest$.inboundSchema = exports.GetWorkspaceFeatureFlagsRequest$inboundSchema;
    /** @deprecated use `GetWorkspaceFeatureFlagsRequest$outboundSchema` instead. */
    GetWorkspaceFeatureFlagsRequest$.outboundSchema = exports.GetWorkspaceFeatureFlagsRequest$outboundSchema;
})(GetWorkspaceFeatureFlagsRequest$ || (exports.GetWorkspaceFeatureFlagsRequest$ = GetWorkspaceFeatureFlagsRequest$ = {}));
function getWorkspaceFeatureFlagsRequestToJSON(getWorkspaceFeatureFlagsRequest) {
    return JSON.stringify(exports.GetWorkspaceFeatureFlagsRequest$outboundSchema.parse(getWorkspaceFeatureFlagsRequest));
}
function getWorkspaceFeatureFlagsRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetWorkspaceFeatureFlagsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetWorkspaceFeatureFlagsRequest' from JSON`);
}
