// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/getworkspaceteam.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetWorkspaceTeamRequest$ = exports.GetWorkspaceTeamRequest$outboundSchema = exports.GetWorkspaceTeamRequest$inboundSchema = exports.GetWorkspaceTeamGlobals$ = exports.GetWorkspaceTeamGlobals$outboundSchema = exports.GetWorkspaceTeamGlobals$inboundSchema = void 0;
exports.getWorkspaceTeamGlobalsToJSON = getWorkspaceTeamGlobalsToJSON;
exports.getWorkspaceTeamGlobalsFromJSON = getWorkspaceTeamGlobalsFromJSON;
exports.getWorkspaceTeamRequestToJSON = getWorkspaceTeamRequestToJSON;
exports.getWorkspaceTeamRequestFromJSON = getWorkspaceTeamRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.GetWorkspaceTeamGlobals$inboundSchema = z.object({
    workspace_id: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "workspace_id": "workspaceId",
    });
});
/** @internal */
exports.GetWorkspaceTeamGlobals$outboundSchema = z.object({
    workspaceId: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        workspaceId: "workspace_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetWorkspaceTeamGlobals$;
(function (GetWorkspaceTeamGlobals$) {
    /** @deprecated use `GetWorkspaceTeamGlobals$inboundSchema` instead. */
    GetWorkspaceTeamGlobals$.inboundSchema = exports.GetWorkspaceTeamGlobals$inboundSchema;
    /** @deprecated use `GetWorkspaceTeamGlobals$outboundSchema` instead. */
    GetWorkspaceTeamGlobals$.outboundSchema = exports.GetWorkspaceTeamGlobals$outboundSchema;
})(GetWorkspaceTeamGlobals$ || (exports.GetWorkspaceTeamGlobals$ = GetWorkspaceTeamGlobals$ = {}));
function getWorkspaceTeamGlobalsToJSON(getWorkspaceTeamGlobals) {
    return JSON.stringify(exports.GetWorkspaceTeamGlobals$outboundSchema.parse(getWorkspaceTeamGlobals));
}
function getWorkspaceTeamGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetWorkspaceTeamGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetWorkspaceTeamGlobals' from JSON`);
}
/** @internal */
exports.GetWorkspaceTeamRequest$inboundSchema = z.object({
    workspace_id: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "workspace_id": "workspaceId",
    });
});
/** @internal */
exports.GetWorkspaceTeamRequest$outboundSchema = z.object({
    workspaceId: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        workspaceId: "workspace_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetWorkspaceTeamRequest$;
(function (GetWorkspaceTeamRequest$) {
    /** @deprecated use `GetWorkspaceTeamRequest$inboundSchema` instead. */
    GetWorkspaceTeamRequest$.inboundSchema = exports.GetWorkspaceTeamRequest$inboundSchema;
    /** @deprecated use `GetWorkspaceTeamRequest$outboundSchema` instead. */
    GetWorkspaceTeamRequest$.outboundSchema = exports.GetWorkspaceTeamRequest$outboundSchema;
})(GetWorkspaceTeamRequest$ || (exports.GetWorkspaceTeamRequest$ = GetWorkspaceTeamRequest$ = {}));
function getWorkspaceTeamRequestToJSON(getWorkspaceTeamRequest) {
    return JSON.stringify(exports.GetWorkspaceTeamRequest$outboundSchema.parse(getWorkspaceTeamRequest));
}
function getWorkspaceTeamRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetWorkspaceTeamRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetWorkspaceTeamRequest' from JSON`);
}
