// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/getworkspace.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetWorkspaceRequest$ = exports.GetWorkspaceRequest$outboundSchema = exports.GetWorkspaceRequest$inboundSchema = exports.GetWorkspaceGlobals$ = exports.GetWorkspaceGlobals$outboundSchema = exports.GetWorkspaceGlobals$inboundSchema = void 0;
exports.getWorkspaceGlobalsToJSON = getWorkspaceGlobalsToJSON;
exports.getWorkspaceGlobalsFromJSON = getWorkspaceGlobalsFromJSON;
exports.getWorkspaceRequestToJSON = getWorkspaceRequestToJSON;
exports.getWorkspaceRequestFromJSON = getWorkspaceRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.GetWorkspaceGlobals$inboundSchema = z.object({
    workspace_id: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "workspace_id": "workspaceId",
    });
});
/** @internal */
exports.GetWorkspaceGlobals$outboundSchema = z.object({
    workspaceId: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        workspaceId: "workspace_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetWorkspaceGlobals$;
(function (GetWorkspaceGlobals$) {
    /** @deprecated use `GetWorkspaceGlobals$inboundSchema` instead. */
    GetWorkspaceGlobals$.inboundSchema = exports.GetWorkspaceGlobals$inboundSchema;
    /** @deprecated use `GetWorkspaceGlobals$outboundSchema` instead. */
    GetWorkspaceGlobals$.outboundSchema = exports.GetWorkspaceGlobals$outboundSchema;
})(GetWorkspaceGlobals$ || (exports.GetWorkspaceGlobals$ = GetWorkspaceGlobals$ = {}));
function getWorkspaceGlobalsToJSON(getWorkspaceGlobals) {
    return JSON.stringify(exports.GetWorkspaceGlobals$outboundSchema.parse(getWorkspaceGlobals));
}
function getWorkspaceGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetWorkspaceGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetWorkspaceGlobals' from JSON`);
}
/** @internal */
exports.GetWorkspaceRequest$inboundSchema = z.object({
    workspace_id: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "workspace_id": "workspaceId",
    });
});
/** @internal */
exports.GetWorkspaceRequest$outboundSchema = z.object({
    workspaceId: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        workspaceId: "workspace_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetWorkspaceRequest$;
(function (GetWorkspaceRequest$) {
    /** @deprecated use `GetWorkspaceRequest$inboundSchema` instead. */
    GetWorkspaceRequest$.inboundSchema = exports.GetWorkspaceRequest$inboundSchema;
    /** @deprecated use `GetWorkspaceRequest$outboundSchema` instead. */
    GetWorkspaceRequest$.outboundSchema = exports.GetWorkspaceRequest$outboundSchema;
})(GetWorkspaceRequest$ || (exports.GetWorkspaceRequest$ = GetWorkspaceRequest$ = {}));
function getWorkspaceRequestToJSON(getWorkspaceRequest) {
    return JSON.stringify(exports.GetWorkspaceRequest$outboundSchema.parse(getWorkspaceRequest));
}
function getWorkspaceRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetWorkspaceRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetWorkspaceRequest' from JSON`);
}
