// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/getworkspacetokens.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetWorkspaceTokensRequest$ = exports.GetWorkspaceTokensRequest$outboundSchema = exports.GetWorkspaceTokensRequest$inboundSchema = exports.GetWorkspaceTokensGlobals$ = exports.GetWorkspaceTokensGlobals$outboundSchema = exports.GetWorkspaceTokensGlobals$inboundSchema = void 0;
exports.getWorkspaceTokensGlobalsToJSON = getWorkspaceTokensGlobalsToJSON;
exports.getWorkspaceTokensGlobalsFromJSON = getWorkspaceTokensGlobalsFromJSON;
exports.getWorkspaceTokensRequestToJSON = getWorkspaceTokensRequestToJSON;
exports.getWorkspaceTokensRequestFromJSON = getWorkspaceTokensRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.GetWorkspaceTokensGlobals$inboundSchema = z.object({
    workspace_id: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "workspace_id": "workspaceId",
    });
});
/** @internal */
exports.GetWorkspaceTokensGlobals$outboundSchema = z.object({
    workspaceId: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        workspaceId: "workspace_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetWorkspaceTokensGlobals$;
(function (GetWorkspaceTokensGlobals$) {
    /** @deprecated use `GetWorkspaceTokensGlobals$inboundSchema` instead. */
    GetWorkspaceTokensGlobals$.inboundSchema = exports.GetWorkspaceTokensGlobals$inboundSchema;
    /** @deprecated use `GetWorkspaceTokensGlobals$outboundSchema` instead. */
    GetWorkspaceTokensGlobals$.outboundSchema = exports.GetWorkspaceTokensGlobals$outboundSchema;
})(GetWorkspaceTokensGlobals$ || (exports.GetWorkspaceTokensGlobals$ = GetWorkspaceTokensGlobals$ = {}));
function getWorkspaceTokensGlobalsToJSON(getWorkspaceTokensGlobals) {
    return JSON.stringify(exports.GetWorkspaceTokensGlobals$outboundSchema.parse(getWorkspaceTokensGlobals));
}
function getWorkspaceTokensGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetWorkspaceTokensGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetWorkspaceTokensGlobals' from JSON`);
}
/** @internal */
exports.GetWorkspaceTokensRequest$inboundSchema = z.object({
    workspace_id: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "workspace_id": "workspaceId",
    });
});
/** @internal */
exports.GetWorkspaceTokensRequest$outboundSchema = z.object({
    workspaceId: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        workspaceId: "workspace_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetWorkspaceTokensRequest$;
(function (GetWorkspaceTokensRequest$) {
    /** @deprecated use `GetWorkspaceTokensRequest$inboundSchema` instead. */
    GetWorkspaceTokensRequest$.inboundSchema = exports.GetWorkspaceTokensRequest$inboundSchema;
    /** @deprecated use `GetWorkspaceTokensRequest$outboundSchema` instead. */
    GetWorkspaceTokensRequest$.outboundSchema = exports.GetWorkspaceTokensRequest$outboundSchema;
})(GetWorkspaceTokensRequest$ || (exports.GetWorkspaceTokensRequest$ = GetWorkspaceTokensRequest$ = {}));
function getWorkspaceTokensRequestToJSON(getWorkspaceTokensRequest) {
    return JSON.stringify(exports.GetWorkspaceTokensRequest$outboundSchema.parse(getWorkspaceTokensRequest));
}
function getWorkspaceTokensRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetWorkspaceTokensRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetWorkspaceTokensRequest' from JSON`);
}
