// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/getworkspacetargets.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetWorkspaceTargetsRequest$ = exports.GetWorkspaceTargetsRequest$outboundSchema = exports.GetWorkspaceTargetsRequest$inboundSchema = void 0;
exports.getWorkspaceTargetsRequestToJSON = getWorkspaceTargetsRequestToJSON;
exports.getWorkspaceTargetsRequestFromJSON = getWorkspaceTargetsRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.GetWorkspaceTargetsRequest$inboundSchema = z.object({
    after_last_event_created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "after_last_event_created_at": "afterLastEventCreatedAt",
    });
});
/** @internal */
exports.GetWorkspaceTargetsRequest$outboundSchema = z.object({
    afterLastEventCreatedAt: z.date().transform(v => v.toISOString()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        afterLastEventCreatedAt: "after_last_event_created_at",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetWorkspaceTargetsRequest$;
(function (GetWorkspaceTargetsRequest$) {
    /** @deprecated use `GetWorkspaceTargetsRequest$inboundSchema` instead. */
    GetWorkspaceTargetsRequest$.inboundSchema = exports.GetWorkspaceTargetsRequest$inboundSchema;
    /** @deprecated use `GetWorkspaceTargetsRequest$outboundSchema` instead. */
    GetWorkspaceTargetsRequest$.outboundSchema = exports.GetWorkspaceTargetsRequest$outboundSchema;
})(GetWorkspaceTargetsRequest$ || (exports.GetWorkspaceTargetsRequest$ = GetWorkspaceTargetsRequest$ = {}));
function getWorkspaceTargetsRequestToJSON(getWorkspaceTargetsRequest) {
    return JSON.stringify(exports.GetWorkspaceTargetsRequest$outboundSchema.parse(getWorkspaceTargetsRequest));
}
function getWorkspaceTargetsRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetWorkspaceTargetsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetWorkspaceTargetsRequest' from JSON`);
}
