// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/getworkspaceaccess.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetWorkspaceAccessRequest$ = exports.GetWorkspaceAccessRequest$outboundSchema = exports.GetWorkspaceAccessRequest$inboundSchema = void 0;
exports.getWorkspaceAccessRequestToJSON = getWorkspaceAccessRequestToJSON;
exports.getWorkspaceAccessRequestFromJSON = getWorkspaceAccessRequestFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.GetWorkspaceAccessRequest$inboundSchema = z.object({
    genLockId: z.string().optional(),
    targetType: z.string().optional(),
    passive: z.boolean().optional(),
});
/** @internal */
exports.GetWorkspaceAccessRequest$outboundSchema = z.object({
    genLockId: z.string().optional(),
    targetType: z.string().optional(),
    passive: z.boolean().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetWorkspaceAccessRequest$;
(function (GetWorkspaceAccessRequest$) {
    /** @deprecated use `GetWorkspaceAccessRequest$inboundSchema` instead. */
    GetWorkspaceAccessRequest$.inboundSchema = exports.GetWorkspaceAccessRequest$inboundSchema;
    /** @deprecated use `GetWorkspaceAccessRequest$outboundSchema` instead. */
    GetWorkspaceAccessRequest$.outboundSchema = exports.GetWorkspaceAccessRequest$outboundSchema;
})(GetWorkspaceAccessRequest$ || (exports.GetWorkspaceAccessRequest$ = GetWorkspaceAccessRequest$ = {}));
function getWorkspaceAccessRequestToJSON(getWorkspaceAccessRequest) {
    return JSON.stringify(exports.GetWorkspaceAccessRequest$outboundSchema.parse(getWorkspaceAccessRequest));
}
function getWorkspaceAccessRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetWorkspaceAccessRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetWorkspaceAccessRequest' from JSON`);
}
