// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/githubcheckpublishingsecrets.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GithubCheckPublishingSecretsRequest$ = exports.GithubCheckPublishingSecretsRequest$outboundSchema = exports.GithubCheckPublishingSecretsRequest$inboundSchema = void 0;
exports.githubCheckPublishingSecretsRequestToJSON = githubCheckPublishingSecretsRequestToJSON;
exports.githubCheckPublishingSecretsRequestFromJSON = githubCheckPublishingSecretsRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.GithubCheckPublishingSecretsRequest$inboundSchema = z.object({
    generate_gen_lock_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "generate_gen_lock_id": "generateGenLockId",
    });
});
/** @internal */
exports.GithubCheckPublishingSecretsRequest$outboundSchema = z.object({
    generateGenLockId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        generateGenLockId: "generate_gen_lock_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GithubCheckPublishingSecretsRequest$;
(function (GithubCheckPublishingSecretsRequest$) {
    /** @deprecated use `GithubCheckPublishingSecretsRequest$inboundSchema` instead. */
    GithubCheckPublishingSecretsRequest$.inboundSchema = exports.GithubCheckPublishingSecretsRequest$inboundSchema;
    /** @deprecated use `GithubCheckPublishingSecretsRequest$outboundSchema` instead. */
    GithubCheckPublishingSecretsRequest$.outboundSchema = exports.GithubCheckPublishingSecretsRequest$outboundSchema;
})(GithubCheckPublishingSecretsRequest$ || (exports.GithubCheckPublishingSecretsRequest$ = GithubCheckPublishingSecretsRequest$ = {}));
function githubCheckPublishingSecretsRequestToJSON(githubCheckPublishingSecretsRequest) {
    return JSON.stringify(exports.GithubCheckPublishingSecretsRequest$outboundSchema.parse(githubCheckPublishingSecretsRequest));
}
function githubCheckPublishingSecretsRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GithubCheckPublishingSecretsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GithubCheckPublishingSecretsRequest' from JSON`);
}
