// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/grantuseraccesstoworkspace.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GrantUserAccessToWorkspaceRequest$ = exports.GrantUserAccessToWorkspaceRequest$outboundSchema = exports.GrantUserAccessToWorkspaceRequest$inboundSchema = exports.GrantUserAccessToWorkspaceGlobals$ = exports.GrantUserAccessToWorkspaceGlobals$outboundSchema = exports.GrantUserAccessToWorkspaceGlobals$inboundSchema = void 0;
exports.grantUserAccessToWorkspaceGlobalsToJSON = grantUserAccessToWorkspaceGlobalsToJSON;
exports.grantUserAccessToWorkspaceGlobalsFromJSON = grantUserAccessToWorkspaceGlobalsFromJSON;
exports.grantUserAccessToWorkspaceRequestToJSON = grantUserAccessToWorkspaceRequestToJSON;
exports.grantUserAccessToWorkspaceRequestFromJSON = grantUserAccessToWorkspaceRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.GrantUserAccessToWorkspaceGlobals$inboundSchema = z.object({
    workspace_id: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "workspace_id": "workspaceId",
    });
});
/** @internal */
exports.GrantUserAccessToWorkspaceGlobals$outboundSchema = z.object({
    workspaceId: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        workspaceId: "workspace_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GrantUserAccessToWorkspaceGlobals$;
(function (GrantUserAccessToWorkspaceGlobals$) {
    /** @deprecated use `GrantUserAccessToWorkspaceGlobals$inboundSchema` instead. */
    GrantUserAccessToWorkspaceGlobals$.inboundSchema = exports.GrantUserAccessToWorkspaceGlobals$inboundSchema;
    /** @deprecated use `GrantUserAccessToWorkspaceGlobals$outboundSchema` instead. */
    GrantUserAccessToWorkspaceGlobals$.outboundSchema = exports.GrantUserAccessToWorkspaceGlobals$outboundSchema;
})(GrantUserAccessToWorkspaceGlobals$ || (exports.GrantUserAccessToWorkspaceGlobals$ = GrantUserAccessToWorkspaceGlobals$ = {}));
function grantUserAccessToWorkspaceGlobalsToJSON(grantUserAccessToWorkspaceGlobals) {
    return JSON.stringify(exports.GrantUserAccessToWorkspaceGlobals$outboundSchema.parse(grantUserAccessToWorkspaceGlobals));
}
function grantUserAccessToWorkspaceGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GrantUserAccessToWorkspaceGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GrantUserAccessToWorkspaceGlobals' from JSON`);
}
/** @internal */
exports.GrantUserAccessToWorkspaceRequest$inboundSchema = z.object({
    workspace_id: z.string().optional(),
    email: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "workspace_id": "workspaceId",
    });
});
/** @internal */
exports.GrantUserAccessToWorkspaceRequest$outboundSchema = z.object({
    workspaceId: z.string().optional(),
    email: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        workspaceId: "workspace_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GrantUserAccessToWorkspaceRequest$;
(function (GrantUserAccessToWorkspaceRequest$) {
    /** @deprecated use `GrantUserAccessToWorkspaceRequest$inboundSchema` instead. */
    GrantUserAccessToWorkspaceRequest$.inboundSchema = exports.GrantUserAccessToWorkspaceRequest$inboundSchema;
    /** @deprecated use `GrantUserAccessToWorkspaceRequest$outboundSchema` instead. */
    GrantUserAccessToWorkspaceRequest$.outboundSchema = exports.GrantUserAccessToWorkspaceRequest$outboundSchema;
})(GrantUserAccessToWorkspaceRequest$ || (exports.GrantUserAccessToWorkspaceRequest$ = GrantUserAccessToWorkspaceRequest$ = {}));
function grantUserAccessToWorkspaceRequestToJSON(grantUserAccessToWorkspaceRequest) {
    return JSON.stringify(exports.GrantUserAccessToWorkspaceRequest$outboundSchema.parse(grantUserAccessToWorkspaceRequest));
}
function grantUserAccessToWorkspaceRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GrantUserAccessToWorkspaceRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GrantUserAccessToWorkspaceRequest' from JSON`);
}
