// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/linkgithubaccess.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LinkGithubAccessRequest$ = exports.LinkGithubAccessRequest$outboundSchema = exports.LinkGithubAccessRequest$inboundSchema = void 0;
exports.linkGithubAccessRequestToJSON = linkGithubAccessRequestToJSON;
exports.linkGithubAccessRequestFromJSON = linkGithubAccessRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.LinkGithubAccessRequest$inboundSchema = z.object({
    installation_id: z.string().optional(),
    github_org: z.string().optional(),
    github_oidc_token: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "installation_id": "installationId",
        "github_org": "githubOrg",
        "github_oidc_token": "githubOidcToken",
    });
});
/** @internal */
exports.LinkGithubAccessRequest$outboundSchema = z.object({
    installationId: z.string().optional(),
    githubOrg: z.string().optional(),
    githubOidcToken: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        installationId: "installation_id",
        githubOrg: "github_org",
        githubOidcToken: "github_oidc_token",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LinkGithubAccessRequest$;
(function (LinkGithubAccessRequest$) {
    /** @deprecated use `LinkGithubAccessRequest$inboundSchema` instead. */
    LinkGithubAccessRequest$.inboundSchema = exports.LinkGithubAccessRequest$inboundSchema;
    /** @deprecated use `LinkGithubAccessRequest$outboundSchema` instead. */
    LinkGithubAccessRequest$.outboundSchema = exports.LinkGithubAccessRequest$outboundSchema;
})(LinkGithubAccessRequest$ || (exports.LinkGithubAccessRequest$ = LinkGithubAccessRequest$ = {}));
function linkGithubAccessRequestToJSON(linkGithubAccessRequest) {
    return JSON.stringify(exports.LinkGithubAccessRequest$outboundSchema.parse(linkGithubAccessRequest));
}
function linkGithubAccessRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.LinkGithubAccessRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'LinkGithubAccessRequest' from JSON`);
}
