// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/posttags.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostTagsRequest$ = exports.PostTagsRequest$outboundSchema = exports.PostTagsRequest$inboundSchema = void 0;
exports.postTagsRequestToJSON = postTagsRequestToJSON;
exports.postTagsRequestFromJSON = postTagsRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
const shared = __importStar(require("../shared/index.js"));
/** @internal */
exports.PostTagsRequest$inboundSchema = z.object({
    namespace_name: z.string(),
    AddTags: shared.AddTags$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "namespace_name": "namespaceName",
        "AddTags": "addTags",
    });
});
/** @internal */
exports.PostTagsRequest$outboundSchema = z.object({
    namespaceName: z.string(),
    addTags: shared.AddTags$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        namespaceName: "namespace_name",
        addTags: "AddTags",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PostTagsRequest$;
(function (PostTagsRequest$) {
    /** @deprecated use `PostTagsRequest$inboundSchema` instead. */
    PostTagsRequest$.inboundSchema = exports.PostTagsRequest$inboundSchema;
    /** @deprecated use `PostTagsRequest$outboundSchema` instead. */
    PostTagsRequest$.outboundSchema = exports.PostTagsRequest$outboundSchema;
})(PostTagsRequest$ || (exports.PostTagsRequest$ = PostTagsRequest$ = {}));
function postTagsRequestToJSON(postTagsRequest) {
    return JSON.stringify(exports.PostTagsRequest$outboundSchema.parse(postTagsRequest));
}
function postTagsRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.PostTagsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PostTagsRequest' from JSON`);
}
