// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/revokeuseraccesstoworkspace.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RevokeUserAccessToWorkspaceRequest$ = exports.RevokeUserAccessToWorkspaceRequest$outboundSchema = exports.RevokeUserAccessToWorkspaceRequest$inboundSchema = exports.RevokeUserAccessToWorkspaceGlobals$ = exports.RevokeUserAccessToWorkspaceGlobals$outboundSchema = exports.RevokeUserAccessToWorkspaceGlobals$inboundSchema = void 0;
exports.revokeUserAccessToWorkspaceGlobalsToJSON = revokeUserAccessToWorkspaceGlobalsToJSON;
exports.revokeUserAccessToWorkspaceGlobalsFromJSON = revokeUserAccessToWorkspaceGlobalsFromJSON;
exports.revokeUserAccessToWorkspaceRequestToJSON = revokeUserAccessToWorkspaceRequestToJSON;
exports.revokeUserAccessToWorkspaceRequestFromJSON = revokeUserAccessToWorkspaceRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.RevokeUserAccessToWorkspaceGlobals$inboundSchema = z.object({
    workspace_id: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "workspace_id": "workspaceId",
    });
});
/** @internal */
exports.RevokeUserAccessToWorkspaceGlobals$outboundSchema = z.object({
    workspaceId: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        workspaceId: "workspace_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RevokeUserAccessToWorkspaceGlobals$;
(function (RevokeUserAccessToWorkspaceGlobals$) {
    /** @deprecated use `RevokeUserAccessToWorkspaceGlobals$inboundSchema` instead. */
    RevokeUserAccessToWorkspaceGlobals$.inboundSchema = exports.RevokeUserAccessToWorkspaceGlobals$inboundSchema;
    /** @deprecated use `RevokeUserAccessToWorkspaceGlobals$outboundSchema` instead. */
    RevokeUserAccessToWorkspaceGlobals$.outboundSchema = exports.RevokeUserAccessToWorkspaceGlobals$outboundSchema;
})(RevokeUserAccessToWorkspaceGlobals$ || (exports.RevokeUserAccessToWorkspaceGlobals$ = RevokeUserAccessToWorkspaceGlobals$ = {}));
function revokeUserAccessToWorkspaceGlobalsToJSON(revokeUserAccessToWorkspaceGlobals) {
    return JSON.stringify(exports.RevokeUserAccessToWorkspaceGlobals$outboundSchema.parse(revokeUserAccessToWorkspaceGlobals));
}
function revokeUserAccessToWorkspaceGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.RevokeUserAccessToWorkspaceGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RevokeUserAccessToWorkspaceGlobals' from JSON`);
}
/** @internal */
exports.RevokeUserAccessToWorkspaceRequest$inboundSchema = z.object({
    workspace_id: z.string().optional(),
    userId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "workspace_id": "workspaceId",
    });
});
/** @internal */
exports.RevokeUserAccessToWorkspaceRequest$outboundSchema = z.object({
    workspaceId: z.string().optional(),
    userId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        workspaceId: "workspace_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RevokeUserAccessToWorkspaceRequest$;
(function (RevokeUserAccessToWorkspaceRequest$) {
    /** @deprecated use `RevokeUserAccessToWorkspaceRequest$inboundSchema` instead. */
    RevokeUserAccessToWorkspaceRequest$.inboundSchema = exports.RevokeUserAccessToWorkspaceRequest$inboundSchema;
    /** @deprecated use `RevokeUserAccessToWorkspaceRequest$outboundSchema` instead. */
    RevokeUserAccessToWorkspaceRequest$.outboundSchema = exports.RevokeUserAccessToWorkspaceRequest$outboundSchema;
})(RevokeUserAccessToWorkspaceRequest$ || (exports.RevokeUserAccessToWorkspaceRequest$ = RevokeUserAccessToWorkspaceRequest$ = {}));
function revokeUserAccessToWorkspaceRequestToJSON(revokeUserAccessToWorkspaceRequest) {
    return JSON.stringify(exports.RevokeUserAccessToWorkspaceRequest$outboundSchema.parse(revokeUserAccessToWorkspaceRequest));
}
function revokeUserAccessToWorkspaceRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.RevokeUserAccessToWorkspaceRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RevokeUserAccessToWorkspaceRequest' from JSON`);
}
