// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/suggestopenapiregistry.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SuggestOpenAPIRegistryRequest$ = exports.SuggestOpenAPIRegistryRequest$outboundSchema = exports.SuggestOpenAPIRegistryRequest$inboundSchema = void 0;
exports.suggestOpenAPIRegistryRequestToJSON = suggestOpenAPIRegistryRequestToJSON;
exports.suggestOpenAPIRegistryRequestFromJSON = suggestOpenAPIRegistryRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
const shared = __importStar(require("../shared/index.js"));
/** @internal */
exports.SuggestOpenAPIRegistryRequest$inboundSchema = z.object({
    "x-session-id": z.string(),
    namespace_name: z.string(),
    revision_reference: z.string(),
    SuggestRequestBody: shared.SuggestRequestBody$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "x-session-id": "xSessionId",
        "namespace_name": "namespaceName",
        "revision_reference": "revisionReference",
        "SuggestRequestBody": "suggestRequestBody",
    });
});
/** @internal */
exports.SuggestOpenAPIRegistryRequest$outboundSchema = z.object({
    xSessionId: z.string(),
    namespaceName: z.string(),
    revisionReference: z.string(),
    suggestRequestBody: shared.SuggestRequestBody$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        xSessionId: "x-session-id",
        namespaceName: "namespace_name",
        revisionReference: "revision_reference",
        suggestRequestBody: "SuggestRequestBody",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SuggestOpenAPIRegistryRequest$;
(function (SuggestOpenAPIRegistryRequest$) {
    /** @deprecated use `SuggestOpenAPIRegistryRequest$inboundSchema` instead. */
    SuggestOpenAPIRegistryRequest$.inboundSchema = exports.SuggestOpenAPIRegistryRequest$inboundSchema;
    /** @deprecated use `SuggestOpenAPIRegistryRequest$outboundSchema` instead. */
    SuggestOpenAPIRegistryRequest$.outboundSchema = exports.SuggestOpenAPIRegistryRequest$outboundSchema;
})(SuggestOpenAPIRegistryRequest$ || (exports.SuggestOpenAPIRegistryRequest$ = SuggestOpenAPIRegistryRequest$ = {}));
function suggestOpenAPIRegistryRequestToJSON(suggestOpenAPIRegistryRequest) {
    return JSON.stringify(exports.SuggestOpenAPIRegistryRequest$outboundSchema.parse(suggestOpenAPIRegistryRequest));
}
function suggestOpenAPIRegistryRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SuggestOpenAPIRegistryRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SuggestOpenAPIRegistryRequest' from JSON`);
}
