// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/models/operations/uploadreport.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UploadReportUploadedReport$ = exports.UploadReportUploadedReport$outboundSchema = exports.UploadReportUploadedReport$inboundSchema = exports.UploadReportRequestBody$ = exports.UploadReportRequestBody$outboundSchema = exports.UploadReportRequestBody$inboundSchema = exports.FileT$ = exports.FileT$outboundSchema = exports.FileT$inboundSchema = void 0;
exports.fileToJSON = fileToJSON;
exports.fileFromJSON = fileFromJSON;
exports.uploadReportRequestBodyToJSON = uploadReportRequestBodyToJSON;
exports.uploadReportRequestBodyFromJSON = uploadReportRequestBodyFromJSON;
exports.uploadReportUploadedReportToJSON = uploadReportUploadedReportToJSON;
exports.uploadReportUploadedReportFromJSON = uploadReportUploadedReportFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../../lib/schemas.js");
const blobs_js_1 = require("../../types/blobs.js");
const shared = __importStar(require("../shared/index.js"));
/** @internal */
exports.FileT$inboundSchema = z
    .object({
    content: z.union([
        z.instanceof((ReadableStream)),
        z.instanceof(Blob),
        z.instanceof(ArrayBuffer),
        z.instanceof(Uint8Array),
    ]),
    fileName: z.string(),
});
/** @internal */
exports.FileT$outboundSchema = z.object({
    content: z.union([
        z.instanceof((ReadableStream)),
        z.instanceof(Blob),
        z.instanceof(ArrayBuffer),
        z.instanceof(Uint8Array),
    ]),
    fileName: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FileT$;
(function (FileT$) {
    /** @deprecated use `FileT$inboundSchema` instead. */
    FileT$.inboundSchema = exports.FileT$inboundSchema;
    /** @deprecated use `FileT$outboundSchema` instead. */
    FileT$.outboundSchema = exports.FileT$outboundSchema;
})(FileT$ || (exports.FileT$ = FileT$ = {}));
function fileToJSON(fileT) {
    return JSON.stringify(exports.FileT$outboundSchema.parse(fileT));
}
function fileFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.FileT$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FileT' from JSON`);
}
/** @internal */
exports.UploadReportRequestBody$inboundSchema = z.object({
    data: shared.Report$inboundSchema,
    file: z.lazy(() => exports.FileT$inboundSchema),
});
/** @internal */
exports.UploadReportRequestBody$outboundSchema = z.object({
    data: shared.Report$outboundSchema,
    file: z.lazy(() => exports.FileT$outboundSchema).or(blobs_js_1.blobLikeSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UploadReportRequestBody$;
(function (UploadReportRequestBody$) {
    /** @deprecated use `UploadReportRequestBody$inboundSchema` instead. */
    UploadReportRequestBody$.inboundSchema = exports.UploadReportRequestBody$inboundSchema;
    /** @deprecated use `UploadReportRequestBody$outboundSchema` instead. */
    UploadReportRequestBody$.outboundSchema = exports.UploadReportRequestBody$outboundSchema;
})(UploadReportRequestBody$ || (exports.UploadReportRequestBody$ = UploadReportRequestBody$ = {}));
function uploadReportRequestBodyToJSON(uploadReportRequestBody) {
    return JSON.stringify(exports.UploadReportRequestBody$outboundSchema.parse(uploadReportRequestBody));
}
function uploadReportRequestBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.UploadReportRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UploadReportRequestBody' from JSON`);
}
/** @internal */
exports.UploadReportUploadedReport$inboundSchema = z.object({
    url: z.string(),
});
/** @internal */
exports.UploadReportUploadedReport$outboundSchema = z.object({
    url: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UploadReportUploadedReport$;
(function (UploadReportUploadedReport$) {
    /** @deprecated use `UploadReportUploadedReport$inboundSchema` instead. */
    UploadReportUploadedReport$.inboundSchema = exports.UploadReportUploadedReport$inboundSchema;
    /** @deprecated use `UploadReportUploadedReport$outboundSchema` instead. */
    UploadReportUploadedReport$.outboundSchema = exports.UploadReportUploadedReport$outboundSchema;
})(UploadReportUploadedReport$ || (exports.UploadReportUploadedReport$ = UploadReportUploadedReport$ = {}));
function uploadReportUploadedReportToJSON(uploadReportUploadedReport) {
    return JSON.stringify(exports.UploadReportUploadedReport$outboundSchema.parse(uploadReportUploadedReport));
}
function uploadReportUploadedReportFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.UploadReportUploadedReport$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UploadReportUploadedReport' from JSON`);
}
