// @speakeasy-api/speakeasy-client-sdk-typescript/lib/config.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SDK_METADATA = exports.ServerList = exports.ServerProd = void 0;
exports.serverURLFromOptions = serverURLFromOptions;
const url_js_1 = require("./url.js");
exports.ServerProd = "prod";
/**
 * Contains the list of servers available to the SDK
 */
exports.ServerList = {
    [exports.ServerProd]: "https://api.prod.speakeasy.com",
};
function serverURLFromOptions(options) {
    let serverURL = options.serverURL;
    const params = {};
    if (!serverURL) {
        const server = options.server ?? exports.ServerProd;
        serverURL = exports.ServerList[server] || "";
    }
    const u = (0, url_js_1.pathToFunc)(serverURL)(params);
    return new URL(u);
}
exports.SDK_METADATA = {
    language: "typescript",
    openapiDocVersion: "0.4.0",
    sdkVersion: "5.0.0-rc.4",
    genVersion: "2.628.0",
    userAgent: "speakeasy-sdk/typescript 5.0.0-rc.4 2.628.0 0.4.0 @speakeasy-api/speakeasy-client-sdk-typescript",
};
