// @speakeasy-api/speakeasy-client-sdk-typescript/mcp-server/server.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.createMCPServer = createMCPServer;
const mcp_js_1 = require("@modelcontextprotocol/sdk/server/mcp.js");
const core_js_1 = require("../core.js");
const prompts_js_1 = require("./prompts.js");
const resources_js_1 = require("./resources.js");
const tools_js_1 = require("./tools.js");
const artifactsCreateRemoteSource_js_1 = require("./tools/artifactsCreateRemoteSource.js");
const artifactsGetBlob_js_1 = require("./tools/artifactsGetBlob.js");
const artifactsGetManifest_js_1 = require("./tools/artifactsGetManifest.js");
const artifactsGetNamespaces_js_1 = require("./tools/artifactsGetNamespaces.js");
const artifactsGetRevisions_js_1 = require("./tools/artifactsGetRevisions.js");
const artifactsGetTags_js_1 = require("./tools/artifactsGetTags.js");
const artifactsListRemoteSources_js_1 = require("./tools/artifactsListRemoteSources.js");
const artifactsPostTags_js_1 = require("./tools/artifactsPostTags.js");
const artifactsPreflight_js_1 = require("./tools/artifactsPreflight.js");
const artifactsSetArchived_js_1 = require("./tools/artifactsSetArchived.js");
const artifactsSetVisibility_js_1 = require("./tools/artifactsSetVisibility.js");
const authGetAccess_js_1 = require("./tools/authGetAccess.js");
const authGetAccessToken_js_1 = require("./tools/authGetAccessToken.js");
const authGetUser_js_1 = require("./tools/authGetUser.js");
const authValidateApiKey_js_1 = require("./tools/authValidateApiKey.js");
const codeSamplesGenerateCodeSamplePreview_js_1 = require("./tools/codeSamplesGenerateCodeSamplePreview.js");
const codeSamplesGenerateCodeSamplePreviewAsync_js_1 = require("./tools/codeSamplesGenerateCodeSamplePreviewAsync.js");
const codeSamplesGet_js_1 = require("./tools/codeSamplesGet.js");
const codeSamplesGetCodeSamplePreviewAsync_js_1 = require("./tools/codeSamplesGetCodeSamplePreviewAsync.js");
const eventsGetEventsByTarget_js_1 = require("./tools/eventsGetEventsByTarget.js");
const eventsGetTargets_js_1 = require("./tools/eventsGetTargets.js");
const eventsGetTargetsDeprecated_js_1 = require("./tools/eventsGetTargetsDeprecated.js");
const eventsPost_js_1 = require("./tools/eventsPost.js");
const eventsSearch_js_1 = require("./tools/eventsSearch.js");
const githubCheckAccess_js_1 = require("./tools/githubCheckAccess.js");
const githubCheckPublishingPRs_js_1 = require("./tools/githubCheckPublishingPRs.js");
const githubCheckPublishingSecrets_js_1 = require("./tools/githubCheckPublishingSecrets.js");
const githubConfigureCodeSamples_js_1 = require("./tools/githubConfigureCodeSamples.js");
const githubConfigureMintlifyRepo_js_1 = require("./tools/githubConfigureMintlifyRepo.js");
const githubConfigureTarget_js_1 = require("./tools/githubConfigureTarget.js");
const githubGetAction_js_1 = require("./tools/githubGetAction.js");
const githubGetSetup_js_1 = require("./tools/githubGetSetup.js");
const githubLinkGithub_js_1 = require("./tools/githubLinkGithub.js");
const githubStorePublishingSecrets_js_1 = require("./tools/githubStorePublishingSecrets.js");
const githubTriggerAction_js_1 = require("./tools/githubTriggerAction.js");
const organizationsCreate_js_1 = require("./tools/organizationsCreate.js");
const organizationsCreateBillingAddOns_js_1 = require("./tools/organizationsCreateBillingAddOns.js");
const organizationsCreateFreeTrial_js_1 = require("./tools/organizationsCreateFreeTrial.js");
const organizationsDeleteBillingAddOn_js_1 = require("./tools/organizationsDeleteBillingAddOn.js");
const organizationsGet_js_1 = require("./tools/organizationsGet.js");
const organizationsGetAll_js_1 = require("./tools/organizationsGetAll.js");
const organizationsGetBillingAddOns_js_1 = require("./tools/organizationsGetBillingAddOns.js");
const organizationsGetUsage_js_1 = require("./tools/organizationsGetUsage.js");
const publishingTokensCreate_js_1 = require("./tools/publishingTokensCreate.js");
const publishingTokensDelete_js_1 = require("./tools/publishingTokensDelete.js");
const publishingTokensGet_js_1 = require("./tools/publishingTokensGet.js");
const publishingTokensList_js_1 = require("./tools/publishingTokensList.js");
const publishingTokensResolveMetadata_js_1 = require("./tools/publishingTokensResolveMetadata.js");
const publishingTokensResolveTarget_js_1 = require("./tools/publishingTokensResolveTarget.js");
const publishingTokensUpdate_js_1 = require("./tools/publishingTokensUpdate.js");
const reportsGetChangesReportSignedUrl_js_1 = require("./tools/reportsGetChangesReportSignedUrl.js");
const reportsGetLintingReportSignedUrl_js_1 = require("./tools/reportsGetLintingReportSignedUrl.js");
const reportsUploadReport_js_1 = require("./tools/reportsUploadReport.js");
const schemaStoreCreateSchemaStoreItem_js_1 = require("./tools/schemaStoreCreateSchemaStoreItem.js");
const shortURLsCreate_js_1 = require("./tools/shortURLsCreate.js");
const subscriptionsActivateSubscriptionNamespace_js_1 = require("./tools/subscriptionsActivateSubscriptionNamespace.js");
const subscriptionsIgnoreSubscriptionNamespace_js_1 = require("./tools/subscriptionsIgnoreSubscriptionNamespace.js");
const suggestSuggest_js_1 = require("./tools/suggestSuggest.js");
const suggestSuggestItems_js_1 = require("./tools/suggestSuggestItems.js");
const suggestSuggestOpenAPI_js_1 = require("./tools/suggestSuggestOpenAPI.js");
const suggestSuggestOpenAPIRegistry_js_1 = require("./tools/suggestSuggestOpenAPIRegistry.js");
const workspacesCreate_js_1 = require("./tools/workspacesCreate.js");
const workspacesCreateToken_js_1 = require("./tools/workspacesCreateToken.js");
const workspacesDeleteToken_js_1 = require("./tools/workspacesDeleteToken.js");
const workspacesGet_js_1 = require("./tools/workspacesGet.js");
const workspacesGetAll_js_1 = require("./tools/workspacesGetAll.js");
const workspacesGetByID_js_1 = require("./tools/workspacesGetByID.js");
const workspacesGetFeatureFlags_js_1 = require("./tools/workspacesGetFeatureFlags.js");
const workspacesGetSettings_js_1 = require("./tools/workspacesGetSettings.js");
const workspacesGetTeam_js_1 = require("./tools/workspacesGetTeam.js");
const workspacesGetTokens_js_1 = require("./tools/workspacesGetTokens.js");
const workspacesGrantAccess_js_1 = require("./tools/workspacesGrantAccess.js");
const workspacesRevokeAccess_js_1 = require("./tools/workspacesRevokeAccess.js");
const workspacesSetFeatureFlags_js_1 = require("./tools/workspacesSetFeatureFlags.js");
const workspacesUpdate_js_1 = require("./tools/workspacesUpdate.js");
const workspacesUpdateSettings_js_1 = require("./tools/workspacesUpdateSettings.js");
function createMCPServer(deps) {
    const server = new mcp_js_1.McpServer({
        name: "Speakeasy",
        version: "5.0.0-rc.4",
    });
    const client = new core_js_1.SpeakeasyCore({
        security: deps.security,
        workspaceId: deps.workspaceId,
        serverURL: deps.serverURL,
        server: deps.server,
    });
    const scopes = new Set(deps.scopes);
    const allowedTools = deps.allowedTools && new Set(deps.allowedTools);
    const tool = (0, tools_js_1.createRegisterTool)(deps.logger, server, client, scopes, allowedTools);
    const resource = (0, resources_js_1.createRegisterResource)(deps.logger, server, client, scopes);
    const resourceTemplate = (0, resources_js_1.createRegisterResourceTemplate)(deps.logger, server, client, scopes);
    const prompt = (0, prompts_js_1.createRegisterPrompt)(deps.logger, server, client, scopes);
    const register = { tool, resource, resourceTemplate, prompt };
    void register; // suppress unused warnings
    tool(artifactsCreateRemoteSource_js_1.tool$artifactsCreateRemoteSource);
    tool(artifactsGetBlob_js_1.tool$artifactsGetBlob);
    tool(artifactsGetManifest_js_1.tool$artifactsGetManifest);
    tool(artifactsGetNamespaces_js_1.tool$artifactsGetNamespaces);
    tool(artifactsGetRevisions_js_1.tool$artifactsGetRevisions);
    tool(artifactsGetTags_js_1.tool$artifactsGetTags);
    tool(artifactsListRemoteSources_js_1.tool$artifactsListRemoteSources);
    tool(artifactsPostTags_js_1.tool$artifactsPostTags);
    tool(artifactsPreflight_js_1.tool$artifactsPreflight);
    tool(artifactsSetArchived_js_1.tool$artifactsSetArchived);
    tool(artifactsSetVisibility_js_1.tool$artifactsSetVisibility);
    tool(authGetAccess_js_1.tool$authGetAccess);
    tool(authGetAccessToken_js_1.tool$authGetAccessToken);
    tool(authGetUser_js_1.tool$authGetUser);
    tool(authValidateApiKey_js_1.tool$authValidateApiKey);
    tool(codeSamplesGenerateCodeSamplePreview_js_1.tool$codeSamplesGenerateCodeSamplePreview);
    tool(codeSamplesGenerateCodeSamplePreviewAsync_js_1.tool$codeSamplesGenerateCodeSamplePreviewAsync);
    tool(codeSamplesGet_js_1.tool$codeSamplesGet);
    tool(codeSamplesGetCodeSamplePreviewAsync_js_1.tool$codeSamplesGetCodeSamplePreviewAsync);
    tool(githubCheckAccess_js_1.tool$githubCheckAccess);
    tool(githubCheckPublishingPRs_js_1.tool$githubCheckPublishingPRs);
    tool(githubCheckPublishingSecrets_js_1.tool$githubCheckPublishingSecrets);
    tool(githubConfigureCodeSamples_js_1.tool$githubConfigureCodeSamples);
    tool(githubConfigureMintlifyRepo_js_1.tool$githubConfigureMintlifyRepo);
    tool(githubConfigureTarget_js_1.tool$githubConfigureTarget);
    tool(githubGetAction_js_1.tool$githubGetAction);
    tool(githubGetSetup_js_1.tool$githubGetSetup);
    tool(githubLinkGithub_js_1.tool$githubLinkGithub);
    tool(githubStorePublishingSecrets_js_1.tool$githubStorePublishingSecrets);
    tool(githubTriggerAction_js_1.tool$githubTriggerAction);
    tool(organizationsCreate_js_1.tool$organizationsCreate);
    tool(organizationsCreateBillingAddOns_js_1.tool$organizationsCreateBillingAddOns);
    tool(organizationsCreateFreeTrial_js_1.tool$organizationsCreateFreeTrial);
    tool(organizationsDeleteBillingAddOn_js_1.tool$organizationsDeleteBillingAddOn);
    tool(organizationsGet_js_1.tool$organizationsGet);
    tool(organizationsGetAll_js_1.tool$organizationsGetAll);
    tool(organizationsGetBillingAddOns_js_1.tool$organizationsGetBillingAddOns);
    tool(organizationsGetUsage_js_1.tool$organizationsGetUsage);
    tool(publishingTokensCreate_js_1.tool$publishingTokensCreate);
    tool(publishingTokensDelete_js_1.tool$publishingTokensDelete);
    tool(publishingTokensGet_js_1.tool$publishingTokensGet);
    tool(publishingTokensList_js_1.tool$publishingTokensList);
    tool(publishingTokensResolveMetadata_js_1.tool$publishingTokensResolveMetadata);
    tool(publishingTokensResolveTarget_js_1.tool$publishingTokensResolveTarget);
    tool(publishingTokensUpdate_js_1.tool$publishingTokensUpdate);
    tool(reportsGetChangesReportSignedUrl_js_1.tool$reportsGetChangesReportSignedUrl);
    tool(reportsGetLintingReportSignedUrl_js_1.tool$reportsGetLintingReportSignedUrl);
    tool(reportsUploadReport_js_1.tool$reportsUploadReport);
    tool(schemaStoreCreateSchemaStoreItem_js_1.tool$schemaStoreCreateSchemaStoreItem);
    tool(shortURLsCreate_js_1.tool$shortURLsCreate);
    tool(subscriptionsActivateSubscriptionNamespace_js_1.tool$subscriptionsActivateSubscriptionNamespace);
    tool(subscriptionsIgnoreSubscriptionNamespace_js_1.tool$subscriptionsIgnoreSubscriptionNamespace);
    tool(suggestSuggest_js_1.tool$suggestSuggest);
    tool(suggestSuggestItems_js_1.tool$suggestSuggestItems);
    tool(suggestSuggestOpenAPI_js_1.tool$suggestSuggestOpenAPI);
    tool(suggestSuggestOpenAPIRegistry_js_1.tool$suggestSuggestOpenAPIRegistry);
    tool(workspacesCreate_js_1.tool$workspacesCreate);
    tool(workspacesCreateToken_js_1.tool$workspacesCreateToken);
    tool(workspacesDeleteToken_js_1.tool$workspacesDeleteToken);
    tool(workspacesGet_js_1.tool$workspacesGet);
    tool(workspacesGetAll_js_1.tool$workspacesGetAll);
    tool(workspacesGetByID_js_1.tool$workspacesGetByID);
    tool(workspacesGetFeatureFlags_js_1.tool$workspacesGetFeatureFlags);
    tool(workspacesGetSettings_js_1.tool$workspacesGetSettings);
    tool(workspacesGetTeam_js_1.tool$workspacesGetTeam);
    tool(workspacesGetTokens_js_1.tool$workspacesGetTokens);
    tool(workspacesGrantAccess_js_1.tool$workspacesGrantAccess);
    tool(workspacesRevokeAccess_js_1.tool$workspacesRevokeAccess);
    tool(workspacesSetFeatureFlags_js_1.tool$workspacesSetFeatureFlags);
    tool(workspacesUpdate_js_1.tool$workspacesUpdate);
    tool(workspacesUpdateSettings_js_1.tool$workspacesUpdateSettings);
    tool(eventsGetEventsByTarget_js_1.tool$eventsGetEventsByTarget);
    tool(eventsGetTargets_js_1.tool$eventsGetTargets);
    tool(eventsGetTargetsDeprecated_js_1.tool$eventsGetTargetsDeprecated);
    tool(eventsPost_js_1.tool$eventsPost);
    tool(eventsSearch_js_1.tool$eventsSearch);
    return server;
}
