// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/events.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Events = void 0;
const eventsGetEventsByTarget_js_1 = require("../funcs/eventsGetEventsByTarget.js");
const eventsGetTargets_js_1 = require("../funcs/eventsGetTargets.js");
const eventsGetTargetsDeprecated_js_1 = require("../funcs/eventsGetTargetsDeprecated.js");
const eventsPost_js_1 = require("../funcs/eventsPost.js");
const eventsSearch_js_1 = require("../funcs/eventsSearch.js");
const sdks_js_1 = require("../lib/sdks.js");
const fp_js_1 = require("./types/fp.js");
class Events extends sdks_js_1.ClientSDK {
    /**
     * Load recent events for a particular workspace
     */
    async getEventsByTarget(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, eventsGetEventsByTarget_js_1.eventsGetEventsByTarget)(this, request, options));
    }
    /**
     * Load targets for a particular workspace
     */
    async getTargets(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, eventsGetTargets_js_1.eventsGetTargets)(this, request, options));
    }
    /**
     * Load targets for a particular workspace
     */
    async getTargetsDeprecated(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, eventsGetTargetsDeprecated_js_1.eventsGetTargetsDeprecated)(this, request, options));
    }
    /**
     * Post events for a specific workspace
     *
     * @remarks
     * Sends an array of events to be stored for a particular workspace.
     */
    async post(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, eventsPost_js_1.eventsPost)(this, request, options));
    }
    /**
     * Search events for a particular workspace by any field
     */
    async search(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, eventsSearch_js_1.eventsSearch)(this, request, options));
    }
}
exports.Events = Events;
