// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/github.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Github = void 0;
const githubCheckAccess_js_1 = require("../funcs/githubCheckAccess.js");
const githubCheckPublishingPRs_js_1 = require("../funcs/githubCheckPublishingPRs.js");
const githubCheckPublishingSecrets_js_1 = require("../funcs/githubCheckPublishingSecrets.js");
const githubConfigureCodeSamples_js_1 = require("../funcs/githubConfigureCodeSamples.js");
const githubConfigureMintlifyRepo_js_1 = require("../funcs/githubConfigureMintlifyRepo.js");
const githubConfigureTarget_js_1 = require("../funcs/githubConfigureTarget.js");
const githubGetAction_js_1 = require("../funcs/githubGetAction.js");
const githubGetSetup_js_1 = require("../funcs/githubGetSetup.js");
const githubLinkGithub_js_1 = require("../funcs/githubLinkGithub.js");
const githubStorePublishingSecrets_js_1 = require("../funcs/githubStorePublishingSecrets.js");
const githubTriggerAction_js_1 = require("../funcs/githubTriggerAction.js");
const sdks_js_1 = require("../lib/sdks.js");
const fp_js_1 = require("./types/fp.js");
class Github extends sdks_js_1.ClientSDK {
    async checkAccess(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, githubCheckAccess_js_1.githubCheckAccess)(this, request, options));
    }
    async checkPublishingPRs(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, githubCheckPublishingPRs_js_1.githubCheckPublishingPRs)(this, request, options));
    }
    async checkPublishingSecrets(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, githubCheckPublishingSecrets_js_1.githubCheckPublishingSecrets)(this, request, options));
    }
    async configureCodeSamples(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, githubConfigureCodeSamples_js_1.githubConfigureCodeSamples)(this, request, options));
    }
    async configureMintlifyRepo(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, githubConfigureMintlifyRepo_js_1.githubConfigureMintlifyRepo)(this, request, options));
    }
    async configureTarget(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, githubConfigureTarget_js_1.githubConfigureTarget)(this, request, options));
    }
    async getAction(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, githubGetAction_js_1.githubGetAction)(this, request, options));
    }
    async getSetup(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, githubGetSetup_js_1.githubGetSetup)(this, request, options));
    }
    async linkGithub(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, githubLinkGithub_js_1.githubLinkGithub)(this, request, options));
    }
    async storePublishingSecrets(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, githubStorePublishingSecrets_js_1.githubStorePublishingSecrets)(this, request, options));
    }
    async triggerAction(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, githubTriggerAction_js_1.githubTriggerAction)(this, request, options));
    }
}
exports.Github = Github;
