// @speakeasy-api/speakeasy-client-sdk-typescript/sdk/sdk.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Speakeasy = void 0;
const sdks_js_1 = require("../lib/sdks.js");
const artifacts_js_1 = require("./artifacts.js");
const auth_js_1 = require("./auth.js");
const codesamples_js_1 = require("./codesamples.js");
const events_js_1 = require("./events.js");
const github_js_1 = require("./github.js");
const organizations_js_1 = require("./organizations.js");
const publishingtokens_js_1 = require("./publishingtokens.js");
const reports_js_1 = require("./reports.js");
const schemastore_js_1 = require("./schemastore.js");
const shorturls_js_1 = require("./shorturls.js");
const subscriptions_js_1 = require("./subscriptions.js");
const suggest_js_1 = require("./suggest.js");
const workspaces_js_1 = require("./workspaces.js");
class Speakeasy extends sdks_js_1.ClientSDK {
    get artifacts() {
        return (this._artifacts ?? (this._artifacts = new artifacts_js_1.Artifacts(this._options)));
    }
    get auth() {
        return (this._auth ?? (this._auth = new auth_js_1.Auth(this._options)));
    }
    get codeSamples() {
        return (this._codeSamples ?? (this._codeSamples = new codesamples_js_1.CodeSamples(this._options)));
    }
    get github() {
        return (this._github ?? (this._github = new github_js_1.Github(this._options)));
    }
    get organizations() {
        return (this._organizations ?? (this._organizations = new organizations_js_1.Organizations(this._options)));
    }
    get publishingTokens() {
        return (this._publishingTokens ?? (this._publishingTokens = new publishingtokens_js_1.PublishingTokens(this._options)));
    }
    get reports() {
        return (this._reports ?? (this._reports = new reports_js_1.Reports(this._options)));
    }
    get schemaStore() {
        return (this._schemaStore ?? (this._schemaStore = new schemastore_js_1.SchemaStore(this._options)));
    }
    get shortURLs() {
        return (this._shortURLs ?? (this._shortURLs = new shorturls_js_1.ShortURLs(this._options)));
    }
    get subscriptions() {
        return (this._subscriptions ?? (this._subscriptions = new subscriptions_js_1.Subscriptions(this._options)));
    }
    get suggest() {
        return (this._suggest ?? (this._suggest = new suggest_js_1.Suggest(this._options)));
    }
    get workspaces() {
        return (this._workspaces ?? (this._workspaces = new workspaces_js_1.Workspaces(this._options)));
    }
    get events() {
        return (this._events ?? (this._events = new events_js_1.Events(this._options)));
    }
}
exports.Speakeasy = Speakeasy;
